// @wix/auto_sdk_members_user-member/build/internal/meta.mjs
// src/members-usermembers-v1-user-member-user-member.http.ts
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixMembersUsermembersV1UserMembersUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/user-members-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/user-members",
        destPath: "/v1/user-members"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_user-member";
function queryUserMembers(payload) {
  function __queryUserMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.usermembers.v1.user_member",
      method: "POST",
      methodFqn: "wix.members.usermembers.v1.UserMembers.QueryUserMembers",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersUsermembersV1UserMembersUrl({
        protoPath: "/v1/user-members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "userMembers.createdDate" },
            { path: "userMembers.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryUserMembers;
}

// src/members-usermembers-v1-user-member-user-member.meta.ts
function queryUserMembers2() {
  const payload = {};
  const getRequestOptions = queryUserMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/user-members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  queryUserMembers2 as queryUserMembers
};
