// @wix/auto_sdk_bookings_attendance/build/internal/cjs/src/bookings-v2-attendance-attendance.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.AttendanceStatus = exports.queryAttendance = exports.bulkSetAttendance = exports.setAttendance = exports.getAttendance = exports.__metadata = void 0;
const bookings_v2_attendance_attendance_universal_js_1 = require("./bookings-v2-attendance-attendance.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function getAttendance(httpClient) {
    return (attendanceId) => (0, bookings_v2_attendance_attendance_universal_js_1.getAttendance)(attendanceId, 
    // @ts-ignore
    { httpClient });
}
exports.getAttendance = getAttendance;
function setAttendance(httpClient) {
    return (attendance, options) => (0, bookings_v2_attendance_attendance_universal_js_1.setAttendance)(attendance, options, 
    // @ts-ignore
    { httpClient });
}
exports.setAttendance = setAttendance;
function bulkSetAttendance(httpClient) {
    return (options) => (0, bookings_v2_attendance_attendance_universal_js_1.bulkSetAttendance)(options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkSetAttendance = bulkSetAttendance;
function queryAttendance(httpClient) {
    return () => (0, bookings_v2_attendance_attendance_universal_js_1.queryAttendance)(
    // @ts-ignore
    { httpClient });
}
exports.queryAttendance = queryAttendance;
var bookings_v2_attendance_attendance_universal_js_2 = require("./bookings-v2-attendance-attendance.universal.js");
Object.defineProperty(exports, "AttendanceStatus", { enumerable: true, get: function () { return bookings_v2_attendance_attendance_universal_js_2.AttendanceStatus; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_v2_attendance_attendance_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_v2_attendance_attendance_universal_js_2.WebhookIdentityType; } });
