// @wix/auto_sdk_bookings_resource-types/build/cjs/src/bookings-resources-v2-resource-type-resource-types.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.CreateResourceTypeErrors = exports.RequestedFields = void 0;
var RequestedFields;
(function (RequestedFields) {
    RequestedFields["TOTAL_RESOURCE_COUNT"] = "TOTAL_RESOURCE_COUNT";
    RequestedFields["SPECIFIC_LOCATION_TYPE_RESOURCE_COUNTS"] = "SPECIFIC_LOCATION_TYPE_RESOURCE_COUNTS";
    RequestedFields["DISTINCT_RESOURCE_LOCATIONS"] = "DISTINCT_RESOURCE_LOCATIONS";
})(RequestedFields || (exports.RequestedFields = RequestedFields = {}));
var CreateResourceTypeErrors;
(function (CreateResourceTypeErrors) {
    CreateResourceTypeErrors["UNKNOWN_CREATE_RESOURCE_TYPE_ERROR"] = "UNKNOWN_CREATE_RESOURCE_TYPE_ERROR";
    /** Failed to create requested `quantity` of resources for the resource type */
    CreateResourceTypeErrors["FAILED_TO_CREATE_RESOURCES"] = "FAILED_TO_CREATE_RESOURCES";
})(CreateResourceTypeErrors || (exports.CreateResourceTypeErrors = CreateResourceTypeErrors = {}));
var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (exports.SortOrder = SortOrder = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
