// @wix/auto_sdk_bookings_resource-types/build/cjs/src/bookings-resources-v2-resource-type-resource-types.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.RequestedFields = exports.CreateResourceTypeErrors = exports.onResourceTypeUpdated = exports.onResourceTypeDeleted = exports.onResourceTypeCreated = exports.countResourceTypes = exports.queryResourceTypes = exports.deleteResourceType = exports.updateResourceType = exports.getResourceType = exports.createResourceType = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_resources_v2_resource_type_resource_types_universal_js_1 = require("./bookings-resources-v2-resource-type-resource-types.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function createResourceType(httpClient) {
    return (resourceType) => (0, bookings_resources_v2_resource_type_resource_types_universal_js_1.createResourceType)(resourceType, 
    // @ts-ignore
    { httpClient });
}
exports.createResourceType = createResourceType;
function getResourceType(httpClient) {
    return (resourceTypeId) => (0, bookings_resources_v2_resource_type_resource_types_universal_js_1.getResourceType)(resourceTypeId, 
    // @ts-ignore
    { httpClient });
}
exports.getResourceType = getResourceType;
function updateResourceType(httpClient) {
    return (_id, resourceType) => (0, bookings_resources_v2_resource_type_resource_types_universal_js_1.updateResourceType)(_id, resourceType, 
    // @ts-ignore
    { httpClient });
}
exports.updateResourceType = updateResourceType;
function deleteResourceType(httpClient) {
    return (resourceTypeId) => (0, bookings_resources_v2_resource_type_resource_types_universal_js_1.deleteResourceType)(resourceTypeId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteResourceType = deleteResourceType;
function queryResourceTypes(httpClient) {
    return (options) => (0, bookings_resources_v2_resource_type_resource_types_universal_js_1.queryResourceTypes)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryResourceTypes = queryResourceTypes;
function countResourceTypes(httpClient) {
    return (options) => (0, bookings_resources_v2_resource_type_resource_types_universal_js_1.countResourceTypes)(options, 
    // @ts-ignore
    { httpClient });
}
exports.countResourceTypes = countResourceTypes;
exports.onResourceTypeCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.resources.v2.resource_type_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onResourceTypeDeleted = (0, sdk_types_1.EventDefinition)('wix.bookings.resources.v2.resource_type_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onResourceTypeUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.resources.v2.resource_type_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var bookings_resources_v2_resource_type_resource_types_universal_js_2 = require("./bookings-resources-v2-resource-type-resource-types.universal.js");
Object.defineProperty(exports, "CreateResourceTypeErrors", { enumerable: true, get: function () { return bookings_resources_v2_resource_type_resource_types_universal_js_2.CreateResourceTypeErrors; } });
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return bookings_resources_v2_resource_type_resource_types_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_resources_v2_resource_type_resource_types_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_resources_v2_resource_type_resource_types_universal_js_2.WebhookIdentityType; } });
