// @wix/auto_sdk_bookings_resource-types/build/es/src/bookings-resources-v2-resource-type-resource-types.context.js
import { createResourceType as publicCreateResourceType, getResourceType as publicGetResourceType, updateResourceType as publicUpdateResourceType, deleteResourceType as publicDeleteResourceType, queryResourceTypes as publicQueryResourceTypes, countResourceTypes as publicCountResourceTypes, } from './bookings-resources-v2-resource-type-resource-types.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onResourceTypeCreated as publicOnResourceTypeCreated } from './bookings-resources-v2-resource-type-resource-types.public.js';
import { onResourceTypeDeleted as publicOnResourceTypeDeleted } from './bookings-resources-v2-resource-type-resource-types.public.js';
import { onResourceTypeUpdated as publicOnResourceTypeUpdated } from './bookings-resources-v2-resource-type-resource-types.public.js';
export const createResourceType = /*#__PURE__*/ createRESTModule(publicCreateResourceType);
export const getResourceType = /*#__PURE__*/ createRESTModule(publicGetResourceType);
export const updateResourceType = /*#__PURE__*/ createRESTModule(publicUpdateResourceType);
export const deleteResourceType = /*#__PURE__*/ createRESTModule(publicDeleteResourceType);
export const queryResourceTypes = /*#__PURE__*/ createRESTModule(publicQueryResourceTypes);
export const countResourceTypes = /*#__PURE__*/ createRESTModule(publicCountResourceTypes);
/**
 * Triggered when a resource type is created.
 */
export const onResourceTypeCreated = createEventModule(publicOnResourceTypeCreated);
/**
 * Triggered when a resource type is deleted.
 */
export const onResourceTypeDeleted = createEventModule(publicOnResourceTypeDeleted);
/**
 * Triggered when a resource type is updated.
 */
export const onResourceTypeUpdated = createEventModule(publicOnResourceTypeUpdated);
export { RequestedFields, CreateResourceTypeErrors, SortOrder, WebhookIdentityType, } from './bookings-resources-v2-resource-type-resource-types.universal.js';
