// @wix/auto_sdk_bookings_categories/build/internal/es/src/bookings-v1-category-categories.context.js
import { listCategories as publicListCategories, createCategory as publicCreateCategory, updateCategory as publicUpdateCategory, deleteCategory as publicDeleteCategory, } from './bookings-v1-category-categories.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onCategoryNotification as publicOnCategoryNotification } from './bookings-v1-category-categories.public.js';
export const listCategories = /*#__PURE__*/ createRESTModule(publicListCategories);
export const createCategory = /*#__PURE__*/ createRESTModule(publicCreateCategory);
export const updateCategory = /*#__PURE__*/ createRESTModule(publicUpdateCategory);
export const deleteCategory = /*#__PURE__*/ createRESTModule(publicDeleteCategory);
/** */
export const onCategoryNotification = createEventModule(publicOnCategoryNotification);
export { Status, Event, WebhookIdentityType, } from './bookings-v1-category-categories.universal.js';
