// @wix/auto_sdk_members_authorization/build/internal/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  assignRole: () => assignRole2,
  getRoles: () => getRoles2,
  queryRoles: () => queryRoles2,
  unassignRole: () => unassignRole2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-role-authorization.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersMemberrolesMemberRolesUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/roles",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_authorization";
function assignRole(payload) {
  function __assignRole({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "POST",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.AssignRole",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/{memberId}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __assignRole;
}
function unassignRole(payload) {
  function __unassignRole({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "DELETE",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.UnassignRole",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __unassignRole;
}
function getRoles(payload) {
  function __getRoles({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "GET",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.GetRoles",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getRoles;
}
function queryRoles(payload) {
  function __queryRoles({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "POST",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.QueryRoles",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/query",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryRoles;
}

// src/members-v1-role-authorization.meta.ts
function assignRole2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = assignRole(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/roles/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function unassignRole2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = unassignRole(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/roles/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRoles2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = getRoles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/roles/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRoles2() {
  const payload = {};
  const getRequestOptions = queryRoles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/roles/members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  assignRole,
  getRoles,
  queryRoles,
  unassignRole
});
