// @wix/auto_sdk_members_authorization/build/internal/meta.mjs
// src/members-v1-role-authorization.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersMemberrolesMemberRolesUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/roles",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_authorization";
function assignRole(payload) {
  function __assignRole({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "POST",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.AssignRole",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/{memberId}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __assignRole;
}
function unassignRole(payload) {
  function __unassignRole({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "DELETE",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.UnassignRole",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/{memberId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __unassignRole;
}
function getRoles(payload) {
  function __getRoles({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "GET",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.GetRoles",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/{memberId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getRoles;
}
function queryRoles(payload) {
  function __queryRoles({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.role",
      method: "POST",
      methodFqn: "com.wixpress.members.memberroles.MemberRoles.QueryRoles",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRolesUrl({
        protoPath: "/v1/roles/members/query",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryRoles;
}

// src/members-v1-role-authorization.meta.ts
function assignRole2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = assignRole(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/roles/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function unassignRole2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = unassignRole(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/roles/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRoles2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = getRoles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/roles/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryRoles2() {
  const payload = {};
  const getRequestOptions = queryRoles(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/roles/members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  assignRole2 as assignRole,
  getRoles2 as getRoles,
  queryRoles2 as queryRoles,
  unassignRole2 as unassignRole
};
