// @wix/auto_sdk_bookings_resource-types/build/internal/es/src/bookings-resources-v2-resource-type-resource-types.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { countResourceTypes as universalCountResourceTypes, createResourceType as universalCreateResourceType, deleteResourceType as universalDeleteResourceType, getResourceType as universalGetResourceType, queryResourceTypes as universalQueryResourceTypes, updateResourceType as universalUpdateResourceType, } from './bookings-resources-v2-resource-type-resource-types.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/bookings' };
export function createResourceType(httpClient) {
    return (resourceType) => universalCreateResourceType(resourceType, 
    // @ts-ignore
    { httpClient });
}
export function getResourceType(httpClient) {
    return (resourceTypeId) => universalGetResourceType(resourceTypeId, 
    // @ts-ignore
    { httpClient });
}
export function updateResourceType(httpClient) {
    return (_id, resourceType) => universalUpdateResourceType(_id, resourceType, 
    // @ts-ignore
    { httpClient });
}
export function deleteResourceType(httpClient) {
    return (resourceTypeId) => universalDeleteResourceType(resourceTypeId, 
    // @ts-ignore
    { httpClient });
}
export function queryResourceTypes(httpClient) {
    return (options) => universalQueryResourceTypes(options, 
    // @ts-ignore
    { httpClient });
}
export function countResourceTypes(httpClient) {
    return (options) => universalCountResourceTypes(options, 
    // @ts-ignore
    { httpClient });
}
export const onResourceTypeCreated = EventDefinition('wix.bookings.resources.v2.resource_type_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onResourceTypeDeleted = EventDefinition('wix.bookings.resources.v2.resource_type_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onResourceTypeUpdated = EventDefinition('wix.bookings.resources.v2.resource_type_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export { CreateResourceTypeErrors, RequestedFields, SortOrder, WebhookIdentityType, } from './bookings-resources-v2-resource-type-resource-types.universal.js';
