// @wix/auto_sdk_bookings_resource-types/build/internal/es/src/bookings-resources-v2-resource-type-resource-types.meta.js
import * as ambassadorWixBookingsResourcesV2ResourceType from './bookings-resources-v2-resource-type-resource-types.http.js';
export function createResourceType() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.createResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/resource-types',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getResourceType() {
    const payload = { resourceTypeId: ':resourceTypeId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.getResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/resources/resource-types/{resourceTypeId}',
        pathParams: { resourceTypeId: 'resourceTypeId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateResourceType() {
    const payload = { resourceType: { id: ':resourceTypeId' } };
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.updateResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/resources/resource-types/{resourceType.id}',
        pathParams: { resourceTypeId: 'resourceTypeId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteResourceType() {
    const payload = { resourceTypeId: ':resourceTypeId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.deleteResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/resources/resource-types/{resourceTypeId}',
        pathParams: { resourceTypeId: 'resourceTypeId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryResourceTypes() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.queryResourceTypes(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/resource-types/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countResourceTypes() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.countResourceTypes(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/resource-types/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
