// @wix/auto_sdk_ecom_tax-calculation-provider/build/es/src/interfaces-billing-v1-tax-calculation-provider.public.js
import { ServicePluginDefinition } from '@wix/sdk-types';
import { transformRESTAddressToSDKAddress } from '@wix/sdk-runtime/transformations/address';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
export const provideHandlers = ServicePluginDefinition('TAX_CALCULATION_PROVIDER', [
    {
        name: 'calculateTax',
        primaryHttpMappingPath: '/v1/calculate-tax',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return renameKeysFromSDKRequestToRESTRequest(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = transformPaths(payload, [
                    {
                        transformFn: transformRESTAddressToSDKAddress,
                        paths: [{ path: 'request.addresses', isRepeated: true }],
                    },
                ]);
                return renameKeysFromRESTResponseToSDKResponse(fromRestRequest);
            },
        },
    },
]);
