// @wix/auto_sdk_bookings_extended-bookings/build/internal/es/src/bookings-reader-v2-extended-booking-extended-bookings.meta.js
import * as ambassadorWixBookingsReaderV2ExtendedBooking from './bookings-reader-v2-extended-booking-extended-bookings.http.js';
export function query() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsReaderV2ExtendedBooking.query(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/extended-booking/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryExtendedBookings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsReaderV2ExtendedBooking.queryExtendedBookings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/extended-bookings/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countExtendedBookings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsReaderV2ExtendedBooking.countExtendedBookings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/extended-bookings/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
