// @wix/auto_sdk_bookings_resources/build/internal/es/src/bookings-resources-v2-resource-resources.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { bulkCreateResources as universalBulkCreateResources, bulkDeleteResources as universalBulkDeleteResources, countResources as universalCountResources, createResource as universalCreateResource, deleteResource as universalDeleteResource, getResource as universalGetResource, queryResources as universalQueryResources, searchResources as universalSearchResources, updateResource as universalUpdateResource, } from './bookings-resources-v2-resource-resources.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/bookings' };
export function createResource(httpClient) {
    return (resource) => universalCreateResource(resource, 
    // @ts-ignore
    { httpClient });
}
export function bulkCreateResources(httpClient) {
    return (resources, options) => universalBulkCreateResources(resources, options, 
    // @ts-ignore
    { httpClient });
}
export function getResource(httpClient) {
    return (resourceId) => universalGetResource(resourceId, 
    // @ts-ignore
    { httpClient });
}
export function updateResource(httpClient) {
    return (_id, resource) => universalUpdateResource(_id, resource, 
    // @ts-ignore
    { httpClient });
}
export function deleteResource(httpClient) {
    return (resourceId) => universalDeleteResource(resourceId, 
    // @ts-ignore
    { httpClient });
}
export function bulkDeleteResources(httpClient) {
    return (ids) => universalBulkDeleteResources(ids, 
    // @ts-ignore
    { httpClient });
}
export function searchResources(httpClient) {
    return (options) => universalSearchResources(options, 
    // @ts-ignore
    { httpClient });
}
export function queryResources(httpClient) {
    return () => universalQueryResources(
    // @ts-ignore
    { httpClient });
}
export function countResources(httpClient) {
    return (options) => universalCountResources(options, 
    // @ts-ignore
    { httpClient });
}
export const onResourceCreated = EventDefinition('wix.bookings.resources.v2.resource_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onResourceDeleted = EventDefinition('wix.bookings.resources.v2.resource_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onResourceUpdated = EventDefinition('wix.bookings.resources.v2.resource_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export { AggregationType, Interval, ManagementType, MissingValues, Mode, NestedAggregationType, ScalarType, SortDirection, SortOrder, SortType, WebhookIdentityType, } from './bookings-resources-v2-resource-resources.universal.js';
