// @wix/auto_sdk_bookings_resources/build/internal/es/src/bookings-resources-v2-resource-resources.context.js
import { createResource as publicCreateResource, bulkCreateResources as publicBulkCreateResources, getResource as publicGetResource, updateResource as publicUpdateResource, deleteResource as publicDeleteResource, bulkDeleteResources as publicBulkDeleteResources, searchResources as publicSearchResources, queryResources as publicQueryResources, countResources as publicCountResources, } from './bookings-resources-v2-resource-resources.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onResourceCreated as publicOnResourceCreated } from './bookings-resources-v2-resource-resources.public.js';
import { onResourceDeleted as publicOnResourceDeleted } from './bookings-resources-v2-resource-resources.public.js';
import { onResourceUpdated as publicOnResourceUpdated } from './bookings-resources-v2-resource-resources.public.js';
export const createResource = /*#__PURE__*/ createRESTModule(publicCreateResource);
export const bulkCreateResources = /*#__PURE__*/ createRESTModule(publicBulkCreateResources);
export const getResource = /*#__PURE__*/ createRESTModule(publicGetResource);
export const updateResource = /*#__PURE__*/ createRESTModule(publicUpdateResource);
export const deleteResource = /*#__PURE__*/ createRESTModule(publicDeleteResource);
export const bulkDeleteResources = /*#__PURE__*/ createRESTModule(publicBulkDeleteResources);
export const searchResources = /*#__PURE__*/ createRESTModule(publicSearchResources);
export const queryResources = /*#__PURE__*/ createRESTModule(publicQueryResources);
export const countResources = /*#__PURE__*/ createRESTModule(publicCountResources);
/**
 * Triggered when a resource is created.
 */
export const onResourceCreated = createEventModule(publicOnResourceCreated);
/**
 * Triggered when a resource is deleted.
 */
export const onResourceDeleted = createEventModule(publicOnResourceDeleted);
/**
 * Triggered when a resource is updated.
 */
export const onResourceUpdated = createEventModule(publicOnResourceUpdated);
export { ManagementType, SortOrder, SortType, SortDirection, MissingValues, ScalarType, NestedAggregationType, Interval, AggregationType, Mode, WebhookIdentityType, } from './bookings-resources-v2-resource-resources.universal.js';
