// @wix/auto_sdk_bookings_staff-members/build/es/src/bookings-staff-v1-staff-member-staff-members.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTImageToSDKImage } from '@wix/sdk-runtime/transformations/image';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { assignCustomSchedule as universalAssignCustomSchedule, assignWorkingHoursSchedule as universalAssignWorkingHoursSchedule, bulkUpdateStaffMemberTags as universalBulkUpdateStaffMemberTags, bulkUpdateStaffMemberTagsByFilter as universalBulkUpdateStaffMemberTagsByFilter, connectStaffMemberToUser as universalConnectStaffMemberToUser, countStaffMembers as universalCountStaffMembers, createStaffMember as universalCreateStaffMember, deleteStaffMember as universalDeleteStaffMember, disconnectStaffMemberFromUser as universalDisconnectStaffMemberFromUser, getStaffMember as universalGetStaffMember, queryStaffMembers as universalQueryStaffMembers, searchStaffMembers as universalSearchStaffMembers, updateStaffMember as universalUpdateStaffMember, } from './bookings-staff-v1-staff-member-staff-members.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/bookings' };
export function createStaffMember(httpClient) {
    return (staffMember, options) => universalCreateStaffMember(staffMember, options, 
    // @ts-ignore
    { httpClient });
}
export function getStaffMember(httpClient) {
    return (staffMemberId, options) => universalGetStaffMember(staffMemberId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateStaffMember(httpClient) {
    return (_id, staffMember, options) => universalUpdateStaffMember(_id, staffMember, options, 
    // @ts-ignore
    { httpClient });
}
export function deleteStaffMember(httpClient) {
    return (staffMemberId) => universalDeleteStaffMember(staffMemberId, 
    // @ts-ignore
    { httpClient });
}
export function queryStaffMembers(httpClient) {
    return (options) => universalQueryStaffMembers(options, 
    // @ts-ignore
    { httpClient });
}
export function countStaffMembers(httpClient) {
    return (options) => universalCountStaffMembers(options, 
    // @ts-ignore
    { httpClient });
}
export function connectStaffMemberToUser(httpClient) {
    return (staffMemberId, options) => universalConnectStaffMemberToUser(staffMemberId, options, 
    // @ts-ignore
    { httpClient });
}
export function searchStaffMembers(httpClient) {
    return (search, options) => universalSearchStaffMembers(search, options, 
    // @ts-ignore
    { httpClient });
}
export function disconnectStaffMemberFromUser(httpClient) {
    return (staffMemberId, options) => universalDisconnectStaffMemberFromUser(staffMemberId, options, 
    // @ts-ignore
    { httpClient });
}
export function assignWorkingHoursSchedule(httpClient) {
    return (staffMemberId, scheduleId, options) => universalAssignWorkingHoursSchedule(staffMemberId, scheduleId, options, 
    // @ts-ignore
    { httpClient });
}
export function assignCustomSchedule(httpClient) {
    return (staffMemberId, scheduleId, options) => universalAssignCustomSchedule(staffMemberId, scheduleId, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkUpdateStaffMemberTags(httpClient) {
    return (ids, options) => universalBulkUpdateStaffMemberTags(ids, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkUpdateStaffMemberTagsByFilter(httpClient) {
    return (filter, options) => universalBulkUpdateStaffMemberTagsByFilter(filter, options, 
    // @ts-ignore
    { httpClient });
}
export const onStaffMemberConnectedToUser = EventDefinition('wix.bookings.staff.v1.staff_member_connected_to_user', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.staffMember.createdDate' },
            { path: 'data.staffMember.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'data.staffMember.mainMedia.image' }],
    },
])))();
export const onStaffMemberCreated = EventDefinition('wix.bookings.staff.v1.staff_member_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'entity.mainMedia.image' }],
    },
])))();
export const onStaffMemberDeleted = EventDefinition('wix.bookings.staff.v1.staff_member_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'entity.mainMedia.image' }],
    },
])))();
export const onStaffMemberDisconnectedFromUser = EventDefinition('wix.bookings.staff.v1.staff_member_disconnected_from_user', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.staffMember.createdDate' },
            { path: 'data.staffMember.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'data.staffMember.mainMedia.image' }],
    },
])))();
export const onStaffMemberFullyCreated = EventDefinition('wix.bookings.staff.v1.staff_member_fully_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.staffMember.createdDate' },
            { path: 'data.staffMember.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'data.staffMember.mainMedia.image' }],
    },
])))();
export const onStaffMemberUpdated = EventDefinition('wix.bookings.staff.v1.staff_member_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'entity.mainMedia.image' }],
    },
])))();
export { AggregationType, ApprovalStatus, AssociatedWixIdentityConnectionStatusEnumConnectionStatus, CalendarType, ConferenceType, ConnectionStatus, Day, DayOfWeek, IdentityType, LocationStatus, LocationType, LocationsLocationType, Mode, RecurringIntervalType, RequestedFields, ScalarType, ScheduleStatus, SessionType, SortOrder, Status, Transparency, WebhookIdentityType, } from './bookings-staff-v1-staff-member-staff-members.universal.js';
