// @wix/auto_sdk_bookings_staff-members/build/es/src/bookings-staff-v1-staff-member-staff-members.meta.js
import * as ambassadorWixBookingsStaffV1StaffMember from './bookings-staff-v1-staff-member-staff-members.http.js';
export function createStaffMember() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.createStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getStaffMember() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.getStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/staff-members/{staffMemberId}',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateStaffMember() {
    const payload = { staffMember: { id: ':staffMemberId' } };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.updateStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/staff-members/{staffMember.id}',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteStaffMember() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.deleteStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/staff-members/{staffMemberId}',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryStaffMembers() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.queryStaffMembers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countStaffMembers() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.countStaffMembers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function connectStaffMemberToUser() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.connectStaffMemberToUser(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/connect-staff-member-to-user',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchStaffMembers() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.searchStaffMembers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function disconnectStaffMemberFromUser() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.disconnectStaffMemberFromUser(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/disconnect-staff-member-from-user',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function assignWorkingHoursSchedule() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.assignWorkingHoursSchedule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/assign-working-hours-schedule',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function assignCustomSchedule() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.assignCustomSchedule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/assign-custom-schedule',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateStaffMemberTags() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.bulkUpdateStaffMemberTags(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/staff-members/update-tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateStaffMemberTagsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.bulkUpdateStaffMemberTagsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/staff-members/update-tags-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
