// @wix/auto_sdk_bookings_attendance/build/internal/es/src/bookings-v2-attendance-attendance.meta.js
import * as ambassadorWixBookingsV2Attendance from './bookings-v2-attendance-attendance.http.js';
export function getAttendance() {
    const payload = { attendanceId: ':attendanceId' };
    const getRequestOptions = ambassadorWixBookingsV2Attendance.getAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/attendance/{attendanceId}',
        pathParams: { attendanceId: 'attendanceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setAttendance() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Attendance.setAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/attendance/set',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkSetAttendance() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Attendance.bulkSetAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/attendance/set',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryAttendance() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Attendance.queryAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/attendance/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
