// @wix/auto_sdk_bookings_resources/build/es/src/bookings-resources-v2-resource-resources.meta.js
import * as ambassadorWixBookingsResourcesV2Resource from './bookings-resources-v2-resource-resources.http.js';
export function createResource() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.createResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.bulkCreateResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/resources/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getResource() {
    const payload = { resourceId: ':resourceId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.getResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/resources/{resourceId}',
        pathParams: { resourceId: 'resourceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateResource() {
    const payload = { resource: { id: ':resourceId' } };
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.updateResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/resources/{resource.id}',
        pathParams: { resourceId: 'resourceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteResource() {
    const payload = { resourceId: ':resourceId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.deleteResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/resources/{resourceId}',
        pathParams: { resourceId: 'resourceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.bulkDeleteResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/resources/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.searchResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.queryResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.countResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
