// @wix/auto_sdk_members_member-report/build/meta.mjs
// src/members-v1-member-report-member-report.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersMemberreportsMemberReportsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members-reports-api",
        destPath: ""
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    _: [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-report";
function reportMember(payload) {
  function __reportMember({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "memberReport.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.ReportMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberReport.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __reportMember;
}
function queryMemberReports(payload) {
  function __queryMemberReports({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.QueryMemberReports",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberReports.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryMemberReports;
}
function listReportedMembers(payload) {
  function __listReportedMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "GET",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.ListReportedMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "reportedMembers.lastReportDate" }]
        }
      ])
    };
    return metadata;
  }
  return __listReportedMembers;
}
function queryReportedMembers(payload) {
  function __queryReportedMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.QueryReportedMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "reportedMembers.lastReportDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryReportedMembers;
}
function deleteMemberReports(payload) {
  function __deleteMemberReports({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "DELETE",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.DeleteMemberReports",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members/{memberId}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteMemberReports;
}

// src/members-v1-member-report-member-report.meta.ts
function reportMember2() {
  const payload = {};
  const getRequestOptions = reportMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/member-reports",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMemberReports2() {
  const payload = {};
  const getRequestOptions = queryMemberReports(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/member-reports/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listReportedMembers2() {
  const payload = {};
  const getRequestOptions = listReportedMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/member-reports/members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryReportedMembers2() {
  const payload = {};
  const getRequestOptions = queryReportedMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/member-reports/members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberReports2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = deleteMemberReports(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/member-reports/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  deleteMemberReports2 as deleteMemberReports,
  listReportedMembers2 as listReportedMembers,
  queryMemberReports2 as queryMemberReports,
  queryReportedMembers2 as queryReportedMembers,
  reportMember2 as reportMember
};
