// @wix/auto_sdk_bookings_service-options-and-variants/build/internal/es/src/bookings-catalog-v1-service-options-and-variants-service-options-and-variants.meta.js
import * as ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants from './bookings-catalog-v1-service-options-and-variants-service-options-and-variants.http.js';
export function createServiceOptionsAndVariants() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants.createServiceOptionsAndVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/serviceOptionsAndVariants',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cloneServiceOptionsAndVariants() {
    const payload = { cloneFromId: ':cloneFromId' };
    const getRequestOptions = ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants.cloneServiceOptionsAndVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/serviceOptionsAndVariants/{cloneFromId}/clone',
        pathParams: { cloneFromId: 'cloneFromId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getServiceOptionsAndVariants() {
    const payload = {
        serviceOptionsAndVariantsId: ':serviceOptionsAndVariantsId',
    };
    const getRequestOptions = ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants.getServiceOptionsAndVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/serviceOptionsAndVariants/{serviceOptionsAndVariantsId}',
        pathParams: { serviceOptionsAndVariantsId: 'serviceOptionsAndVariantsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getServiceOptionsAndVariantsByServiceId() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants.getServiceOptionsAndVariantsByServiceId(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/serviceOptionsAndVariants/service_id/{serviceId}',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateServiceOptionsAndVariants() {
    const payload = {
        serviceOptionsAndVariants: { id: ':serviceOptionsAndVariantsId' },
    };
    const getRequestOptions = ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants.updateServiceOptionsAndVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/serviceOptionsAndVariants/{serviceOptionsAndVariants.id}',
        pathParams: { serviceOptionsAndVariantsId: 'serviceOptionsAndVariantsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteServiceOptionsAndVariants() {
    const payload = {
        serviceOptionsAndVariantsId: ':serviceOptionsAndVariantsId',
    };
    const getRequestOptions = ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants.deleteServiceOptionsAndVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/serviceOptionsAndVariants/{serviceOptionsAndVariantsId}',
        pathParams: { serviceOptionsAndVariantsId: 'serviceOptionsAndVariantsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryServiceOptionsAndVariants() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCatalogV1ServiceOptionsAndVariants.queryServiceOptionsAndVariants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/serviceOptionsAndVariants/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
