// @wix/auto_sdk_members_default-privacy/build/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Privacy: () => Privacy,
  getDefaultPrivacyStatus: () => getDefaultPrivacyStatus4,
  setDefaultPrivacyStatus: () => setDefaultPrivacyStatus4
});
module.exports = __toCommonJS(index_exports);

// src/members-v1-default-privacy-default-privacy.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/members-v1-default-privacy-default-privacy.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members-default-privacy-status-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_default-privacy";
function getDefaultPrivacyStatus(payload) {
  function __getDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "GET",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.GetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getDefaultPrivacyStatus;
}
function setDefaultPrivacyStatus(payload) {
  function __setDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "PATCH",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.SetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __setDefaultPrivacyStatus;
}

// src/members-v1-default-privacy-default-privacy.universal.ts
var Privacy = /* @__PURE__ */ ((Privacy2) => {
  Privacy2["PRIVATE"] = "PRIVATE";
  Privacy2["PUBLIC"] = "PUBLIC";
  return Privacy2;
})(Privacy || {});
async function getDefaultPrivacyStatus2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getDefaultPrivacyStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function setDefaultPrivacyStatus2(defaultPrivacy) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    defaultPrivacy
  });
  const reqOpts = setDefaultPrivacyStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { defaultPrivacy: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["defaultPrivacy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/members-v1-default-privacy-default-privacy.public.ts
function getDefaultPrivacyStatus3(httpClient) {
  return () => getDefaultPrivacyStatus2(
    // @ts-ignore
    { httpClient }
  );
}
function setDefaultPrivacyStatus3(httpClient) {
  return (defaultPrivacy) => setDefaultPrivacyStatus2(
    defaultPrivacy,
    // @ts-ignore
    { httpClient }
  );
}

// src/members-v1-default-privacy-default-privacy.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var getDefaultPrivacyStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getDefaultPrivacyStatus3);
var setDefaultPrivacyStatus4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(setDefaultPrivacyStatus3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Privacy,
  getDefaultPrivacyStatus,
  setDefaultPrivacyStatus
});
