// @wix/auto_sdk_members_default-privacy/build/meta.mjs
// src/members-v1-default-privacy-default-privacy.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members-default-privacy-status-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_default-privacy";
function getDefaultPrivacyStatus(payload) {
  function __getDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "GET",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.GetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getDefaultPrivacyStatus;
}
function setDefaultPrivacyStatus(payload) {
  function __setDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "PATCH",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.SetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __setDefaultPrivacyStatus;
}

// src/members-v1-default-privacy-default-privacy.meta.ts
function getDefaultPrivacyStatus2() {
  const payload = {};
  const getRequestOptions = getDefaultPrivacyStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/default-privacy-status",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setDefaultPrivacyStatus2() {
  const payload = {};
  const getRequestOptions = setDefaultPrivacyStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/default-privacy-status",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getDefaultPrivacyStatus2 as getDefaultPrivacyStatus,
  setDefaultPrivacyStatus2 as setDefaultPrivacyStatus
};
