// @wix/auto_sdk_bookings_availability-calendar/build/internal/cjs/src/bookings-availability-v1-slot-availability-availability-calendar.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.calculateMultiSlotAvailability = exports.getScheduleAvailability = exports.queryAvailability = void 0;
const ambassadorWixBookingsAvailabilityV1SlotAvailability = __importStar(require("./bookings-availability-v1-slot-availability-availability-calendar.http.js"));
function queryAvailability() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsAvailabilityV1SlotAvailability.queryAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/availability/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryAvailability = queryAvailability;
function getScheduleAvailability() {
    const payload = { scheduleId: ':scheduleId' };
    const getRequestOptions = ambassadorWixBookingsAvailabilityV1SlotAvailability.getScheduleAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/availability/schedule/{scheduleId}',
        pathParams: { scheduleId: 'scheduleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getScheduleAvailability = getScheduleAvailability;
function calculateMultiSlotAvailability() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsAvailabilityV1SlotAvailability.calculateMultiSlotAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/availability/multislot',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.calculateMultiSlotAvailability = calculateMultiSlotAvailability;
