// @wix/auto_sdk_bookings_availability-calendar/build/internal/cjs/src/bookings-availability-v1-slot-availability-availability-calendar.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationLocationType = exports.SortOrder = exports.LocationType = void 0;
var LocationType;
(function (LocationType) {
    LocationType["UNDEFINED"] = "UNDEFINED";
    LocationType["OWNER_BUSINESS"] = "OWNER_BUSINESS";
    LocationType["OWNER_CUSTOM"] = "OWNER_CUSTOM";
    LocationType["CUSTOM"] = "CUSTOM";
})(LocationType || (exports.LocationType = LocationType = {}));
var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (exports.SortOrder = SortOrder = {}));
var LocationLocationType;
(function (LocationLocationType) {
    LocationLocationType["UNKNOWN_LOCATION_TYPE"] = "UNKNOWN_LOCATION_TYPE";
    /** A business location, either the default business address, or locations defined for the business by the Business Info. */
    LocationLocationType["BUSINESS"] = "BUSINESS";
    /** The location is unique to this service and isn't defined as one of the business locations. */
    LocationLocationType["CUSTOM"] = "CUSTOM";
    /** The location can be determined by the customer and is not set up beforehand. */
    LocationLocationType["CUSTOMER"] = "CUSTOMER";
})(LocationLocationType || (exports.LocationLocationType = LocationLocationType = {}));
