// @wix/auto_sdk_bookings_pricing/build/cjs/src/bookings-v2-price-info-pricing.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueType = exports.SelectedPaymentOption = exports.Platform = exports.PaymentStatus = exports.MultiServiceBookingType = exports.LocationType = exports.IdentityType = exports.BookingStatus = exports.Actor = exports.calculatePrice = exports.previewPrice = exports.__metadata = void 0;
const bookings_v2_price_info_pricing_universal_js_1 = require("./bookings-v2-price-info-pricing.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function previewPrice(httpClient) {
    return (bookingLineItems) => (0, bookings_v2_price_info_pricing_universal_js_1.previewPrice)(bookingLineItems, 
    // @ts-ignore
    { httpClient });
}
exports.previewPrice = previewPrice;
function calculatePrice(httpClient) {
    return (booking) => (0, bookings_v2_price_info_pricing_universal_js_1.calculatePrice)(booking, 
    // @ts-ignore
    { httpClient });
}
exports.calculatePrice = calculatePrice;
var bookings_v2_price_info_pricing_universal_js_2 = require("./bookings-v2-price-info-pricing.universal.js");
Object.defineProperty(exports, "Actor", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.Actor; } });
Object.defineProperty(exports, "BookingStatus", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.BookingStatus; } });
Object.defineProperty(exports, "IdentityType", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.IdentityType; } });
Object.defineProperty(exports, "LocationType", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.LocationType; } });
Object.defineProperty(exports, "MultiServiceBookingType", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.MultiServiceBookingType; } });
Object.defineProperty(exports, "PaymentStatus", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.PaymentStatus; } });
Object.defineProperty(exports, "Platform", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.Platform; } });
Object.defineProperty(exports, "SelectedPaymentOption", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.SelectedPaymentOption; } });
Object.defineProperty(exports, "ValueType", { enumerable: true, get: function () { return bookings_v2_price_info_pricing_universal_js_2.ValueType; } });
