// @wix/auto_sdk_bookings_availability-time-slots/build/cjs/src/bookings-availability-v2-time-slot-availability-time-slots.universal.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAvailabilityTimeSlot = exports.listAvailabilityTimeSlots = exports.LocationType = void 0;
const transform_error_1 = require("@wix/sdk-runtime/transform-error");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const ambassadorWixBookingsAvailabilityV2TimeSlot = __importStar(require("./bookings-availability-v2-time-slot-availability-time-slots.http.js"));
var LocationType;
(function (LocationType) {
    LocationType["UNKNOWN_LOCATION_TYPE"] = "UNKNOWN_LOCATION_TYPE";
    /** A business location, either the default business address, or locations defined for the business by the Business Info. */
    LocationType["BUSINESS"] = "BUSINESS";
    /** The location is unique to this service and isn't defined as one of the business locations. */
    LocationType["CUSTOM"] = "CUSTOM";
    /** The location can be determined by the customer and is not set up beforehand. */
    LocationType["CUSTOMER"] = "CUSTOMER";
})(LocationType || (exports.LocationType = LocationType = {}));
/**
 * Retrieves a list of `TimeSlot`s that match the provided filters.
 *
 * <blockquote class="important">
 * <p><strong>Important:</strong>
 * Currently supported only for services of type <code class="grey-background">APPOINTMENT</code>.</p>
 * </blockquote>
 *
 * The request body __must__ include either:
 * + All of the following filters:  `serviceId`, `fromLocalDate`, `toLocalDate`, and `timeZone`.
 * You may add additional filters as you wish.
 * + A `cursorPaging` with a valid `cursor` from previous response.
 *
 *
 * Each [TimeSlot](https://bo.wix.com/wix-docs/rest/all-apis/wix-service-availability/availability-time-slots/time-slot-object) in response
 * represents the availability of the service in a specific location,
 * and has a `localStartDate` within the range of the provided `fromLocalDate` and `toLocalDate` exclusive.
 * The `localEndDate` of a `TimeSlot` is calculated as the sum of the `TimeSlot`'s `localStartDate` and the duration of the service.
 *
 * By default,
 * the response contains at most 1000 results.
 * If there are more than 1000 results, we return a `cursorPagingMetadata` with
 * a cursor for the next page of results, regardless of whether a `cursorPaging`
 * was provided in request.
 *
 * ### AvailableResources in response
 * The `TimeSlot`'s `AvailableResources` contains information about the resources that are available to provide the service
 * within the time slot. Each `AvailableResources` contains information about available resources of the same type.
 *
 * <blockquote class="important">
 * <p><strong>Important:</strong><br/>
 * By default,
 * if you don't specify <code class="grey-background">includeResourceTypeIds</code>
 * or <code class="grey-background">resourceIds</code> filters in request,
 * we return TimeSlots with an empty <code class="grey-background">AvailableResources</code>.
 * <br/><strong>Note:</strong><br/> Not specifying resources filters can be handy in case you want to avoid large response in flows that only
 * interested of whether the time slots are available. [Finding the next available slot within the next 3 months](https://bo.wix.com/wix-docs/rest/all-apis/service-availability/availability-time-slots/sample-flows?localViewerId=inbari#all-apis_service-availability_availability-time-slots_sample-flows_find-the-first-date-within-the-next-3-months-that-the-selected-service-is-available-for)
 * is an example for such flow.
 * </p>
 * </blockquote>
 *
 * If you wish to get a list of available resources for each `TimeSlot` you should either:
 * + provide `resourceIds` in request.
 * + provide `includeResourceTypeIds` in request.
 *
 * __Notes:__
 * + In both cases the returned TimeSlots contains  __up__ to 10 `AvailableResources` that match the provided filters.
 * Each `AvailableResources` contains __up__ to 10 available `resources` of the same type that match the provided filters.
 * + If an `AvailableResources` has more available resources which are not listed within it,
 * we return `AvailableResources`.`hasMoreAvailableResources` as true.
 * > __Note:__
 * > If you wish to get the full available resources list for a specific `TimeSlot`,
 * > you should call [GetAvailabilityTimeSlot](https://bo.wix.com/wix-docs/rest/all-apis/wix-service-availability/service-availability-time-slots/get-availability-time-slot).
 *
 *
 * ### Availability VS Bookability
 * An `available` time slot is not necessarily `bookable`.
 * The `bookable` field of a `TimeSlot` indicates whether a customer can book the service within the given time slot,
 * at a specific period of time.
 * Read more about [Availability VS Bookability](https://bo.wix.com/wix-docs/rest/all-apis/wix-service-availability/service-availability-time-slots/introduction#all-apis_wix-service-availability_service-availability-time-slots_introduction_availability-vs-bookability).
 *
 * By default,
 * + The response does not contains `unavailable` TimeSlots.
 * For example,
 * if there are no available resources to provide the service from `2024-01-30T14:30:00` to `2024-01-30T15:30:00`,
 * we don't return TimeSlots with `localStartDate` within this range.
 * + The response contains both `bookable` and `un-bookable` TimeSlots.
 * For example,
 * if the service has a booking policy which enforces booking the service up to 10 minutes before the session starts,
 * we return TimeSlots with the violating `localStartDate` with `bookable` as false.
 * If you wish to list only available __bookable__ TimeSlots you should pass `bookable` filter as true.
 *
 * <blockquote class="important">
 * <p><strong>Important:</strong>
 * Because of DST, there are edge cases where certain times either do not exist or exist twice for a local time zone.
 * Read more about <a href="https://bo.wix.com/wix-docs/rest/all-apis/service-availability/availability-time-slots/introduction?localViewerId=inbari#all-apis_service-availability_availability-time-slots_introduction_daylight-saving-time-dst-handling">DST Handling</a></p>
 * </blockquote>
 *
 *
 * ### ListAvailabilityTimeSlots runs with the following defaults
 * + `localStartDate` is sorted in `ASC` order
 * + `cursorPaging`.`limit` is `1000`
 * @public
 * @documentationMaturity preview
 * @permissionId BOOKINGS.AVAILABILITY_READ_TIME_SLOTS
 * @permissionScope Read Bookings - Public Data
 * @permissionScopeId SCOPE.DC-BOOKINGS.READ-BOOKINGS-PUBLIC
 * @permissionScope Manage Bookings Services and Settings
 * @permissionScopeId SCOPE.BOOKINGS.CONFIGURATION
 * @permissionScope Manage Bookings
 * @permissionScopeId SCOPE.DC-BOOKINGS.MANAGE-BOOKINGS
 * @permissionScope Read Bookings - Including Participants
 * @permissionScopeId SCOPE.DC-BOOKINGS.READ-BOOKINGS-SENSITIVE
 * @permissionScope Read Bookings - all read permissions
 * @permissionScopeId SCOPE.DC-BOOKINGS-MEGA.READ-BOOKINGS
 * @permissionScope Read Bookings Calendar Availability
 * @permissionScopeId SCOPE.DC-BOOKINGS.READ-CALENDAR
 * @permissionScope Manage Bookings - all permissions
 * @permissionScopeId SCOPE.DC-BOOKINGS-MEGA.MANAGE-BOOKINGS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn com.wixpress.bookings.availability.v2.AvailabilityTimeSlots.ListAvailabilityTimeSlots
 */
async function listAvailabilityTimeSlots(options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        serviceId: options?.serviceId,
        fromLocalDate: options?.fromLocalDate,
        toLocalDate: options?.toLocalDate,
        timeZone: options?.timeZone,
        locations: options?.locations,
        resourceIds: options?.resourceIds,
        includeResourceTypeIds: options?.includeResourceTypeIds,
        bookable: options?.bookable,
        bookingPolicyViolations: options?.bookingPolicyViolations,
        timeSlotsPerDay: options?.timeSlotsPerDay,
        cursorPaging: options?.cursorPaging,
        resourceTypes: options?.resourceTypes,
    });
    const reqOpts = ambassadorWixBookingsAvailabilityV2TimeSlot.listAvailabilityTimeSlots(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                serviceId: '$[0].serviceId',
                fromLocalDate: '$[0].fromLocalDate',
                toLocalDate: '$[0].toLocalDate',
                timeZone: '$[0].timeZone',
                locations: '$[0].locations',
                resourceIds: '$[0].resourceIds',
                includeResourceTypeIds: '$[0].includeResourceTypeIds',
                bookable: '$[0].bookable',
                bookingPolicyViolations: '$[0].bookingPolicyViolations',
                timeSlotsPerDay: '$[0].timeSlotsPerDay',
                cursorPaging: '$[0].cursorPaging',
                resourceTypes: '$[0].resourceTypes',
            },
            singleArgumentUnchanged: false,
        }, ['options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.listAvailabilityTimeSlots = listAvailabilityTimeSlots;
/**
 * Retrieves an available `TimeSlot` that match the provided filters.
 *
 * Throws `SlotNotFound` if there is no such available time slot.
 *
 * <blockquote class="important">
 * <p><strong>Important:</strong><br/>
 * Currently supported only for services of type <code class="grey-background">APPOINTMENT</code>.</p>
 * </blockquote>
 *
 * By default,
 * if you don't provide `includeResourceTypeIds` or `resourceIds` in request,
 * we return all `AvailableResources` with all `AvailableResources`.`resources` which are available to provide
 * the service within the time slot.
 *
 * If you specify `includeResourceTypeIds` or `resourceIds` in request,
 * the returned `TimeSlot` will contain only `AvailableResources` with at least one available resource
 * which match the given resources filters,
 * each contains all available resources out of those requested.
 *
 *
 * <blockquote class="tip">
 * <p><strong>Tip:</strong><br/>
 * Use this API in order to get the availability of a specific <code class="grey-background">TimeSlot</code> out of those returned from  <code class="grey-background">ListAvailabilityTimeSlots</code> API.</p>
 * </blockquote>
 * @param serviceId - Service ID of the time slot.
 * Currently supported only for services of type `APPOINTMENT`.
 * @param localStartDate - Local start date of the time slot, in ISO-8601 format.
 * For example, "2024-01-30T13:30:00".
 * @param localEndDate - Local end date of the time slot, in ISO-8601 format.
 * For example, "2024-01-30T14:30:00".
 * @param timeZone - Time zone, in IANA time zone format.
 * @param location - The location of the time slot.
 *
 * You must provide a specific `locationType`.
 * If locationType is `BUSINESS`, you __must__ also provide a `locationId`.
 *
 * <blockquote class="warning">
 * <p>Supports filtering by location type, or by location ID.
 * Other fields like <code class="grey-background">name</code> are ignored.</p>
 * </blockquote>
 * @public
 * @documentationMaturity preview
 * @requiredField localEndDate
 * @requiredField localStartDate
 * @requiredField location
 * @requiredField serviceId
 * @requiredField timeZone
 * @permissionId BOOKINGS.AVAILABILITY_READ_TIME_SLOTS
 * @permissionScope Read Bookings - Public Data
 * @permissionScopeId SCOPE.DC-BOOKINGS.READ-BOOKINGS-PUBLIC
 * @permissionScope Manage Bookings Services and Settings
 * @permissionScopeId SCOPE.BOOKINGS.CONFIGURATION
 * @permissionScope Manage Bookings
 * @permissionScopeId SCOPE.DC-BOOKINGS.MANAGE-BOOKINGS
 * @permissionScope Read Bookings - Including Participants
 * @permissionScopeId SCOPE.DC-BOOKINGS.READ-BOOKINGS-SENSITIVE
 * @permissionScope Read Bookings - all read permissions
 * @permissionScopeId SCOPE.DC-BOOKINGS-MEGA.READ-BOOKINGS
 * @permissionScope Read Bookings Calendar Availability
 * @permissionScopeId SCOPE.DC-BOOKINGS.READ-CALENDAR
 * @permissionScope Manage Bookings - all permissions
 * @permissionScopeId SCOPE.DC-BOOKINGS-MEGA.MANAGE-BOOKINGS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn com.wixpress.bookings.availability.v2.AvailabilityTimeSlots.GetAvailabilityTimeSlot
 */
async function getAvailabilityTimeSlot(serviceId, localStartDate, localEndDate, timeZone, location, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[6];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        serviceId: serviceId,
        localStartDate: localStartDate,
        localEndDate: localEndDate,
        timeZone: timeZone,
        location: location,
        resourceIds: options?.resourceIds,
        includeResourceTypeIds: options?.includeResourceTypeIds,
    });
    const reqOpts = ambassadorWixBookingsAvailabilityV2TimeSlot.getAvailabilityTimeSlot(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                serviceId: '$[0]',
                localStartDate: '$[1]',
                localEndDate: '$[2]',
                timeZone: '$[3]',
                location: '$[4]',
                resourceIds: '$[5].resourceIds',
                includeResourceTypeIds: '$[5].includeResourceTypeIds',
            },
            singleArgumentUnchanged: false,
        }, [
            'serviceId',
            'localStartDate',
            'localEndDate',
            'timeZone',
            'location',
            'options',
        ]);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.getAvailabilityTimeSlot = getAvailabilityTimeSlot;
