// @wix/auto_sdk_ecom_shipping-options/build/internal/cjs/src/ecom-v1-shipping-option-shipping-options.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.LogicalOperator = exports.ConditionType = exports.updateExtendedFields = exports.queryShippingOptions = exports.deleteShippingOption = exports.updateShippingOption = exports.getShippingOption = exports.createShippingOption = exports.__metadata = void 0;
const ecom_v1_shipping_option_shipping_options_universal_js_1 = require("./ecom-v1-shipping-option-shipping-options.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createShippingOption(httpClient) {
    return (shippingOption) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.createShippingOption)(shippingOption, 
    // @ts-ignore
    { httpClient });
}
exports.createShippingOption = createShippingOption;
function getShippingOption(httpClient) {
    return (shippingOptionId) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.getShippingOption)(shippingOptionId, 
    // @ts-ignore
    { httpClient });
}
exports.getShippingOption = getShippingOption;
function updateShippingOption(httpClient) {
    return (_id, shippingOption) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.updateShippingOption)(_id, shippingOption, 
    // @ts-ignore
    { httpClient });
}
exports.updateShippingOption = updateShippingOption;
function deleteShippingOption(httpClient) {
    return (shippingOptionId) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.deleteShippingOption)(shippingOptionId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteShippingOption = deleteShippingOption;
function queryShippingOptions(httpClient) {
    return () => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.queryShippingOptions)(
    // @ts-ignore
    { httpClient });
}
exports.queryShippingOptions = queryShippingOptions;
function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => (0, ecom_v1_shipping_option_shipping_options_universal_js_1.updateExtendedFields)(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateExtendedFields = updateExtendedFields;
var ecom_v1_shipping_option_shipping_options_universal_js_2 = require("./ecom-v1-shipping-option-shipping-options.universal.js");
Object.defineProperty(exports, "ConditionType", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.ConditionType; } });
Object.defineProperty(exports, "LogicalOperator", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.LogicalOperator; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return ecom_v1_shipping_option_shipping_options_universal_js_2.WebhookIdentityType; } });
