// @wix/auto_sdk_members_default-privacy/build/internal/index.mjs
// src/members-v1-default-privacy-default-privacy.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/members-v1-default-privacy-default-privacy.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members-default-privacy-status-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_default-privacy";
function getDefaultPrivacyStatus(payload) {
  function __getDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "GET",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.GetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getDefaultPrivacyStatus;
}
function setDefaultPrivacyStatus(payload) {
  function __setDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "PATCH",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.SetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __setDefaultPrivacyStatus;
}

// src/members-v1-default-privacy-default-privacy.universal.ts
var Privacy = /* @__PURE__ */ ((Privacy2) => {
  Privacy2["PRIVATE"] = "PRIVATE";
  Privacy2["PUBLIC"] = "PUBLIC";
  return Privacy2;
})(Privacy || {});
async function getDefaultPrivacyStatus2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = getDefaultPrivacyStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function setDefaultPrivacyStatus2(defaultPrivacy) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    defaultPrivacy
  });
  const reqOpts = setDefaultPrivacyStatus(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { defaultPrivacy: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["defaultPrivacy"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/members-v1-default-privacy-default-privacy.public.ts
function getDefaultPrivacyStatus3(httpClient) {
  return () => getDefaultPrivacyStatus2(
    // @ts-ignore
    { httpClient }
  );
}
function setDefaultPrivacyStatus3(httpClient) {
  return (defaultPrivacy) => setDefaultPrivacyStatus2(
    defaultPrivacy,
    // @ts-ignore
    { httpClient }
  );
}

// src/members-v1-default-privacy-default-privacy.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getDefaultPrivacyStatus4 = /* @__PURE__ */ createRESTModule(getDefaultPrivacyStatus3);
var setDefaultPrivacyStatus4 = /* @__PURE__ */ createRESTModule(setDefaultPrivacyStatus3);
export {
  Privacy,
  getDefaultPrivacyStatus4 as getDefaultPrivacyStatus,
  setDefaultPrivacyStatus4 as setDefaultPrivacyStatus
};
