// @wix/auto_sdk_members_default-privacy/build/internal/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getDefaultPrivacyStatus: () => getDefaultPrivacyStatus2,
  setDefaultPrivacyStatus: () => setDefaultPrivacyStatus2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-default-privacy-default-privacy.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members-default-privacy-status-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/default-privacy-status",
        destPath: "/v1/default-privacy-status"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_default-privacy";
function getDefaultPrivacyStatus(payload) {
  function __getDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "GET",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.GetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getDefaultPrivacyStatus;
}
function setDefaultPrivacyStatus(payload) {
  function __setDefaultPrivacyStatus({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.default_privacy",
      method: "PATCH",
      methodFqn: "com.wixpress.members.defaultprivacy.DefaultPrivacyStatus.SetDefaultPrivacyStatus",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersDefaultprivacyDefaultPrivacyStatusUrl({
        protoPath: "/v1/default-privacy-status",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __setDefaultPrivacyStatus;
}

// src/members-v1-default-privacy-default-privacy.meta.ts
function getDefaultPrivacyStatus2() {
  const payload = {};
  const getRequestOptions = getDefaultPrivacyStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/default-privacy-status",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setDefaultPrivacyStatus2() {
  const payload = {};
  const getRequestOptions = setDefaultPrivacyStatus(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/default-privacy-status",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getDefaultPrivacyStatus,
  setDefaultPrivacyStatus
});
