// @wix/auto_sdk_members_custom-field-suggestions/build/index.mjs
// src/members-v1-custom-field-suggestion-custom-field-suggestions.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import { queryBuilder } from "@wix/sdk-runtime/query-builder";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/members-v1-custom-field-suggestion-custom-field-suggestions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members-custom-fields-suggestions-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-field-suggestions";
function listCustomFieldSuggestions(payload) {
  function __listCustomFieldSuggestions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_suggestion",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.suggestions.CustomFieldSuggestions.ListCustomFieldSuggestions",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(
        { protoPath: "/v1/custom-field-suggestions", data: payload, host }
      ),
      params: toURLSearchParams(payload, true)
    };
    return metadata;
  }
  return __listCustomFieldSuggestions;
}
function queryCustomFieldSuggestions(payload) {
  function __queryCustomFieldSuggestions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_suggestion",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.suggestions.CustomFieldSuggestions.QueryCustomFieldSuggestions",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(
        { protoPath: "/v1/custom-field-suggestions/query", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __queryCustomFieldSuggestions;
}

// src/members-v1-custom-field-suggestion-custom-field-suggestions.universal.ts
var Privacy = /* @__PURE__ */ ((Privacy2) => {
  Privacy2["UNKNOWN"] = "UNKNOWN";
  Privacy2["PUBLIC"] = "PUBLIC";
  Privacy2["PRIVATE"] = "PRIVATE";
  return Privacy2;
})(Privacy || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["TEXT"] = "TEXT";
  Type2["NUMBER"] = "NUMBER";
  Type2["DATE"] = "DATE";
  Type2["URL"] = "URL";
  Type2["SOCIAL"] = "SOCIAL";
  return Type2;
})(Type || {});
var SocialTypeType = /* @__PURE__ */ ((SocialTypeType2) => {
  SocialTypeType2["UNKNOWN"] = "UNKNOWN";
  SocialTypeType2["FACEBOOK"] = "FACEBOOK";
  SocialTypeType2["INSTAGRAM"] = "INSTAGRAM";
  SocialTypeType2["LINKEDIN"] = "LINKEDIN";
  SocialTypeType2["TWITTER"] = "TWITTER";
  SocialTypeType2["YOUTUBE"] = "YOUTUBE";
  SocialTypeType2["PINTEREST"] = "PINTEREST";
  SocialTypeType2["TIKTOK"] = "TIKTOK";
  SocialTypeType2["DEVIANTART"] = "DEVIANTART";
  SocialTypeType2["SOUNDCLOUD"] = "SOUNDCLOUD";
  SocialTypeType2["TUMBLR"] = "TUMBLR";
  SocialTypeType2["VIMEO"] = "VIMEO";
  SocialTypeType2["VKONTAKTE"] = "VKONTAKTE";
  SocialTypeType2["ODNOKLASSNIKI"] = "ODNOKLASSNIKI";
  SocialTypeType2["OTHER"] = "OTHER";
  return SocialTypeType2;
})(SocialTypeType || {});
var Origin = /* @__PURE__ */ ((Origin2) => {
  Origin2["UNKNOWN"] = "UNKNOWN";
  Origin2["CUSTOM"] = "CUSTOM";
  Origin2["CONTACT"] = "CONTACT";
  Origin2["SYSTEM"] = "SYSTEM";
  return Origin2;
})(Origin || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
async function listCustomFieldSuggestions2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = renameKeysFromSDKRequestToRESTRequest({
    paging: options?.paging,
    sorting: options?.sorting
  });
  const reqOpts = listCustomFieldSuggestions(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          paging: "$[0].paging",
          sorting: "$[0].sorting"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryCustomFieldSuggestions2() {
  const { httpClient, sideEffects } = arguments[0];
  return queryBuilder({
    func: async (payload) => {
      const reqOpts = queryCustomFieldSuggestions(
        payload
      );
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return renameKeysFromSDKRequestToRESTRequest({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = renameKeysFromRESTResponseToSDKResponse(data);
      return {
        items: transformedData?.suggestions,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = sdkTransformError(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "OFFSET",
    transformationPaths: {}
  });
}

// src/members-v1-custom-field-suggestion-custom-field-suggestions.public.ts
function listCustomFieldSuggestions3(httpClient) {
  return (options) => listCustomFieldSuggestions2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryCustomFieldSuggestions3(httpClient) {
  return () => queryCustomFieldSuggestions2(
    // @ts-ignore
    { httpClient }
  );
}

// src/members-v1-custom-field-suggestion-custom-field-suggestions.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var listCustomFieldSuggestions4 = /* @__PURE__ */ createRESTModule(listCustomFieldSuggestions3);
var queryCustomFieldSuggestions4 = /* @__PURE__ */ createRESTModule(queryCustomFieldSuggestions3);
export {
  Origin,
  Privacy,
  SocialTypeType,
  SortOrder,
  Type,
  listCustomFieldSuggestions4 as listCustomFieldSuggestions,
  queryCustomFieldSuggestions4 as queryCustomFieldSuggestions
};
