// @wix/auto_sdk_members_custom-field-suggestions/build/meta.mjs
// src/members-v1-custom-field-suggestion-custom-field-suggestions.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members-custom-fields-suggestions-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-field-suggestions";
function listCustomFieldSuggestions(payload) {
  function __listCustomFieldSuggestions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_suggestion",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.suggestions.CustomFieldSuggestions.ListCustomFieldSuggestions",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(
        { protoPath: "/v1/custom-field-suggestions", data: payload, host }
      ),
      params: toURLSearchParams(payload, true)
    };
    return metadata;
  }
  return __listCustomFieldSuggestions;
}
function queryCustomFieldSuggestions(payload) {
  function __queryCustomFieldSuggestions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_suggestion",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.suggestions.CustomFieldSuggestions.QueryCustomFieldSuggestions",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(
        { protoPath: "/v1/custom-field-suggestions/query", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __queryCustomFieldSuggestions;
}

// src/members-v1-custom-field-suggestion-custom-field-suggestions.meta.ts
function listCustomFieldSuggestions2() {
  const payload = {};
  const getRequestOptions = listCustomFieldSuggestions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/custom-field-suggestions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCustomFieldSuggestions2() {
  const payload = {};
  const getRequestOptions = queryCustomFieldSuggestions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-field-suggestions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  listCustomFieldSuggestions2 as listCustomFieldSuggestions,
  queryCustomFieldSuggestions2 as queryCustomFieldSuggestions
};
