// @wix/auto_sdk_bookings_bookings/build/internal/es/src/bookings-v2-booking-bookings.context.js
import { confirmOrDeclineBooking as publicConfirmOrDeclineBooking, bulkConfirmOrDeclineBooking as publicBulkConfirmOrDeclineBooking, createBooking as publicCreateBooking, bulkCreateBooking as publicBulkCreateBooking, rescheduleBooking as publicRescheduleBooking, confirmBooking as publicConfirmBooking, updateExtendedFields as publicUpdateExtendedFields, declineBooking as publicDeclineBooking, cancelBooking as publicCancelBooking, updateNumberOfParticipants as publicUpdateNumberOfParticipants, createMultiServiceBooking as publicCreateMultiServiceBooking, rescheduleMultiServiceBooking as publicRescheduleMultiServiceBooking, getMultiServiceBookingAvailability as publicGetMultiServiceBookingAvailability, cancelMultiServiceBooking as publicCancelMultiServiceBooking, markMultiServiceBookingAsPending as publicMarkMultiServiceBookingAsPending, confirmMultiServiceBooking as publicConfirmMultiServiceBooking, declineMultiServiceBooking as publicDeclineMultiServiceBooking, bulkGetMultiServiceBookingAllowedActions as publicBulkGetMultiServiceBookingAllowedActions, getMultiServiceBooking as publicGetMultiServiceBooking, addBookingsToMultiServiceBooking as publicAddBookingsToMultiServiceBooking, removeBookingsFromMultiServiceBooking as publicRemoveBookingsFromMultiServiceBooking, } from './bookings-v2-booking-bookings.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onBookingCanceled as publicOnBookingCanceled } from './bookings-v2-booking-bookings.public.js';
import { onBookingConfirmed as publicOnBookingConfirmed } from './bookings-v2-booking-bookings.public.js';
import { onBookingCreated as publicOnBookingCreated } from './bookings-v2-booking-bookings.public.js';
import { onBookingDeclined as publicOnBookingDeclined } from './bookings-v2-booking-bookings.public.js';
import { onBookingNumberOfParticipantsUpdated as publicOnBookingNumberOfParticipantsUpdated } from './bookings-v2-booking-bookings.public.js';
import { onBookingRescheduled as publicOnBookingRescheduled } from './bookings-v2-booking-bookings.public.js';
import { onBookingUpdated as publicOnBookingUpdated } from './bookings-v2-booking-bookings.public.js';
export const confirmOrDeclineBooking = /*#__PURE__*/ createRESTModule(publicConfirmOrDeclineBooking);
export const bulkConfirmOrDeclineBooking = /*#__PURE__*/ createRESTModule(publicBulkConfirmOrDeclineBooking);
export const createBooking = /*#__PURE__*/ createRESTModule(publicCreateBooking);
export const bulkCreateBooking = /*#__PURE__*/ createRESTModule(publicBulkCreateBooking);
export const rescheduleBooking = /*#__PURE__*/ createRESTModule(publicRescheduleBooking);
export const confirmBooking = /*#__PURE__*/ createRESTModule(publicConfirmBooking);
export const updateExtendedFields = /*#__PURE__*/ createRESTModule(publicUpdateExtendedFields);
export const declineBooking = /*#__PURE__*/ createRESTModule(publicDeclineBooking);
export const cancelBooking = /*#__PURE__*/ createRESTModule(publicCancelBooking);
export const updateNumberOfParticipants = /*#__PURE__*/ createRESTModule(publicUpdateNumberOfParticipants);
export const createMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicCreateMultiServiceBooking);
export const rescheduleMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicRescheduleMultiServiceBooking);
export const getMultiServiceBookingAvailability = /*#__PURE__*/ createRESTModule(publicGetMultiServiceBookingAvailability);
export const cancelMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicCancelMultiServiceBooking);
export const markMultiServiceBookingAsPending = /*#__PURE__*/ createRESTModule(publicMarkMultiServiceBookingAsPending);
export const confirmMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicConfirmMultiServiceBooking);
export const declineMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicDeclineMultiServiceBooking);
export const bulkGetMultiServiceBookingAllowedActions = /*#__PURE__*/ createRESTModule(publicBulkGetMultiServiceBookingAllowedActions);
export const getMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicGetMultiServiceBooking);
export const addBookingsToMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicAddBookingsToMultiServiceBooking);
export const removeBookingsFromMultiServiceBooking = /*#__PURE__*/ createRESTModule(publicRemoveBookingsFromMultiServiceBooking);
/**
 * Triggered when a booking is canceled.
 */
export const onBookingCanceled = createEventModule(publicOnBookingCanceled);
/**
 * Triggered when a booking is confirmed.
 */
export const onBookingConfirmed = createEventModule(publicOnBookingConfirmed);
/**
 * Triggered when a booking is created.
 */
export const onBookingCreated = createEventModule(publicOnBookingCreated);
/**
 * Triggered when a booking is declined.
 */
export const onBookingDeclined = createEventModule(publicOnBookingDeclined);
/**
 * Triggered when the number of participants is updated.
 */
export const onBookingNumberOfParticipantsUpdated = createEventModule(publicOnBookingNumberOfParticipantsUpdated);
/**
 * Triggered when a booking is rescheduled.
 */
export const onBookingRescheduled = createEventModule(publicOnBookingRescheduled);
/**
 * Triggered when a booked schedule is updated.
 */
export const onBookingUpdated = createEventModule(publicOnBookingUpdated);
export { MultiServiceBookingType, LocationType, ValueType, BookingStatus, PaymentStatus, SelectedPaymentOption, Platform, Actor, IdentificationDataIdentityType, WebhookIdentityType, LocationLocationType, IdentityType, SortOrder, } from './bookings-v2-booking-bookings.universal.js';
