// @wix/auto_sdk_ecom_tax-regions/build/internal/cjs/src/billing-v1-tax-region-tax-regions.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryTaxRegions = exports.deleteTaxRegion = exports.updateTaxRegion = exports.getTaxRegion = exports.createTaxRegion = void 0;
const ambassadorWixBillingV1TaxRegion = __importStar(require("./billing-v1-tax-region-tax-regions.http.js"));
function createTaxRegion() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.createTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-regions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createTaxRegion = createTaxRegion;
function getTaxRegion() {
    const payload = { taxRegionId: ':taxRegionId' };
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.getTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tax-regions/{taxRegionId}',
        pathParams: { taxRegionId: 'taxRegionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTaxRegion = getTaxRegion;
function updateTaxRegion() {
    const payload = { taxRegion: { id: ':taxRegionId' } };
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.updateTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tax-regions/{taxRegion.id}',
        pathParams: { taxRegionId: 'taxRegionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateTaxRegion = updateTaxRegion;
function deleteTaxRegion() {
    const payload = { taxRegionId: ':taxRegionId' };
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.deleteTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tax-regions/{taxRegionId}',
        pathParams: { taxRegionId: 'taxRegionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteTaxRegion = deleteTaxRegion;
function queryTaxRegions() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.queryTaxRegions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-regions/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryTaxRegions = queryTaxRegions;
