// @wix/auto_sdk_ecom_tax-regions/build/internal/cjs/src/billing-v1-tax-region-tax-regions.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.onTaxRegionUpdated = exports.onTaxRegionDeleted = exports.onTaxRegionCreated = exports.queryTaxRegions = exports.deleteTaxRegion = exports.updateTaxRegion = exports.getTaxRegion = exports.createTaxRegion = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const billing_v1_tax_region_tax_regions_universal_js_1 = require("./billing-v1-tax-region-tax-regions.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createTaxRegion(httpClient) {
    return (taxRegion) => (0, billing_v1_tax_region_tax_regions_universal_js_1.createTaxRegion)(taxRegion, 
    // @ts-ignore
    { httpClient });
}
exports.createTaxRegion = createTaxRegion;
function getTaxRegion(httpClient) {
    return (taxRegionId) => (0, billing_v1_tax_region_tax_regions_universal_js_1.getTaxRegion)(taxRegionId, 
    // @ts-ignore
    { httpClient });
}
exports.getTaxRegion = getTaxRegion;
function updateTaxRegion(httpClient) {
    return (_id, taxRegion) => (0, billing_v1_tax_region_tax_regions_universal_js_1.updateTaxRegion)(_id, taxRegion, 
    // @ts-ignore
    { httpClient });
}
exports.updateTaxRegion = updateTaxRegion;
function deleteTaxRegion(httpClient) {
    return (taxRegionId) => (0, billing_v1_tax_region_tax_regions_universal_js_1.deleteTaxRegion)(taxRegionId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteTaxRegion = deleteTaxRegion;
function queryTaxRegions(httpClient) {
    return () => (0, billing_v1_tax_region_tax_regions_universal_js_1.queryTaxRegions)(
    // @ts-ignore
    { httpClient });
}
exports.queryTaxRegions = queryTaxRegions;
exports.onTaxRegionCreated = (0, sdk_types_1.EventDefinition)('wix.billing.v1.tax_region_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onTaxRegionDeleted = (0, sdk_types_1.EventDefinition)('wix.billing.v1.tax_region_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onTaxRegionUpdated = (0, sdk_types_1.EventDefinition)('wix.billing.v1.tax_region_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var billing_v1_tax_region_tax_regions_universal_js_2 = require("./billing-v1-tax-region-tax-regions.universal.js");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return billing_v1_tax_region_tax_regions_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return billing_v1_tax_region_tax_regions_universal_js_2.WebhookIdentityType; } });
