// @wix/auto_sdk_ecom_tax-calculation-provider/build/internal/cjs/src/interfaces-billing-v1-tax-calculation-provider.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('TAX_CALCULATION_PROVIDER', [
    {
        name: 'calculateTax',
        primaryHttpMappingPath: '/v1/calculate-tax',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = (0, transform_paths_1.transformPaths)(payload, [
                    {
                        transformFn: address_1.transformRESTAddressToSDKAddress,
                        paths: [{ path: 'request.addresses', isRepeated: true }],
                    },
                ]);
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
