// @wix/auto_sdk_bookings_app-settings/build/cjs/src/bookings-v2-bookings-app-settings-app-settings.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateBookingsAppSettings = exports.getBookingsAppSettings = exports.createBookingsAppSettings = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBookingsV2BookingsAppSettingsServiceUrl(opts) {
    const domainToMappings = {
        'dev._base_domain_': [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/_api/bookings/v2/app-settings',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_bookings_app-settings';
/** Creates a BookingsAppSettings. */
function createBookingsAppSettings(payload) {
    function __createBookingsAppSettings({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'bookingsAppSettings.createdDate' },
                    { path: 'bookingsAppSettings.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.bookings.v2.bookings_app_settings',
            method: 'POST',
            methodFqn: 'wix.bookings.v2.BookingsAppSettingsService.CreateBookingsAppSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                protoPath: '/v2/bookings-app-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'bookingsAppSettings.createdDate' },
                        { path: 'bookingsAppSettings.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createBookingsAppSettings;
}
exports.createBookingsAppSettings = createBookingsAppSettings;
/** Retrieves a BookingsAppSettings. */
function getBookingsAppSettings(payload) {
    function __getBookingsAppSettings({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.v2.bookings_app_settings',
            method: 'GET',
            methodFqn: 'wix.bookings.v2.BookingsAppSettingsService.GetBookingsAppSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                protoPath: '/v2/settings',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'bookingsAppSettings.createdDate' },
                        { path: 'bookingsAppSettings.updatedDate' },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'GET',
                    url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                        protoPath: '/v2/settings',
                        data: payload,
                        host,
                    }),
                    params: (0, rest_modules_1.toURLSearchParams)(payload),
                },
            ],
        };
        return metadata;
    }
    return __getBookingsAppSettings;
}
exports.getBookingsAppSettings = getBookingsAppSettings;
/**
 * Updates a BookingsAppSettings.
 *
 * Each time the BookingsAppSettings is updated,  `revision` increments by 1.
 * The current `revision` must be passed when updating the BookingsAppSettings.
 * This ensures you're working with the latest BookingsAppSettings  and prevents unintended overwrites.
 */
function updateBookingsAppSettings(payload) {
    function __updateBookingsAppSettings({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'bookingsAppSettings.createdDate' },
                    { path: 'bookingsAppSettings.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.bookings.v2.bookings_app_settings',
            method: 'PATCH',
            methodFqn: 'wix.bookings.v2.BookingsAppSettingsService.UpdateBookingsAppSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                protoPath: '/v2/bookings-app-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'bookingsAppSettings.createdDate' },
                        { path: 'bookingsAppSettings.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateBookingsAppSettings;
}
exports.updateBookingsAppSettings = updateBookingsAppSettings;
