// @wix/auto_sdk_members_member-report/build/internal/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Category: () => Category,
  SortOrder: () => SortOrder,
  WebhookIdentityType: () => WebhookIdentityType,
  deleteMemberReports: () => deleteMemberReports4,
  listReportedMembers: () => listReportedMembers4,
  onMemberReportCreated: () => onMemberReportCreated2,
  onMemberReportDeleted: () => onMemberReportDeleted2,
  onMemberReportReportedMemberCreated: () => onMemberReportReportedMemberCreated2,
  onMemberReportReportedMemberDeleted: () => onMemberReportReportedMemberDeleted2,
  queryMemberReports: () => queryMemberReports4,
  queryReportedMembers: () => queryReportedMembers4,
  reportMember: () => reportMember4
});
module.exports = __toCommonJS(index_exports);

// src/members-v1-member-report-member-report.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/members-v1-member-report-member-report.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/members-v1-member-report-member-report.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersMemberreportsMemberReportsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members-reports-api",
        destPath: ""
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    _: [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-report";
function reportMember(payload) {
  function __reportMember({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "memberReport.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.ReportMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberReport.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __reportMember;
}
function queryMemberReports(payload) {
  function __queryMemberReports({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.QueryMemberReports",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberReports.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryMemberReports;
}
function listReportedMembers(payload) {
  function __listReportedMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "GET",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.ListReportedMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "reportedMembers.lastReportDate" }]
        }
      ])
    };
    return metadata;
  }
  return __listReportedMembers;
}
function queryReportedMembers(payload) {
  function __queryReportedMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.QueryReportedMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "reportedMembers.lastReportDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryReportedMembers;
}
function deleteMemberReports(payload) {
  function __deleteMemberReports({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "DELETE",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.DeleteMemberReports",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMemberReports;
}

// src/members-v1-member-report-member-report.universal.ts
var Category = /* @__PURE__ */ ((Category2) => {
  Category2["UNKNOWN"] = "UNKNOWN";
  Category2["SPAM"] = "SPAM";
  Category2["IMPERSONATION"] = "IMPERSONATION";
  Category2["HARASSMENT"] = "HARASSMENT";
  Category2["OTHER"] = "OTHER";
  return Category2;
})(Category || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function reportMember2(memberReport) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberReport
  });
  const reqOpts = reportMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberReport: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberReport"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryMemberReports2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryMemberReports(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(data);
      return {
        items: transformedData?.memberReports,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function listReportedMembers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listReportedMembers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { cursorPaging: "$[0].cursorPaging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function queryReportedMembers2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    query: options?.query
  });
  const reqOpts = queryReportedMembers(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0].query" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteMemberReports2(memberId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ memberId });
  const reqOpts = deleteMemberReports(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/members-v1-member-report-member-report.public.ts
function reportMember3(httpClient) {
  return (memberReport) => reportMember2(
    memberReport,
    // @ts-ignore
    { httpClient }
  );
}
function queryMemberReports3(httpClient) {
  return () => queryMemberReports2(
    // @ts-ignore
    { httpClient }
  );
}
function listReportedMembers3(httpClient) {
  return (options) => listReportedMembers2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function queryReportedMembers3(httpClient) {
  return (options) => queryReportedMembers2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function deleteMemberReports3(httpClient) {
  return (memberId) => deleteMemberReports2(
    memberId,
    // @ts-ignore
    { httpClient }
  );
}
var onMemberReportCreated = (0, import_sdk_types.EventDefinition)(
  "wix.members.v1.member_report_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.createdDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMemberReportDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.members.v1.member_report_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.createdDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMemberReportReportedMemberCreated = (0, import_sdk_types.EventDefinition)(
  "wix.members.v1.member_report_reported_member_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "data.reportedMember.lastReportDate" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMemberReportReportedMemberDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.members.v1.member_report_reported_member_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: "metadata.eventTime" }]
      }
    ])
  )
)();

// src/members-v1-member-report-member-report.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var reportMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(reportMember3);
var queryMemberReports4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryMemberReports3);
var listReportedMembers4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listReportedMembers3);
var queryReportedMembers4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryReportedMembers3);
var deleteMemberReports4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteMemberReports3);
var onMemberReportCreated2 = (0, import_event_definition_modules.createEventModule)(onMemberReportCreated);
var onMemberReportDeleted2 = (0, import_event_definition_modules.createEventModule)(onMemberReportDeleted);
var onMemberReportReportedMemberCreated2 = (0, import_event_definition_modules.createEventModule)(onMemberReportReportedMemberCreated);
var onMemberReportReportedMemberDeleted2 = (0, import_event_definition_modules.createEventModule)(onMemberReportReportedMemberDeleted);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Category,
  SortOrder,
  WebhookIdentityType,
  deleteMemberReports,
  listReportedMembers,
  onMemberReportCreated,
  onMemberReportDeleted,
  onMemberReportReportedMemberCreated,
  onMemberReportReportedMemberDeleted,
  queryMemberReports,
  queryReportedMembers,
  reportMember
});
