// @wix/auto_sdk_members_member-report/build/internal/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  deleteMemberReports: () => deleteMemberReports2,
  listReportedMembers: () => listReportedMembers2,
  queryMemberReports: () => queryMemberReports2,
  queryReportedMembers: () => queryReportedMembers2,
  reportMember: () => reportMember2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-member-report-member-report.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersMemberreportsMemberReportsUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members-reports-api",
        destPath: ""
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    _: [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/members/v1/member-reports",
        destPath: "/v1/member-reports"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-report";
function reportMember(payload) {
  function __reportMember({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "memberReport.createdDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.ReportMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberReport.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __reportMember;
}
function queryMemberReports(payload) {
  function __queryMemberReports({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.QueryMemberReports",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberReports.createdDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryMemberReports;
}
function listReportedMembers(payload) {
  function __listReportedMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "GET",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.ListReportedMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "reportedMembers.lastReportDate" }]
        }
      ])
    };
    return metadata;
  }
  return __listReportedMembers;
}
function queryReportedMembers(payload) {
  function __queryReportedMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "POST",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.QueryReportedMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "reportedMembers.lastReportDate" }]
        }
      ])
    };
    return metadata;
  }
  return __queryReportedMembers;
}
function deleteMemberReports(payload) {
  function __deleteMemberReports({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_report",
      method: "DELETE",
      methodFqn: "com.wixpress.members.memberreports.MemberReports.DeleteMemberReports",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberreportsMemberReportsUrl({
        protoPath: "/v1/member-reports/members/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMemberReports;
}

// src/members-v1-member-report-member-report.meta.ts
function reportMember2() {
  const payload = {};
  const getRequestOptions = reportMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/member-reports",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMemberReports2() {
  const payload = {};
  const getRequestOptions = queryMemberReports(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/member-reports/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listReportedMembers2() {
  const payload = {};
  const getRequestOptions = listReportedMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/member-reports/members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryReportedMembers2() {
  const payload = {};
  const getRequestOptions = queryReportedMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/member-reports/members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberReports2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = deleteMemberReports(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/member-reports/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  deleteMemberReports,
  listReportedMembers,
  queryMemberReports,
  queryReportedMembers,
  reportMember
});
