// @wix/auto_sdk_bookings_app-settings/build/internal/es/src/bookings-v2-bookings-app-settings-app-settings.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformSDKTimestampToRESTTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformSDKFieldMaskToRESTFieldMask } from '@wix/sdk-runtime/transformations/field-mask';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixBookingsV2BookingsAppSettingsServiceUrl(opts) {
    const domainToMappings = {
        'dev._base_domain_': [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/_api/bookings/v2/app-settings',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/bookings-app-settings',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_bookings_app-settings';
/** Creates a BookingsAppSettings. */
export function createBookingsAppSettings(payload) {
    function __createBookingsAppSettings({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'bookingsAppSettings.createdDate' },
                    { path: 'bookingsAppSettings.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.bookings.v2.bookings_app_settings',
            method: 'POST',
            methodFqn: 'wix.bookings.v2.BookingsAppSettingsService.CreateBookingsAppSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                protoPath: '/v2/bookings-app-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'bookingsAppSettings.createdDate' },
                        { path: 'bookingsAppSettings.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createBookingsAppSettings;
}
/** Retrieves a BookingsAppSettings. */
export function getBookingsAppSettings(payload) {
    function __getBookingsAppSettings({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.v2.bookings_app_settings',
            method: 'GET',
            methodFqn: 'wix.bookings.v2.BookingsAppSettingsService.GetBookingsAppSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                protoPath: '/v2/settings',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'bookingsAppSettings.createdDate' },
                        { path: 'bookingsAppSettings.updatedDate' },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'GET',
                    url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                        protoPath: '/v2/settings',
                        data: payload,
                        host,
                    }),
                    params: toURLSearchParams(payload),
                },
            ],
        };
        return metadata;
    }
    return __getBookingsAppSettings;
}
/**
 * Updates a BookingsAppSettings.
 *
 * Each time the BookingsAppSettings is updated,  `revision` increments by 1.
 * The current `revision` must be passed when updating the BookingsAppSettings.
 * This ensures you're working with the latest BookingsAppSettings  and prevents unintended overwrites.
 */
export function updateBookingsAppSettings(payload) {
    function __updateBookingsAppSettings({ host }) {
        const serializedData = transformPaths(payload, [
            {
                transformFn: transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'bookingsAppSettings.createdDate' },
                    { path: 'bookingsAppSettings.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.bookings.v2.bookings_app_settings',
            method: 'PATCH',
            methodFqn: 'wix.bookings.v2.BookingsAppSettingsService.UpdateBookingsAppSettings',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsV2BookingsAppSettingsServiceUrl({
                protoPath: '/v2/bookings-app-settings',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'bookingsAppSettings.createdDate' },
                        { path: 'bookingsAppSettings.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateBookingsAppSettings;
}
