// @wix/auto_sdk_bookings_booking-fees/build/cjs/src/bookings-fees-v1-booking-fee-booking-fees.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.collectAppliedBookingFees = exports.applyBookingFeesToOrder = exports.listBookingFeesByBookingIds = void 0;
const ambassadorWixBookingsFeesV1BookingFee = __importStar(require("./bookings-fees-v1-booking-fee-booking-fees.http.js"));
function listBookingFeesByBookingIds() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsFeesV1BookingFee.listBookingFeesByBookingIds(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/booking-fees',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listBookingFeesByBookingIds = listBookingFeesByBookingIds;
function applyBookingFeesToOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsFeesV1BookingFee.applyBookingFeesToOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-fees/apply',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.applyBookingFeesToOrder = applyBookingFeesToOrder;
function collectAppliedBookingFees() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsFeesV1BookingFee.collectAppliedBookingFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-fees/collect',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.collectAppliedBookingFees = collectAppliedBookingFees;
