// @wix/auto_sdk_bookings_staff-members/build/internal/cjs/src/bookings-staff-v1-staff-member-staff-members.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bulkUpdateStaffMemberTagsByFilter = exports.bulkUpdateStaffMemberTags = exports.assignCustomSchedule = exports.assignWorkingHoursSchedule = exports.disconnectStaffMemberFromUser = exports.searchStaffMembers = exports.connectStaffMemberToUser = exports.countStaffMembers = exports.queryStaffMembers = exports.deleteStaffMember = exports.updateStaffMember = exports.getStaffMember = exports.createStaffMember = void 0;
const ambassadorWixBookingsStaffV1StaffMember = __importStar(require("./bookings-staff-v1-staff-member-staff-members.http.js"));
function createStaffMember() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.createStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createStaffMember = createStaffMember;
function getStaffMember() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.getStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/staff-members/{staffMemberId}',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getStaffMember = getStaffMember;
function updateStaffMember() {
    const payload = { staffMember: { id: ':staffMemberId' } };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.updateStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/staff-members/{staffMember.id}',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateStaffMember = updateStaffMember;
function deleteStaffMember() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.deleteStaffMember(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/staff-members/{staffMemberId}',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteStaffMember = deleteStaffMember;
function queryStaffMembers() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.queryStaffMembers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryStaffMembers = queryStaffMembers;
function countStaffMembers() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.countStaffMembers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.countStaffMembers = countStaffMembers;
function connectStaffMemberToUser() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.connectStaffMemberToUser(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/connect-staff-member-to-user',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.connectStaffMemberToUser = connectStaffMemberToUser;
function searchStaffMembers() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.searchStaffMembers(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchStaffMembers = searchStaffMembers;
function disconnectStaffMemberFromUser() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.disconnectStaffMemberFromUser(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/disconnect-staff-member-from-user',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.disconnectStaffMemberFromUser = disconnectStaffMemberFromUser;
function assignWorkingHoursSchedule() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.assignWorkingHoursSchedule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/assign-working-hours-schedule',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.assignWorkingHoursSchedule = assignWorkingHoursSchedule;
function assignCustomSchedule() {
    const payload = { staffMemberId: ':staffMemberId' };
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.assignCustomSchedule(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/staff-members/{staffMemberId}/assign-custom-schedule',
        pathParams: { staffMemberId: 'staffMemberId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.assignCustomSchedule = assignCustomSchedule;
function bulkUpdateStaffMemberTags() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.bulkUpdateStaffMemberTags(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/staff-members/update-tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateStaffMemberTags = bulkUpdateStaffMemberTags;
function bulkUpdateStaffMemberTagsByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsStaffV1StaffMember.bulkUpdateStaffMemberTagsByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/bulk/staff-members/update-tags-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateStaffMemberTagsByFilter = bulkUpdateStaffMemberTagsByFilter;
