// @wix/auto_sdk_bookings_staff-members/build/internal/cjs/src/bookings-staff-v1-staff-member-staff-members.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Transparency = exports.Status = exports.SortOrder = exports.SessionType = exports.ScheduleStatus = exports.ScalarType = exports.RequestedFields = exports.RecurringIntervalType = exports.Mode = exports.LocationsLocationType = exports.LocationType = exports.LocationStatus = exports.IdentityType = exports.DayOfWeek = exports.Day = exports.ConnectionStatus = exports.ConferenceType = exports.CalendarType = exports.AssociatedWixIdentityConnectionStatusEnumConnectionStatus = exports.ApprovalStatus = exports.AggregationType = exports.onStaffMemberUpdated = exports.onStaffMemberFullyCreated = exports.onStaffMemberDisconnectedFromUser = exports.onStaffMemberDeleted = exports.onStaffMemberCreated = exports.onStaffMemberConnectedToUser = exports.bulkUpdateStaffMemberTagsByFilter = exports.bulkUpdateStaffMemberTags = exports.assignCustomSchedule = exports.assignWorkingHoursSchedule = exports.disconnectStaffMemberFromUser = exports.searchStaffMembers = exports.connectStaffMemberToUser = exports.countStaffMembers = exports.queryStaffMembers = exports.deleteStaffMember = exports.updateStaffMember = exports.getStaffMember = exports.createStaffMember = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_staff_v1_staff_member_staff_members_universal_js_1 = require("./bookings-staff-v1-staff-member-staff-members.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function createStaffMember(httpClient) {
    return (staffMember, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.createStaffMember)(staffMember, options, 
    // @ts-ignore
    { httpClient });
}
exports.createStaffMember = createStaffMember;
function getStaffMember(httpClient) {
    return (staffMemberId, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.getStaffMember)(staffMemberId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getStaffMember = getStaffMember;
function updateStaffMember(httpClient) {
    return (_id, staffMember, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.updateStaffMember)(_id, staffMember, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateStaffMember = updateStaffMember;
function deleteStaffMember(httpClient) {
    return (staffMemberId) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.deleteStaffMember)(staffMemberId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteStaffMember = deleteStaffMember;
function queryStaffMembers(httpClient) {
    return (options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.queryStaffMembers)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryStaffMembers = queryStaffMembers;
function countStaffMembers(httpClient) {
    return (options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.countStaffMembers)(options, 
    // @ts-ignore
    { httpClient });
}
exports.countStaffMembers = countStaffMembers;
function connectStaffMemberToUser(httpClient) {
    return (staffMemberId, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.connectStaffMemberToUser)(staffMemberId, options, 
    // @ts-ignore
    { httpClient });
}
exports.connectStaffMemberToUser = connectStaffMemberToUser;
function searchStaffMembers(httpClient) {
    return (search, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.searchStaffMembers)(search, options, 
    // @ts-ignore
    { httpClient });
}
exports.searchStaffMembers = searchStaffMembers;
function disconnectStaffMemberFromUser(httpClient) {
    return (staffMemberId, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.disconnectStaffMemberFromUser)(staffMemberId, options, 
    // @ts-ignore
    { httpClient });
}
exports.disconnectStaffMemberFromUser = disconnectStaffMemberFromUser;
function assignWorkingHoursSchedule(httpClient) {
    return (staffMemberId, scheduleId, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.assignWorkingHoursSchedule)(staffMemberId, scheduleId, options, 
    // @ts-ignore
    { httpClient });
}
exports.assignWorkingHoursSchedule = assignWorkingHoursSchedule;
function assignCustomSchedule(httpClient) {
    return (staffMemberId, scheduleId, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.assignCustomSchedule)(staffMemberId, scheduleId, options, 
    // @ts-ignore
    { httpClient });
}
exports.assignCustomSchedule = assignCustomSchedule;
function bulkUpdateStaffMemberTags(httpClient) {
    return (ids, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.bulkUpdateStaffMemberTags)(ids, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateStaffMemberTags = bulkUpdateStaffMemberTags;
function bulkUpdateStaffMemberTagsByFilter(httpClient) {
    return (filter, options) => (0, bookings_staff_v1_staff_member_staff_members_universal_js_1.bulkUpdateStaffMemberTagsByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateStaffMemberTagsByFilter = bulkUpdateStaffMemberTagsByFilter;
exports.onStaffMemberConnectedToUser = (0, sdk_types_1.EventDefinition)('wix.bookings.staff.v1.staff_member_connected_to_user', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.staffMember.createdDate' },
            { path: 'data.staffMember.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'data.staffMember.mainMedia.image' }],
    },
])))();
exports.onStaffMemberCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.staff.v1.staff_member_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.mainMedia.image' }],
    },
])))();
exports.onStaffMemberDeleted = (0, sdk_types_1.EventDefinition)('wix.bookings.staff.v1.staff_member_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.mainMedia.image' }],
    },
])))();
exports.onStaffMemberDisconnectedFromUser = (0, sdk_types_1.EventDefinition)('wix.bookings.staff.v1.staff_member_disconnected_from_user', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.staffMember.createdDate' },
            { path: 'data.staffMember.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'data.staffMember.mainMedia.image' }],
    },
])))();
exports.onStaffMemberFullyCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.staff.v1.staff_member_fully_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.staffMember.createdDate' },
            { path: 'data.staffMember.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'data.staffMember.mainMedia.image' }],
    },
])))();
exports.onStaffMemberUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.staff.v1.staff_member_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.mainMedia.image' }],
    },
])))();
var bookings_staff_v1_staff_member_staff_members_universal_js_2 = require("./bookings-staff-v1-staff-member-staff-members.universal.js");
Object.defineProperty(exports, "AggregationType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.AggregationType; } });
Object.defineProperty(exports, "ApprovalStatus", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.ApprovalStatus; } });
Object.defineProperty(exports, "AssociatedWixIdentityConnectionStatusEnumConnectionStatus", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.AssociatedWixIdentityConnectionStatusEnumConnectionStatus; } });
Object.defineProperty(exports, "CalendarType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.CalendarType; } });
Object.defineProperty(exports, "ConferenceType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.ConferenceType; } });
Object.defineProperty(exports, "ConnectionStatus", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.ConnectionStatus; } });
Object.defineProperty(exports, "Day", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.Day; } });
Object.defineProperty(exports, "DayOfWeek", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.DayOfWeek; } });
Object.defineProperty(exports, "IdentityType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.IdentityType; } });
Object.defineProperty(exports, "LocationStatus", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.LocationStatus; } });
Object.defineProperty(exports, "LocationType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.LocationType; } });
Object.defineProperty(exports, "LocationsLocationType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.LocationsLocationType; } });
Object.defineProperty(exports, "Mode", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.Mode; } });
Object.defineProperty(exports, "RecurringIntervalType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.RecurringIntervalType; } });
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "ScalarType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.ScalarType; } });
Object.defineProperty(exports, "ScheduleStatus", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.ScheduleStatus; } });
Object.defineProperty(exports, "SessionType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.SessionType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.Status; } });
Object.defineProperty(exports, "Transparency", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.Transparency; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_staff_v1_staff_member_staff_members_universal_js_2.WebhookIdentityType; } });
