// @wix/auto_sdk_members_custom-field-applications/build/internal/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  AppliesToAppliesTo: () => AppliesToAppliesTo,
  ExclusionType: () => ExclusionType,
  FieldTypeType: () => FieldTypeType,
  Origin: () => Origin,
  Privacy: () => Privacy,
  Section: () => Section,
  SocialTypeType: () => SocialTypeType,
  Type: () => Type,
  WebhookIdentityType: () => WebhookIdentityType,
  createCustomFieldApplication: () => createCustomFieldApplication4,
  deleteCustomFieldApplication: () => deleteCustomFieldApplication4,
  getCustomFieldApplication: () => getCustomFieldApplication4,
  getCustomFieldApplications: () => getCustomFieldApplications4,
  getMembersCustomFieldApplications: () => getMembersCustomFieldApplications4,
  getRolesCustomFieldApplications: () => getRolesCustomFieldApplications4,
  updateCustomFieldApplication: () => updateCustomFieldApplication4
});
module.exports = __toCommonJS(index_exports);

// src/members-v1-custom-field-application-custom-field-applications.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/members-v1-custom-field-application-custom-field-applications.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-field-applications";
function createCustomFieldApplication(payload) {
  function __createCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.CreateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        { protoPath: "/v1/custom-fields-applications", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __createCustomFieldApplication;
}
function deleteCustomFieldApplication(payload) {
  function __deleteCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "DELETE",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.DeleteCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCustomFieldApplication;
}
function updateCustomFieldApplication(payload) {
  function __updateCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "PATCH",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.UpdateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{application.customFieldId}",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __updateCustomFieldApplication;
}
function getCustomFieldApplication(payload) {
  function __getCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCustomFieldApplication;
}
function getCustomFieldApplications(payload) {
  function __getCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/applications",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __getCustomFieldApplications;
}
function getMembersCustomFieldApplications(payload) {
  function __getMembersCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetMembersCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/members",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMembersCustomFieldApplications;
}
function getRolesCustomFieldApplications(payload) {
  function __getRolesCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetRolesCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/roles",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRolesCustomFieldApplications;
}

// src/members-v1-custom-field-application-custom-field-applications.universal.ts
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["ROLE"] = "ROLE";
  Type2["BADGE"] = "BADGE";
  Type2["PRICING_PLAN"] = "PRICING_PLAN";
  Type2["MEMBER"] = "MEMBER";
  return Type2;
})(Type || {});
var ExclusionType = /* @__PURE__ */ ((ExclusionType2) => {
  ExclusionType2["UNKNOWN"] = "UNKNOWN";
  ExclusionType2["ROLE"] = "ROLE";
  ExclusionType2["BADGE"] = "BADGE";
  ExclusionType2["PRICING_PLAN"] = "PRICING_PLAN";
  ExclusionType2["MEMBER"] = "MEMBER";
  return ExclusionType2;
})(ExclusionType || {});
var Privacy = /* @__PURE__ */ ((Privacy2) => {
  Privacy2["UNKNOWN"] = "UNKNOWN";
  Privacy2["PUBLIC"] = "PUBLIC";
  Privacy2["PRIVATE"] = "PRIVATE";
  return Privacy2;
})(Privacy || {});
var FieldTypeType = /* @__PURE__ */ ((FieldTypeType2) => {
  FieldTypeType2["UNKNOWN"] = "UNKNOWN";
  FieldTypeType2["TEXT"] = "TEXT";
  FieldTypeType2["NUMBER"] = "NUMBER";
  FieldTypeType2["DATE"] = "DATE";
  FieldTypeType2["URL"] = "URL";
  FieldTypeType2["SOCIAL"] = "SOCIAL";
  return FieldTypeType2;
})(FieldTypeType || {});
var SocialTypeType = /* @__PURE__ */ ((SocialTypeType2) => {
  SocialTypeType2["UNKNOWN"] = "UNKNOWN";
  SocialTypeType2["FACEBOOK"] = "FACEBOOK";
  SocialTypeType2["INSTAGRAM"] = "INSTAGRAM";
  SocialTypeType2["LINKEDIN"] = "LINKEDIN";
  SocialTypeType2["TWITTER"] = "TWITTER";
  SocialTypeType2["YOUTUBE"] = "YOUTUBE";
  SocialTypeType2["PINTEREST"] = "PINTEREST";
  SocialTypeType2["TIKTOK"] = "TIKTOK";
  SocialTypeType2["DEVIANTART"] = "DEVIANTART";
  SocialTypeType2["SOUNDCLOUD"] = "SOUNDCLOUD";
  SocialTypeType2["TUMBLR"] = "TUMBLR";
  SocialTypeType2["VIMEO"] = "VIMEO";
  SocialTypeType2["VKONTAKTE"] = "VKONTAKTE";
  SocialTypeType2["ODNOKLASSNIKI"] = "ODNOKLASSNIKI";
  SocialTypeType2["OTHER"] = "OTHER";
  return SocialTypeType2;
})(SocialTypeType || {});
var Origin = /* @__PURE__ */ ((Origin2) => {
  Origin2["UNKNOWN"] = "UNKNOWN";
  Origin2["CUSTOM"] = "CUSTOM";
  Origin2["CONTACT"] = "CONTACT";
  Origin2["SYSTEM"] = "SYSTEM";
  return Origin2;
})(Origin || {});
var AppliesToAppliesTo = /* @__PURE__ */ ((AppliesToAppliesTo2) => {
  AppliesToAppliesTo2["ALL_MEMBERS"] = "ALL_MEMBERS";
  AppliesToAppliesTo2["SELECTED_MEMBERS"] = "SELECTED_MEMBERS";
  return AppliesToAppliesTo2;
})(AppliesToAppliesTo || {});
var Section = /* @__PURE__ */ ((Section2) => {
  Section2["GENERAL"] = "GENERAL";
  Section2["SOCIAL"] = "SOCIAL";
  Section2["DISPLAY_INFO"] = "DISPLAY_INFO";
  Section2["ADDRESS"] = "ADDRESS";
  return Section2;
})(Section || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createCustomFieldApplication2(application) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    application
  });
  const reqOpts = createCustomFieldApplication(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.application;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { application: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["application"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteCustomFieldApplication2(customFieldId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    customFieldId
  });
  const reqOpts = deleteCustomFieldApplication(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { customFieldId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["customFieldId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateCustomFieldApplication2(customFieldId, application) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    application: { ...application, customFieldId }
  });
  const reqOpts = updateCustomFieldApplication(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.application;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { application: "$[1]" },
        explicitPathsToArguments: { "application.customFieldId": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["customFieldId", "application"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCustomFieldApplication2(customFieldId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    customFieldId
  });
  const reqOpts = getCustomFieldApplication(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { customFieldId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["customFieldId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getCustomFieldApplications2(customFieldIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    customFieldIds
  });
  const reqOpts = getCustomFieldApplications(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { customFieldIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["customFieldIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMembersCustomFieldApplications2(memberIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberIds
  });
  const reqOpts = getMembersCustomFieldApplications(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getRolesCustomFieldApplications2(roleIds) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ roleIds });
  const reqOpts = getRolesCustomFieldApplications(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { roleIds: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["roleIds"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/members-v1-custom-field-application-custom-field-applications.public.ts
function createCustomFieldApplication3(httpClient) {
  return (application) => createCustomFieldApplication2(
    application,
    // @ts-ignore
    { httpClient }
  );
}
function deleteCustomFieldApplication3(httpClient) {
  return (customFieldId) => deleteCustomFieldApplication2(
    customFieldId,
    // @ts-ignore
    { httpClient }
  );
}
function updateCustomFieldApplication3(httpClient) {
  return (customFieldId, application) => updateCustomFieldApplication2(
    customFieldId,
    application,
    // @ts-ignore
    { httpClient }
  );
}
function getCustomFieldApplication3(httpClient) {
  return (customFieldId) => getCustomFieldApplication2(
    customFieldId,
    // @ts-ignore
    { httpClient }
  );
}
function getCustomFieldApplications3(httpClient) {
  return (customFieldIds) => getCustomFieldApplications2(
    customFieldIds,
    // @ts-ignore
    { httpClient }
  );
}
function getMembersCustomFieldApplications3(httpClient) {
  return (memberIds) => getMembersCustomFieldApplications2(
    memberIds,
    // @ts-ignore
    { httpClient }
  );
}
function getRolesCustomFieldApplications3(httpClient) {
  return (roleIds) => getRolesCustomFieldApplications2(
    roleIds,
    // @ts-ignore
    { httpClient }
  );
}

// src/members-v1-custom-field-application-custom-field-applications.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var createCustomFieldApplication4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createCustomFieldApplication3);
var deleteCustomFieldApplication4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteCustomFieldApplication3);
var updateCustomFieldApplication4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateCustomFieldApplication3);
var getCustomFieldApplication4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getCustomFieldApplication3);
var getCustomFieldApplications4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getCustomFieldApplications3);
var getMembersCustomFieldApplications4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getMembersCustomFieldApplications3);
var getRolesCustomFieldApplications4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getRolesCustomFieldApplications3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AppliesToAppliesTo,
  ExclusionType,
  FieldTypeType,
  Origin,
  Privacy,
  Section,
  SocialTypeType,
  Type,
  WebhookIdentityType,
  createCustomFieldApplication,
  deleteCustomFieldApplication,
  getCustomFieldApplication,
  getCustomFieldApplications,
  getMembersCustomFieldApplications,
  getRolesCustomFieldApplications,
  updateCustomFieldApplication
});
