// @wix/bookings_app-extensions/build/cjs/src/extensions/dashboard-page-builders.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.staffList = exports.staffEdit = exports.servicesList = exports.syncPersonalCalendar = exports.exportBookingData = exports.serviceForm = void 0;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Service Form page in the dashboard.
 * @dashobardPageBuilder
 */
function serviceForm(options) {
    const { serviceId } = options;
    return {
        pageId: 'db6319e8-f8d6-4de3-830e-917d4e5c6d47',
        relativeUrl: `${serviceId}`,
    };
}
exports.serviceForm = serviceForm;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Export Booking Data page in the dashboard.
 * @dashobardPageBuilder
 */
function exportBookingData() {
    return {
        pageId: 'bbbc8c7d-19b9-4a86-bfc5-1bf59208b01b',
        relativeUrl: `?openModal=exportBookingData`,
    };
}
exports.exportBookingData = exportBookingData;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Sync Personal Calendar page in the dashboard.
 * @dashobardPageBuilder
 */
function syncPersonalCalendar() {
    return {
        pageId: 'bbbc8c7d-19b9-4a86-bfc5-1bf59208b01b',
        relativeUrl: `?openModal=syncPersonalCalendar`,
    };
}
exports.syncPersonalCalendar = syncPersonalCalendar;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Services List page in the dashboard.
 * @dashobardPageBuilder
 */
function servicesList() {
    return { pageId: 'c756378f-3438-41c6-a686-85e9fb0193ac' };
}
exports.servicesList = servicesList;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Staff Edit page in the dashboard.
 * @dashobardPageBuilder
 */
function staffEdit(options) {
    const { staffMemberId } = options;
    return {
        pageId: 'a6a6a0ee-865f-43cc-a8f2-b50984f74457',
        relativeUrl: `${staffMemberId}`,
    };
}
exports.staffEdit = staffEdit;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Staff List page in the dashboard.
 * @dashobardPageBuilder
 */
function staffList() {
    return { pageId: 'a4f36fac-ea18-483b-8236-095e29fcb726' };
}
exports.staffList = staffList;
