// @wix/auto_sdk_bookings_resources/build/internal/cjs/src/bookings-resources-v2-resource-resources.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortType = exports.SortOrder = exports.SortDirection = exports.ScalarType = exports.NestedAggregationType = exports.Mode = exports.MissingValues = exports.ManagementType = exports.Interval = exports.AggregationType = exports.onResourceUpdated = exports.onResourceDeleted = exports.onResourceCreated = exports.countResources = exports.queryResources = exports.searchResources = exports.bulkDeleteResources = exports.deleteResource = exports.updateResource = exports.getResource = exports.bulkCreateResources = exports.createResource = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_resources_v2_resource_resources_universal_js_1 = require("./bookings-resources-v2-resource-resources.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function createResource(httpClient) {
    return (resource) => (0, bookings_resources_v2_resource_resources_universal_js_1.createResource)(resource, 
    // @ts-ignore
    { httpClient });
}
exports.createResource = createResource;
function bulkCreateResources(httpClient) {
    return (resources, options) => (0, bookings_resources_v2_resource_resources_universal_js_1.bulkCreateResources)(resources, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateResources = bulkCreateResources;
function getResource(httpClient) {
    return (resourceId) => (0, bookings_resources_v2_resource_resources_universal_js_1.getResource)(resourceId, 
    // @ts-ignore
    { httpClient });
}
exports.getResource = getResource;
function updateResource(httpClient) {
    return (_id, resource) => (0, bookings_resources_v2_resource_resources_universal_js_1.updateResource)(_id, resource, 
    // @ts-ignore
    { httpClient });
}
exports.updateResource = updateResource;
function deleteResource(httpClient) {
    return (resourceId) => (0, bookings_resources_v2_resource_resources_universal_js_1.deleteResource)(resourceId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteResource = deleteResource;
function bulkDeleteResources(httpClient) {
    return (ids) => (0, bookings_resources_v2_resource_resources_universal_js_1.bulkDeleteResources)(ids, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteResources = bulkDeleteResources;
function searchResources(httpClient) {
    return (options) => (0, bookings_resources_v2_resource_resources_universal_js_1.searchResources)(options, 
    // @ts-ignore
    { httpClient });
}
exports.searchResources = searchResources;
function queryResources(httpClient) {
    return () => (0, bookings_resources_v2_resource_resources_universal_js_1.queryResources)(
    // @ts-ignore
    { httpClient });
}
exports.queryResources = queryResources;
function countResources(httpClient) {
    return (options) => (0, bookings_resources_v2_resource_resources_universal_js_1.countResources)(options, 
    // @ts-ignore
    { httpClient });
}
exports.countResources = countResources;
exports.onResourceCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.resources.v2.resource_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onResourceDeleted = (0, sdk_types_1.EventDefinition)('wix.bookings.resources.v2.resource_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onResourceUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.resources.v2.resource_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var bookings_resources_v2_resource_resources_universal_js_2 = require("./bookings-resources-v2-resource-resources.universal.js");
Object.defineProperty(exports, "AggregationType", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.AggregationType; } });
Object.defineProperty(exports, "Interval", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.Interval; } });
Object.defineProperty(exports, "ManagementType", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.ManagementType; } });
Object.defineProperty(exports, "MissingValues", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.MissingValues; } });
Object.defineProperty(exports, "Mode", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.Mode; } });
Object.defineProperty(exports, "NestedAggregationType", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.NestedAggregationType; } });
Object.defineProperty(exports, "ScalarType", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.ScalarType; } });
Object.defineProperty(exports, "SortDirection", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.SortDirection; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "SortType", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.SortType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_resources_v2_resource_resources_universal_js_2.WebhookIdentityType; } });
