// @wix/auto_sdk_members_member-followers/build/internal/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  followMember: () => followMember2,
  listCurrentMemberFollowers: () => listCurrentMemberFollowers,
  listCurrentMemberFollowing: () => listCurrentMemberFollowing,
  listMemberFollowers: () => listMemberFollowers2,
  listMemberFollowing: () => listMemberFollowing2,
  queryCurrentMemberConnections: () => queryCurrentMemberConnections,
  queryMemberConnections: () => queryMemberConnections2,
  unfollowMember: () => unfollowMember2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v3-follow-member-followers.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersFollowV3MemberFollowersUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      },
      {
        srcPath: "/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v3/followers",
        destPath: "/v3/followers"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-followers";
function followMember(payload) {
  function __followMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "POST",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.FollowMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __followMember;
}
function unfollowMember(payload) {
  function __unfollowMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "DELETE",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.UnfollowMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __unfollowMember;
}
function listMyMemberFollowing(payload) {
  function __listMyMemberFollowing({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMyMemberFollowing",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/my/following",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMyMemberFollowing;
}
function listMemberFollowing(payload) {
  function __listMemberFollowing({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMemberFollowing",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}/following",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMemberFollowing;
}
function listMyMemberFollowers(payload) {
  function __listMyMemberFollowers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMyMemberFollowers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/my",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMyMemberFollowers;
}
function listMemberFollowers(payload) {
  function __listMemberFollowers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "GET",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.ListMemberFollowers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMemberFollowers;
}
function queryMyMemberConnections(payload) {
  function __queryMyMemberConnections({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "POST",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.QueryMyMemberConnections",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/my/connections",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryMyMemberConnections;
}
function queryMemberConnections(payload) {
  function __queryMemberConnections({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v3.follow",
      method: "POST",
      methodFqn: "com.wixpress.members.follow.v3.MemberFollowers.QueryMemberConnections",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersFollowV3MemberFollowersUrl({
        protoPath: "/v3/followers/{memberId}/connections",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __queryMemberConnections;
}

// src/members-v3-follow-member-followers.meta.ts
function followMember2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = followMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/followers/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function unfollowMember2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = unfollowMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/followers/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCurrentMemberFollowing() {
  const payload = {};
  const getRequestOptions = listMyMemberFollowing(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/followers/my/following",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMemberFollowing2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = listMemberFollowing(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/followers/{memberId}/following",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCurrentMemberFollowers() {
  const payload = {};
  const getRequestOptions = listMyMemberFollowers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/followers/my",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMemberFollowers2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = listMemberFollowers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/followers/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCurrentMemberConnections() {
  const payload = {};
  const getRequestOptions = queryMyMemberConnections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/followers/my/connections",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMemberConnections2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = queryMemberConnections(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/followers/{memberId}/connections",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  followMember,
  listCurrentMemberFollowers,
  listCurrentMemberFollowing,
  listMemberFollowers,
  listMemberFollowing,
  queryCurrentMemberConnections,
  queryMemberConnections,
  unfollowMember
});
