// @wix/auto_sdk_bookings_resources/build/cjs/src/bookings-resources-v2-resource-resources.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.countResources = exports.queryResources = exports.searchResources = exports.bulkDeleteResources = exports.deleteResource = exports.updateResource = exports.getResource = exports.bulkCreateResources = exports.createResource = void 0;
const ambassadorWixBookingsResourcesV2Resource = __importStar(require("./bookings-resources-v2-resource-resources.http.js"));
function createResource() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.createResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createResource = createResource;
function bulkCreateResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.bulkCreateResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/resources/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateResources = bulkCreateResources;
function getResource() {
    const payload = { resourceId: ':resourceId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.getResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/resources/{resourceId}',
        pathParams: { resourceId: 'resourceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getResource = getResource;
function updateResource() {
    const payload = { resource: { id: ':resourceId' } };
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.updateResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/resources/{resource.id}',
        pathParams: { resourceId: 'resourceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateResource = updateResource;
function deleteResource() {
    const payload = { resourceId: ':resourceId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.deleteResource(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/resources/{resourceId}',
        pathParams: { resourceId: 'resourceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteResource = deleteResource;
function bulkDeleteResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.bulkDeleteResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/resources/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteResources = bulkDeleteResources;
function searchResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.searchResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchResources = searchResources;
function queryResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.queryResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryResources = queryResources;
function countResources() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2Resource.countResources(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.countResources = countResources;
