// @wix/bookings/build/cjs/meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.pricing = exports.appSettings = exports.bookings = exports.attendance = exports.categories = exports.bookingPolicies = exports.staffMembers = exports.services = exports.resourceTypes = exports.resources = exports.extendedBookings = exports.bookingPolicySnapshots = exports.bookingFees = exports.serviceCategories = exports.serviceOptionsAndVariants = exports.externalCalendars = exports.multiServiceAvailabilityTimeSlots = exports.availabilityTimeSlots = exports.availabilityCalendar = void 0;
exports.availabilityCalendar = __importStar(require("@wix/auto_sdk_bookings_availability-calendar/meta"));
exports.availabilityTimeSlots = __importStar(require("@wix/auto_sdk_bookings_availability-time-slots/meta"));
exports.multiServiceAvailabilityTimeSlots = __importStar(require("@wix/auto_sdk_bookings_multi-service-availability-time-slots/meta"));
exports.externalCalendars = __importStar(require("@wix/auto_sdk_bookings_external-calendars/meta"));
exports.serviceOptionsAndVariants = __importStar(require("@wix/auto_sdk_bookings_service-options-and-variants/meta"));
exports.serviceCategories = __importStar(require("@wix/auto_sdk_bookings_service-categories/meta"));
exports.bookingFees = __importStar(require("@wix/auto_sdk_bookings_booking-fees/meta"));
exports.bookingPolicySnapshots = __importStar(require("@wix/auto_sdk_bookings_booking-policy-snapshots/meta"));
exports.extendedBookings = __importStar(require("@wix/auto_sdk_bookings_extended-bookings/meta"));
exports.resources = __importStar(require("@wix/auto_sdk_bookings_resources/meta"));
exports.resourceTypes = __importStar(require("@wix/auto_sdk_bookings_resource-types/meta"));
exports.services = __importStar(require("@wix/auto_sdk_bookings_services/meta"));
exports.staffMembers = __importStar(require("@wix/auto_sdk_bookings_staff-members/meta"));
exports.bookingPolicies = __importStar(require("@wix/auto_sdk_bookings_booking-policies/meta"));
exports.categories = __importStar(require("@wix/auto_sdk_bookings_categories/meta"));
exports.attendance = __importStar(require("@wix/auto_sdk_bookings_attendance/meta"));
exports.bookings = __importStar(require("@wix/auto_sdk_bookings_bookings/meta"));
exports.appSettings = __importStar(require("@wix/auto_sdk_bookings_app-settings/meta"));
exports.pricing = __importStar(require("@wix/auto_sdk_bookings_pricing/meta"));
