// @wix/auto_sdk_members_user-member/build/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  queryUserMembers: () => queryUserMembers2
});
module.exports = __toCommonJS(meta_exports);

// src/members-usermembers-v1-user-member-user-member.http.ts
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
function resolveWixMembersUsermembersV1UserMembersUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/user-members-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/user-members",
        destPath: "/v1/user-members"
      }
    ]
  };
  return (0, import_rest_modules.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_user-member";
function queryUserMembers(payload) {
  function __queryUserMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.usermembers.v1.user_member",
      method: "POST",
      methodFqn: "wix.members.usermembers.v1.UserMembers.QueryUserMembers",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersUsermembersV1UserMembersUrl({
        protoPath: "/v1/user-members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "userMembers.createdDate" },
            { path: "userMembers.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryUserMembers;
}

// src/members-usermembers-v1-user-member-user-member.meta.ts
function queryUserMembers2() {
  const payload = {};
  const getRequestOptions = queryUserMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/user-members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  queryUserMembers
});
