// @wix/auto_sdk_bookings_external-calendars/build/cjs/src/bookings-calendar-v2-external-calendar-external-calendars.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transparency = exports.SyncToErrorReason = exports.SyncToCalendar = exports.Status = exports.SessionType = exports.SessionStatus = exports.ScheduleStatus = exports.RecurringIntervalType = exports.LocationsLocationType = exports.LocationType = exports.LocationStatus = exports.ListEventFromCalendars = exports.ErrorReason = exports.DayOfWeek = exports.Day = exports.ConnectMethod = exports.ConferenceType = exports.CalendarType = exports.ApprovalStatus = exports.listEvents = exports.disconnect = exports.updateSyncConfig = exports.listCalendars = exports.connectByCredentials = exports.connectByOAuth = exports.listConnections = exports.getConnection = exports.listProviders = exports.__metadata = void 0;
const bookings_calendar_v2_external_calendar_external_calendars_universal_js_1 = require("./bookings-calendar-v2-external-calendar-external-calendars.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function listProviders(httpClient) {
    return () => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.listProviders)(
    // @ts-ignore
    { httpClient });
}
exports.listProviders = listProviders;
function getConnection(httpClient) {
    return (connectionId) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.getConnection)(connectionId, 
    // @ts-ignore
    { httpClient });
}
exports.getConnection = getConnection;
function listConnections(httpClient) {
    return (options) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.listConnections)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listConnections = listConnections;
function connectByOAuth(httpClient) {
    return (providerId, scheduleId, redirectUrl) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.connectByOAuth)(providerId, scheduleId, redirectUrl, 
    // @ts-ignore
    { httpClient });
}
exports.connectByOAuth = connectByOAuth;
function connectByCredentials(httpClient) {
    return (providerId, scheduleId, email, password) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.connectByCredentials)(providerId, scheduleId, email, password, 
    // @ts-ignore
    { httpClient });
}
exports.connectByCredentials = connectByCredentials;
function listCalendars(httpClient) {
    return (connectionId) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.listCalendars)(connectionId, 
    // @ts-ignore
    { httpClient });
}
exports.listCalendars = listCalendars;
function updateSyncConfig(httpClient) {
    return (connectionId, syncConfig) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.updateSyncConfig)(connectionId, syncConfig, 
    // @ts-ignore
    { httpClient });
}
exports.updateSyncConfig = updateSyncConfig;
function disconnect(httpClient) {
    return (connectionId) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.disconnect)(connectionId, 
    // @ts-ignore
    { httpClient });
}
exports.disconnect = disconnect;
function listEvents(httpClient) {
    return (options) => (0, bookings_calendar_v2_external_calendar_external_calendars_universal_js_1.listEvents)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listEvents = listEvents;
var bookings_calendar_v2_external_calendar_external_calendars_universal_js_2 = require("./bookings-calendar-v2-external-calendar-external-calendars.universal.js");
Object.defineProperty(exports, "ApprovalStatus", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.ApprovalStatus; } });
Object.defineProperty(exports, "CalendarType", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.CalendarType; } });
Object.defineProperty(exports, "ConferenceType", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.ConferenceType; } });
Object.defineProperty(exports, "ConnectMethod", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.ConnectMethod; } });
Object.defineProperty(exports, "Day", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.Day; } });
Object.defineProperty(exports, "DayOfWeek", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.DayOfWeek; } });
Object.defineProperty(exports, "ErrorReason", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.ErrorReason; } });
Object.defineProperty(exports, "ListEventFromCalendars", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.ListEventFromCalendars; } });
Object.defineProperty(exports, "LocationStatus", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.LocationStatus; } });
Object.defineProperty(exports, "LocationType", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.LocationType; } });
Object.defineProperty(exports, "LocationsLocationType", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.LocationsLocationType; } });
Object.defineProperty(exports, "RecurringIntervalType", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.RecurringIntervalType; } });
Object.defineProperty(exports, "ScheduleStatus", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.ScheduleStatus; } });
Object.defineProperty(exports, "SessionStatus", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.SessionStatus; } });
Object.defineProperty(exports, "SessionType", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.SessionType; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.Status; } });
Object.defineProperty(exports, "SyncToCalendar", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.SyncToCalendar; } });
Object.defineProperty(exports, "SyncToErrorReason", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.SyncToErrorReason; } });
Object.defineProperty(exports, "Transparency", { enumerable: true, get: function () { return bookings_calendar_v2_external_calendar_external_calendars_universal_js_2.Transparency; } });
