// @wix/auto_sdk_bookings_external-calendars/build/cjs/src/bookings-calendar-v2-external-calendar-external-calendars.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionType = exports.SessionStatus = exports.ConferenceType = exports.ScheduleStatus = exports.ApprovalStatus = exports.DayOfWeek = exports.LocationsLocationType = exports.LocationStatus = exports.LocationType = exports.RecurringIntervalType = exports.Transparency = exports.Day = exports.SyncToErrorReason = exports.ErrorReason = exports.Status = exports.SyncToCalendar = exports.ListEventFromCalendars = exports.ConnectMethod = exports.CalendarType = void 0;
var CalendarType;
(function (CalendarType) {
    /** There is no information about the external calendar type. */
    CalendarType["UNDEFINED"] = "UNDEFINED";
    /** [Google Calendar](https://developers.google.com/calendar/api/guides/overview). */
    CalendarType["GOOGLE"] = "GOOGLE";
    /** Apple iCalendar. */
    CalendarType["I_CAL"] = "I_CAL";
    /** __Deprecated__. Use `MICROSOFT` instead. */
    CalendarType["OUTLOOK"] = "OUTLOOK";
    /** __Deprecated__. Use `MICROSOFT` instead. */
    CalendarType["OFFICE_365"] = "OFFICE_365";
    /** Microsoft Calendar. For example, Office 365 calendar or Outlook calendar. */
    CalendarType["MICROSOFT"] = "MICROSOFT";
    /** A different type of external calendar, not listed here. */
    CalendarType["OTHER"] = "OTHER";
})(CalendarType || (exports.CalendarType = CalendarType = {}));
var ConnectMethod;
(function (ConnectMethod) {
    /**
     * There is no information about how you can connect the external calendar
     * to the Wix site.
     */
    ConnectMethod["UNDEFINED"] = "UNDEFINED";
    /**
     * You can connect the external calendar to the Wix site by calling *Connect By O Auth*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/bookings/external-calendars/connect-by-oauth) | [REST](https://dev.wix.com/docs/rest/business-solutions/bookings/calendar/external-calendar-v2/connect-by-o-auth)).
     */
    ConnectMethod["OAUTH"] = "OAUTH";
    /**
     * You can connect the external calendar to the Wix site by calling *Connect By Credentials*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/bookings/external-calendars/connect-by-credentials) | [REST](https://dev.wix.com/docs/rest/business-solutions/bookings/calendar/external-calendar-v2/connect-by-credentials)).
     */
    ConnectMethod["CREDENTIALS"] = "CREDENTIALS";
})(ConnectMethod || (exports.ConnectMethod = ConnectMethod = {}));
var ListEventFromCalendars;
(function (ListEventFromCalendars) {
    /**
     * There is no information about which external calendar events you
     * can import to the Wix site.
     */
    ListEventFromCalendars["UNDEFINED"] = "UNDEFINED";
    /** You can't import any events from the external calendar to the Wix site. */
    ListEventFromCalendars["NOT_SUPPORTED"] = "NOT_SUPPORTED";
    /**
     * You can import events only for the external calendar that's
     * designated as primary.
     */
    ListEventFromCalendars["PRIMARY_CALENDAR_ONLY"] = "PRIMARY_CALENDAR_ONLY";
    /**
     * You can import events only for a specific external calendar account.
     * You can call _List Calendars_
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/bookings/external-calendars/list-calendars) | [REST](https://dev.wix.com/docs/rest/business-solutions/bookings/calendar/external-calendar-v2/list-calendars))
     * to retrieve a complete list of external calendar accounts.
     */
    ListEventFromCalendars["SPECIFIC_CALENDARS"] = "SPECIFIC_CALENDARS";
})(ListEventFromCalendars || (exports.ListEventFromCalendars = ListEventFromCalendars = {}));
var SyncToCalendar;
(function (SyncToCalendar) {
    /**
     * There is no information about which Wix calendar *events*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/calendar/events/introduction) | [REST](https://dev.wix.com/docs/rest/business-management/calendar/events-v3/introduction))
     * you can export to the external calendar.
     */
    SyncToCalendar["UNDEFINED"] = "UNDEFINED";
    /**
     * You can't export any Wix calendar *events*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/calendar/events/introduction) | [REST](https://dev.wix.com/docs/rest/business-management/calendar/events-v3/introduction))
     * to the external calendar.
     */
    SyncToCalendar["NOT_SUPPORTED"] = "NOT_SUPPORTED";
    /**
     * You can export Wix calendar *events*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/calendar/events/introduction) | [REST](https://dev.wix.com/docs/rest/business-management/calendar/events-v3/introduction))
     * only to the external calendar that's designated as primary.
     */
    SyncToCalendar["PRIMARY_CALENDAR_ONLY"] = "PRIMARY_CALENDAR_ONLY";
    /** Events can be exported to specific external calendars. Use [List Calendars](#list-calendars) to get a list of calendars for a connected external calendar account. */
    SyncToCalendar["SPECIFIC_CALENDAR"] = "SPECIFIC_CALENDAR";
    /**
     * You can export Wix calendar *events*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/calendar/events/introduction) | [REST](https://dev.wix.com/docs/rest/business-management/calendar/events-v3/introduction))
     * only to a dedicated external calendar account. You can
     * call _List Calendars_
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/bookings/external-calendars/list-calendars) | [REST](https://dev.wix.com/docs/rest/business-solutions/bookings/calendar/external-calendar-v2/list-calendars))
     * to retrieve a complete list of external calendar accounts.
     */
    SyncToCalendar["DEDICATED_CALENDAR"] = "DEDICATED_CALENDAR";
})(SyncToCalendar || (exports.SyncToCalendar = SyncToCalendar = {}));
var Status;
(function (Status) {
    /** There is no information about the connection status. */
    Status["UNDEFINED"] = "UNDEFINED";
    /**
     * The external calendar is connected to the Wix *schedule*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/calendar/schedules/introduction) | [REST](https://dev.wix.com/docs/rest/business-management/calendar/schedules-v3/introduction)),
     * but the sync process hasn't started yet.
     */
    Status["CONNECTED"] = "CONNECTED";
    /** Sync process is ongoing. */
    Status["SYNC_IN_PROGRESS"] = "SYNC_IN_PROGRESS";
    /** The Wix calendar and the external calendar are in sync. */
    Status["SYNCED"] = "SYNCED";
    /**
     * The external calender has been disconnected from the Wix *schedule*
     * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/calendar/schedules/introduction) | [REST](https://dev.wix.com/docs/rest/business-management/calendar/schedules-v3/introduction)).
     */
    Status["DISCONNECTED"] = "DISCONNECTED";
    /** The calendars sync is in error state. */
    Status["ERROR"] = "ERROR";
})(Status || (exports.Status = Status = {}));
var ErrorReason;
(function (ErrorReason) {
    /** There is no information about the connection error. */
    ErrorReason["UNDEFINED"] = "UNDEFINED";
    /** The external calendar's access token has been revoked. */
    ErrorReason["TOKEN_REVOKED"] = "TOKEN_REVOKED";
    /** The external calendar couldn't be created. */
    ErrorReason["EXTERNAL_CALENDAR_CREATION_FAILED"] = "EXTERNAL_CALENDAR_CREATION_FAILED";
    /** The external calendar was deleted. */
    ErrorReason["EXTERNAL_CALENDAR_DELETED"] = "EXTERNAL_CALENDAR_DELETED";
})(ErrorReason || (exports.ErrorReason = ErrorReason = {}));
var SyncToErrorReason;
(function (SyncToErrorReason) {
    /** No sync error. */
    SyncToErrorReason["UNDEFINED"] = "UNDEFINED";
    /** Could not create calendar to sync sessions to. */
    SyncToErrorReason["CALENDAR_CREATION_FAILURE"] = "CALENDAR_CREATION_FAILURE";
    /** Calendar was deleted while sync was in progress. */
    SyncToErrorReason["CALENDAR_DELETED"] = "CALENDAR_DELETED";
})(SyncToErrorReason || (exports.SyncToErrorReason = SyncToErrorReason = {}));
var Day;
(function (Day) {
    /** Undefined. */
    Day["UNDEFINED"] = "UNDEFINED";
    /** Monday. */
    Day["MON"] = "MON";
    /** Tuesday. */
    Day["TUE"] = "TUE";
    /** Wednesday. */
    Day["WED"] = "WED";
    /** Thursday. */
    Day["THU"] = "THU";
    /** Friday. */
    Day["FRI"] = "FRI";
    /** Saturday. */
    Day["SAT"] = "SAT";
    /** Sunday. */
    Day["SUN"] = "SUN";
})(Day || (exports.Day = Day = {}));
var Transparency;
(function (Transparency) {
    Transparency["UNDEFINED"] = "UNDEFINED";
    /** The schedule can have available slots during the linked schedule's sessions. */
    Transparency["FREE"] = "FREE";
    /** The schedule can't have available slots during the linked schedule's sessions. */
    Transparency["BUSY"] = "BUSY";
})(Transparency || (exports.Transparency = Transparency = {}));
var RecurringIntervalType;
(function (RecurringIntervalType) {
    /** The default value. Sessions for this interval will be of type EVENT. */
    RecurringIntervalType["UNDEFINED"] = "UNDEFINED";
    /** A recurring interval of events. */
    RecurringIntervalType["EVENT"] = "EVENT";
    /** Deprecated. */
    RecurringIntervalType["TIME_AVAILABILITY"] = "TIME_AVAILABILITY";
    /** A recurring interval for availability. */
    RecurringIntervalType["AVAILABILITY"] = "AVAILABILITY";
})(RecurringIntervalType || (exports.RecurringIntervalType = RecurringIntervalType = {}));
var LocationType;
(function (LocationType) {
    LocationType["UNDEFINED"] = "UNDEFINED";
    LocationType["OWNER_BUSINESS"] = "OWNER_BUSINESS";
    LocationType["OWNER_CUSTOM"] = "OWNER_CUSTOM";
    LocationType["CUSTOM"] = "CUSTOM";
})(LocationType || (exports.LocationType = LocationType = {}));
/** For future use */
var LocationStatus;
(function (LocationStatus) {
    LocationStatus["ACTIVE"] = "ACTIVE";
    LocationStatus["INACTIVE"] = "INACTIVE";
})(LocationStatus || (exports.LocationStatus = LocationStatus = {}));
/** For future use */
var LocationsLocationType;
(function (LocationsLocationType) {
    LocationsLocationType["UNKNOWN"] = "UNKNOWN";
    LocationsLocationType["BRANCH"] = "BRANCH";
    LocationsLocationType["OFFICES"] = "OFFICES";
    LocationsLocationType["RECEPTION"] = "RECEPTION";
    LocationsLocationType["HEADQUARTERS"] = "HEADQUARTERS";
    LocationsLocationType["INVENTORY"] = "INVENTORY";
})(LocationsLocationType || (exports.LocationsLocationType = LocationsLocationType = {}));
/** Enumerates the days of the week. */
var DayOfWeek;
(function (DayOfWeek) {
    DayOfWeek["MONDAY"] = "MONDAY";
    DayOfWeek["TUESDAY"] = "TUESDAY";
    DayOfWeek["WEDNESDAY"] = "WEDNESDAY";
    DayOfWeek["THURSDAY"] = "THURSDAY";
    DayOfWeek["FRIDAY"] = "FRIDAY";
    DayOfWeek["SATURDAY"] = "SATURDAY";
    DayOfWeek["SUNDAY"] = "SUNDAY";
})(DayOfWeek || (exports.DayOfWeek = DayOfWeek = {}));
var ApprovalStatus;
(function (ApprovalStatus) {
    /** Default. */
    ApprovalStatus["UNDEFINED"] = "UNDEFINED";
    /** Pending business approval. */
    ApprovalStatus["PENDING"] = "PENDING";
    /** Approved by the business. */
    ApprovalStatus["APPROVED"] = "APPROVED";
    /** Declined by the business. */
    ApprovalStatus["DECLINED"] = "DECLINED";
})(ApprovalStatus || (exports.ApprovalStatus = ApprovalStatus = {}));
var ScheduleStatus;
(function (ScheduleStatus) {
    /** Undefined schedule status. */
    ScheduleStatus["UNDEFINED"] = "UNDEFINED";
    /** The schedule was created. */
    ScheduleStatus["CREATED"] = "CREATED";
    /** The schedule was cancelled. */
    ScheduleStatus["CANCELLED"] = "CANCELLED";
})(ScheduleStatus || (exports.ScheduleStatus = ScheduleStatus = {}));
var ConferenceType;
(function (ConferenceType) {
    /** Undefined conference type. */
    ConferenceType["UNDEFINED"] = "UNDEFINED";
    /** API-generated online meeting. */
    ConferenceType["ONLINE_MEETING_PROVIDER"] = "ONLINE_MEETING_PROVIDER";
    /** User-defined meeting. */
    ConferenceType["CUSTOM"] = "CUSTOM";
})(ConferenceType || (exports.ConferenceType = ConferenceType = {}));
var SessionStatus;
(function (SessionStatus) {
    /** Undefined status. */
    SessionStatus["UNDEFINED"] = "UNDEFINED";
    /** Session is confirmed. Default status. */
    SessionStatus["CONFIRMED"] = "CONFIRMED";
    /**
     * Session is cancelled.
     * A cancelled session can be the cancellation of a recurring session that should no longer be displayed or a deleted single session.
     * The ListSessions returns cancelled sessions only if 'includeDelete' flag is set to true.
     */
    SessionStatus["CANCELLED"] = "CANCELLED";
})(SessionStatus || (exports.SessionStatus = SessionStatus = {}));
var SessionType;
(function (SessionType) {
    SessionType["UNDEFINED"] = "UNDEFINED";
    /**
     * Creates an event on the calendar for the owner of the schedule that the session belongs to.
     * Default type.
     */
    SessionType["EVENT"] = "EVENT";
    /** Represents a resource's available working hours. */
    SessionType["WORKING_HOURS"] = "WORKING_HOURS";
    /** Deprecated. Please use WORKING_HOURS. */
    SessionType["TIME_AVAILABILITY"] = "TIME_AVAILABILITY";
    /** Deprecated. Represents a resource's available hours. Please use WORKING_HOURS. */
    SessionType["AVAILABILITY"] = "AVAILABILITY";
})(SessionType || (exports.SessionType = SessionType = {}));
