// @wix/auto_sdk_bookings_external-calendars/build/cjs/src/bookings-calendar-v2-external-calendar-external-calendars.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listEvents = exports.disconnect = exports.updateSyncConfig = exports.listCalendars = exports.connectByCredentials = exports.connectByOAuth = exports.listConnections = exports.getConnection = exports.listProviders = void 0;
const ambassadorWixBookingsCalendarV2ExternalCalendar = __importStar(require("./bookings-calendar-v2-external-calendar-external-calendars.http.js"));
function listProviders() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listProviders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/providers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listProviders = listProviders;
function getConnection() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.getConnection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/connections/{connectionId}',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getConnection = getConnection;
function listConnections() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listConnections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/connections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listConnections = listConnections;
function connectByOAuth() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.connectByOAuth(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/external-calendars/connections:connectByOAuth',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.connectByOAuth = connectByOAuth;
function connectByCredentials() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.connectByCredentials(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/external-calendars/connections:connectByCredentials',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.connectByCredentials = connectByCredentials;
function listCalendars() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listCalendars(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/connections/{connectionId}/calendars',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listCalendars = listCalendars;
function updateSyncConfig() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.updateSyncConfig(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/external-calendars/connections/{connectionId}/sync-config',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateSyncConfig = updateSyncConfig;
function disconnect() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.disconnect(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/external-calendars/connections/{connectionId}/disconnect',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.disconnect = disconnect;
function listEvents() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listEvents(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/events',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listEvents = listEvents;
