// @wix/auto_sdk_members_member-role-definition/build/internal/meta.mjs
// src/members-v1-member-role-definition-member-role-definition.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/roles",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/roles",
        destPath: "/v1/roles"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-role-definition";
function createMemberRoleDefinition(payload) {
  function __createMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "POST",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.CreateMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __createMemberRoleDefinition;
}
function listMemberRoleDefinitions(payload) {
  function __listMemberRoleDefinitions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "GET",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.ListMemberRoleDefinitions",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __listMemberRoleDefinitions;
}
function getMemberRoleDefinition(payload) {
  function __getMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "GET",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.GetMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions/{roleKey}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getMemberRoleDefinition;
}
function updateMemberRoleDefinition(payload) {
  function __updateMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "PATCH",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.UpdateMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions/{key}",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __updateMemberRoleDefinition;
}
function deleteMemberRoleDefinition(payload) {
  function __deleteMemberRoleDefinition({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_role_definition",
      method: "DELETE",
      methodFqn: "com.wixpress.members.memberroles.MemberRoleDefinitions.DeleteMemberRoleDefinition",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberrolesMemberRoleDefinitionsUrl({
        protoPath: "/v1/roles/definitions/{roleKey}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteMemberRoleDefinition;
}

// src/members-v1-member-role-definition-member-role-definition.meta.ts
function createMemberRoleDefinition2() {
  const payload = {};
  const getRequestOptions = createMemberRoleDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/roles/definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMemberRoleDefinitions2() {
  const payload = {};
  const getRequestOptions = listMemberRoleDefinitions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/roles/definitions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMemberRoleDefinition2() {
  const payload = { roleKey: ":roleKey" };
  const getRequestOptions = getMemberRoleDefinition(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/roles/definitions/{roleKey}",
    pathParams: { roleKey: "roleKey" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMemberRoleDefinition2() {
  const payload = { key: ":key" };
  const getRequestOptions = updateMemberRoleDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/roles/definitions/{key}",
    pathParams: { key: "key" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberRoleDefinition2() {
  const payload = { roleKey: ":roleKey" };
  const getRequestOptions = deleteMemberRoleDefinition(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/roles/definitions/{roleKey}",
    pathParams: { roleKey: "roleKey" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createMemberRoleDefinition2 as createMemberRoleDefinition,
  deleteMemberRoleDefinition2 as deleteMemberRoleDefinition,
  getMemberRoleDefinition2 as getMemberRoleDefinition,
  listMemberRoleDefinitions2 as listMemberRoleDefinitions,
  updateMemberRoleDefinition2 as updateMemberRoleDefinition
};
