// @wix/auto_sdk_bookings_booking-policies/build/cjs/src/bookings-v1-booking-policy-booking-policies.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.State = exports.SortOrder = exports.SiteCreatedContext = exports.ResolutionMethod = exports.PlacementType = exports.Namespace = exports.DeleteStatus = exports.DayOfWeek = exports.onBookingPolicyUpdated = exports.onBookingPolicyDeleted = exports.onBookingPolicyDefaultBookingPolicySet = exports.onBookingPolicyCreated = exports.countBookingPolicies = exports.queryBookingPolicies = exports.deleteBookingPolicy = exports.setDefaultBookingPolicy = exports.updateBookingPolicy = exports.getStrictestBookingPolicy = exports.getBookingPolicy = exports.createBookingPolicy = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_v1_booking_policy_booking_policies_universal_js_1 = require("./bookings-v1-booking-policy-booking-policies.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function createBookingPolicy(httpClient) {
    return (bookingPolicy) => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.createBookingPolicy)(bookingPolicy, 
    // @ts-ignore
    { httpClient });
}
exports.createBookingPolicy = createBookingPolicy;
function getBookingPolicy(httpClient) {
    return (bookingPolicyId) => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.getBookingPolicy)(bookingPolicyId, 
    // @ts-ignore
    { httpClient });
}
exports.getBookingPolicy = getBookingPolicy;
function getStrictestBookingPolicy(httpClient) {
    return (bookingPolicyIds) => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.getStrictestBookingPolicy)(bookingPolicyIds, 
    // @ts-ignore
    { httpClient });
}
exports.getStrictestBookingPolicy = getStrictestBookingPolicy;
function updateBookingPolicy(httpClient) {
    return (_id, bookingPolicy) => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.updateBookingPolicy)(_id, bookingPolicy, 
    // @ts-ignore
    { httpClient });
}
exports.updateBookingPolicy = updateBookingPolicy;
function setDefaultBookingPolicy(httpClient) {
    return (bookingPolicyId) => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.setDefaultBookingPolicy)(bookingPolicyId, 
    // @ts-ignore
    { httpClient });
}
exports.setDefaultBookingPolicy = setDefaultBookingPolicy;
function deleteBookingPolicy(httpClient) {
    return (bookingPolicyId) => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.deleteBookingPolicy)(bookingPolicyId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteBookingPolicy = deleteBookingPolicy;
function queryBookingPolicies(httpClient) {
    return () => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.queryBookingPolicies)(
    // @ts-ignore
    { httpClient });
}
exports.queryBookingPolicies = queryBookingPolicies;
function countBookingPolicies(httpClient) {
    return (options) => (0, bookings_v1_booking_policy_booking_policies_universal_js_1.countBookingPolicies)(options, 
    // @ts-ignore
    { httpClient });
}
exports.countBookingPolicies = countBookingPolicies;
exports.onBookingPolicyCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.v1.booking_policy_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onBookingPolicyDefaultBookingPolicySet = (0, sdk_types_1.EventDefinition)('wix.bookings.v1.booking_policy_default_booking_policy_set', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.currentDefaultBookingPolicy.createdDate' },
            { path: 'data.currentDefaultBookingPolicy.updatedDate' },
            { path: 'data.previousDefaultBookingPolicy.createdDate' },
            { path: 'data.previousDefaultBookingPolicy.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onBookingPolicyDeleted = (0, sdk_types_1.EventDefinition)('wix.bookings.v1.booking_policy_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onBookingPolicyUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.v1.booking_policy_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var bookings_v1_booking_policy_booking_policies_universal_js_2 = require("./bookings-v1-booking-policy-booking-policies.universal.js");
Object.defineProperty(exports, "DayOfWeek", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.DayOfWeek; } });
Object.defineProperty(exports, "DeleteStatus", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.DeleteStatus; } });
Object.defineProperty(exports, "Namespace", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.Namespace; } });
Object.defineProperty(exports, "PlacementType", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.PlacementType; } });
Object.defineProperty(exports, "ResolutionMethod", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.ResolutionMethod; } });
Object.defineProperty(exports, "SiteCreatedContext", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.SiteCreatedContext; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "State", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.State; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_v1_booking_policy_booking_policies_universal_js_2.WebhookIdentityType; } });
