// @wix/auto_sdk_members_custom-fields/build/internal/meta.mjs
// src/members-v1-custom-field-custom-fields.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersCustomfieldsCustomFieldsUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields",
        destPath: "/v1/custom-fields"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/custom-fields",
        destPath: "/v1/custom-fields"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields",
        destPath: "/v1/custom-fields"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-fields",
        destPath: "/v1/custom-fields"
      }
    ],
    "editor-flow.wixapps.net": [
      {
        srcPath: "/_api/members/v1/custom-fields",
        destPath: "/v1/custom-fields"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/custom-fields",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/custom-fields",
        destPath: "/v1/custom-fields"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-fields";
function createCustomField(payload) {
  function __createCustomField({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "field.createdDate" }, { path: "field.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.CustomFields.CreateCustomField",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsCustomFieldsUrl({
        protoPath: "/v1/custom-fields",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "field.createdDate" },
            { path: "field.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCustomField;
}
function getCustomField(payload) {
  function __getCustomField({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.CustomFields.GetCustomField",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsCustomFieldsUrl({
        protoPath: "/v1/custom-fields/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "field.createdDate" },
            { path: "field.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCustomField;
}
function listCustomFields(payload) {
  function __listCustomFields({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.CustomFields.ListCustomFields",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsCustomFieldsUrl({
        protoPath: "/v1/custom-fields",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fields.createdDate" },
            { path: "fields.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listCustomFields;
}
function updateCustomField(payload) {
  function __updateCustomField({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "field.createdDate" }, { path: "field.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field",
      method: "PATCH",
      methodFqn: "com.wixpress.members.customfields.CustomFields.UpdateCustomField",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsCustomFieldsUrl({
        protoPath: "/v1/custom-fields/{field.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "field.createdDate" },
            { path: "field.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCustomField;
}
function deleteCustomField(payload) {
  function __deleteCustomField({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field",
      method: "DELETE",
      methodFqn: "com.wixpress.members.customfields.CustomFields.DeleteCustomField",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsCustomFieldsUrl({
        protoPath: "/v1/custom-fields/{id}",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCustomField;
}
function hideCustomField(payload) {
  function __hideCustomField({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.CustomFields.HideCustomField",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsCustomFieldsUrl({
        protoPath: "/v1/custom-fields/{id}/hide",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __hideCustomField;
}
function updateCustomFieldsOrder(payload) {
  function __updateCustomFieldsOrder({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.CustomFields.UpdateCustomFieldsOrder",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsCustomFieldsUrl({
        protoPath: "/v1/custom-fields/order",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "fields.createdDate" },
            { path: "fields.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCustomFieldsOrder;
}

// src/members-v1-custom-field-custom-fields.meta.ts
function createCustomField2() {
  const payload = {};
  const getRequestOptions = createCustomField(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomField2() {
  const payload = { id: ":id" };
  const getRequestOptions = getCustomField(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/custom-fields/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCustomFields2() {
  const payload = {};
  const getRequestOptions = listCustomFields(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/custom-fields",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCustomField2() {
  const payload = { field: { id: ":fieldId" } };
  const getRequestOptions = updateCustomField(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/custom-fields/{field.id}",
    pathParams: { fieldId: "fieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCustomField2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteCustomField(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/custom-fields/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function hideCustomField2() {
  const payload = { id: ":id" };
  const getRequestOptions = hideCustomField(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields/{id}/hide",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCustomFieldsOrder2() {
  const payload = {};
  const getRequestOptions = updateCustomFieldsOrder(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields/order",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createCustomField2 as createCustomField,
  deleteCustomField2 as deleteCustomField,
  getCustomField2 as getCustomField,
  hideCustomField2 as hideCustomField,
  listCustomFields2 as listCustomFields,
  updateCustomField2 as updateCustomField,
  updateCustomFieldsOrder2 as updateCustomFieldsOrder
};
