// @wix/auto_sdk_bookings_services/build/internal/cjs/src/bookings-services-v2-service-services.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.cloneService = exports.validateSlug = exports.setCustomSlug = exports.disablePricingPlansForService = exports.enablePricingPlansForService = exports.setServiceLocations = exports.queryCategories = exports.queryLocations = exports.countServices = exports.queryBookingForms = exports.queryPolicies = exports.searchServices = exports.queryServices = exports.bulkDeleteServicesByFilter = exports.bulkDeleteServices = exports.deleteService = exports.bulkUpdateServicesByFilter = exports.bulkUpdateServices = exports.updateService = exports.getService = exports.createService = void 0;
const ambassadorWixBookingsServicesV2Service = __importStar(require("./bookings-services-v2-service-services.http.js"));
function createService() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.createService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createService = createService;
function getService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.getService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/services/{serviceId}',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getService = getService;
function updateService() {
    const payload = { service: { id: ':serviceId' } };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.updateService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/services/{service.id}',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateService = updateService;
function bulkUpdateServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkUpdateServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateServices = bulkUpdateServices;
function bulkUpdateServicesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkUpdateServicesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/update-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateServicesByFilter = bulkUpdateServicesByFilter;
function deleteService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.deleteService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/services/{serviceId}',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteService = deleteService;
function bulkDeleteServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkDeleteServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteServices = bulkDeleteServices;
function bulkDeleteServicesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkDeleteServicesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/delete-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteServicesByFilter = bulkDeleteServicesByFilter;
function queryServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryServices = queryServices;
function searchServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.searchServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.searchServices = searchServices;
function queryPolicies() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryPolicies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/policies/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryPolicies = queryPolicies;
function queryBookingForms() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryBookingForms(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/booking-forms/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryBookingForms = queryBookingForms;
function countServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.countServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.countServices = countServices;
function queryLocations() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryLocations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/locations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryLocations = queryLocations;
function queryCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/categories/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryCategories = queryCategories;
function setServiceLocations() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.setServiceLocations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/locations',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setServiceLocations = setServiceLocations;
function enablePricingPlansForService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.enablePricingPlansForService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/pricing-plans/add',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.enablePricingPlansForService = enablePricingPlansForService;
function disablePricingPlansForService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.disablePricingPlansForService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/pricing-plans/remove',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.disablePricingPlansForService = disablePricingPlansForService;
function setCustomSlug() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.setCustomSlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/slugs/custom',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setCustomSlug = setCustomSlug;
function validateSlug() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.validateSlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/slugs/validate',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.validateSlug = validateSlug;
function cloneService() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.cloneService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/clone',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.cloneService = cloneService;
