// @wix/auto_sdk_bookings_services/build/internal/cjs/src/bookings-services-v2-service-services.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecurringIntervalType = exports.RateType = exports.PlacementType = exports.NestedAggregationType = exports.Mode = exports.MissingValues = exports.LocationsLocationType = exports.LocationTypeEnumLocationType = exports.LocationType = exports.LocationStatus = exports.InvalidSlugError = exports.Event = exports.DayOfWeek = exports.Day = exports.DateHistogramAggregationInterval = exports.CrudType = exports.ConferenceType = exports.CloneErrors = exports.CategoryStatus = exports.CategoryNotificationEvent = exports.CalendarType = exports.BenefitType = exports.ApprovalStatus = exports.AggregationType = exports.Action = exports.onServiceUpdated = exports.onServiceDeleted = exports.onServiceCreated = exports.cloneService = exports.validateSlug = exports.setCustomSlug = exports.disablePricingPlansForService = exports.enablePricingPlansForService = exports.setServiceLocations = exports.queryCategories = exports.queryLocations = exports.countServices = exports.queryBookingForms = exports.queryPolicies = exports.searchServices = exports.queryServices = exports.bulkDeleteServicesByFilter = exports.bulkDeleteServices = exports.deleteService = exports.bulkUpdateServicesByFilter = exports.bulkUpdateServices = exports.updateService = exports.getService = exports.createService = exports.__metadata = void 0;
exports.WebhookIdentityType = exports.V2RequestedFields = exports.Transparency = exports.Status = exports.SortType = exports.SortOrder = exports.SortDirection = exports.SessionType = exports.ServiceType = exports.ScheduleStatus = exports.ScalarType = exports.ResourceStatus = exports.ResourceNotificationEvent = exports.ResolutionMethod = exports.RequestedFields = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const address_1 = require("@wix/sdk-runtime/transformations/address");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_v2_1 = require("@wix/sdk-runtime/transformations/page-url-v2");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_services_v2_service_services_universal_js_1 = require("./bookings-services-v2-service-services.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function createService(httpClient) {
    return (service) => (0, bookings_services_v2_service_services_universal_js_1.createService)(service, 
    // @ts-ignore
    { httpClient });
}
exports.createService = createService;
function getService(httpClient) {
    return (serviceId) => (0, bookings_services_v2_service_services_universal_js_1.getService)(serviceId, 
    // @ts-ignore
    { httpClient });
}
exports.getService = getService;
function updateService(httpClient) {
    return (_id, service) => (0, bookings_services_v2_service_services_universal_js_1.updateService)(_id, service, 
    // @ts-ignore
    { httpClient });
}
exports.updateService = updateService;
function bulkUpdateServices(httpClient) {
    return (options) => (0, bookings_services_v2_service_services_universal_js_1.bulkUpdateServices)(options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateServices = bulkUpdateServices;
function bulkUpdateServicesByFilter(httpClient) {
    return (filter, options) => (0, bookings_services_v2_service_services_universal_js_1.bulkUpdateServicesByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateServicesByFilter = bulkUpdateServicesByFilter;
function deleteService(httpClient) {
    return (serviceId, options) => (0, bookings_services_v2_service_services_universal_js_1.deleteService)(serviceId, options, 
    // @ts-ignore
    { httpClient });
}
exports.deleteService = deleteService;
function bulkDeleteServices(httpClient) {
    return (ids, options) => (0, bookings_services_v2_service_services_universal_js_1.bulkDeleteServices)(ids, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteServices = bulkDeleteServices;
function bulkDeleteServicesByFilter(httpClient) {
    return (filter, options) => (0, bookings_services_v2_service_services_universal_js_1.bulkDeleteServicesByFilter)(filter, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteServicesByFilter = bulkDeleteServicesByFilter;
function queryServices(httpClient) {
    return (options) => (0, bookings_services_v2_service_services_universal_js_1.queryServices)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryServices = queryServices;
function searchServices(httpClient) {
    return (search) => (0, bookings_services_v2_service_services_universal_js_1.searchServices)(search, 
    // @ts-ignore
    { httpClient });
}
exports.searchServices = searchServices;
function queryPolicies(httpClient) {
    return (query) => (0, bookings_services_v2_service_services_universal_js_1.queryPolicies)(query, 
    // @ts-ignore
    { httpClient });
}
exports.queryPolicies = queryPolicies;
function queryBookingForms(httpClient) {
    return (query, options) => (0, bookings_services_v2_service_services_universal_js_1.queryBookingForms)(query, options, 
    // @ts-ignore
    { httpClient });
}
exports.queryBookingForms = queryBookingForms;
function countServices(httpClient) {
    return (options) => (0, bookings_services_v2_service_services_universal_js_1.countServices)(options, 
    // @ts-ignore
    { httpClient });
}
exports.countServices = countServices;
function queryLocations(httpClient) {
    return (options) => (0, bookings_services_v2_service_services_universal_js_1.queryLocations)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryLocations = queryLocations;
function queryCategories(httpClient) {
    return (options) => (0, bookings_services_v2_service_services_universal_js_1.queryCategories)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryCategories = queryCategories;
function setServiceLocations(httpClient) {
    return (serviceId, locations, options) => (0, bookings_services_v2_service_services_universal_js_1.setServiceLocations)(serviceId, locations, options, 
    // @ts-ignore
    { httpClient });
}
exports.setServiceLocations = setServiceLocations;
function enablePricingPlansForService(httpClient) {
    return (serviceId, pricingPlanIds) => (0, bookings_services_v2_service_services_universal_js_1.enablePricingPlansForService)(serviceId, pricingPlanIds, 
    // @ts-ignore
    { httpClient });
}
exports.enablePricingPlansForService = enablePricingPlansForService;
function disablePricingPlansForService(httpClient) {
    return (serviceId, options) => (0, bookings_services_v2_service_services_universal_js_1.disablePricingPlansForService)(serviceId, options, 
    // @ts-ignore
    { httpClient });
}
exports.disablePricingPlansForService = disablePricingPlansForService;
function setCustomSlug(httpClient) {
    return (serviceId, options) => (0, bookings_services_v2_service_services_universal_js_1.setCustomSlug)(serviceId, options, 
    // @ts-ignore
    { httpClient });
}
exports.setCustomSlug = setCustomSlug;
function validateSlug(httpClient) {
    return (serviceId, options) => (0, bookings_services_v2_service_services_universal_js_1.validateSlug)(serviceId, options, 
    // @ts-ignore
    { httpClient });
}
exports.validateSlug = validateSlug;
function cloneService(httpClient) {
    return (sourceServiceId) => (0, bookings_services_v2_service_services_universal_js_1.cloneService)(sourceServiceId, 
    // @ts-ignore
    { httpClient });
}
exports.cloneService = cloneService;
exports.onServiceCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.services.v2.service_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.bookingPolicy.createdDate' },
            { path: 'entity.bookingPolicy.updatedDate' },
            { path: 'entity.schedule.firstSessionStart' },
            { path: 'entity.schedule.lastSessionEnd' },
            { path: 'entity.supportedSlugs.createdDate' },
            { path: 'entity.mainSlug.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.media.items.image' },
            { path: 'entity.media.mainMedia.image' },
            { path: 'entity.media.coverMedia.image' },
            { path: 'entity.staffMembers.mainMedia.image' },
            { path: 'entity.staffMemberDetails.staffMembers.mainMedia.image' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.locations.calculatedAddress' },
            { path: 'entity.locations.business.address' },
            { path: 'entity.locations.custom.address' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [
            { path: 'entity.urls.servicePage' },
            { path: 'entity.urls.bookingPage' },
            { path: 'entity.urls.calendarPage' },
        ],
    },
])))();
exports.onServiceDeleted = (0, sdk_types_1.EventDefinition)('wix.bookings.services.v2.service_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.bookingPolicy.createdDate' },
            { path: 'entity.bookingPolicy.updatedDate' },
            { path: 'entity.schedule.firstSessionStart' },
            { path: 'entity.schedule.lastSessionEnd' },
            { path: 'entity.supportedSlugs.createdDate' },
            { path: 'entity.mainSlug.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.media.items.image' },
            { path: 'entity.media.mainMedia.image' },
            { path: 'entity.media.coverMedia.image' },
            { path: 'entity.staffMembers.mainMedia.image' },
            { path: 'entity.staffMemberDetails.staffMembers.mainMedia.image' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.locations.calculatedAddress' },
            { path: 'entity.locations.business.address' },
            { path: 'entity.locations.custom.address' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [
            { path: 'entity.urls.servicePage' },
            { path: 'entity.urls.bookingPage' },
            { path: 'entity.urls.calendarPage' },
        ],
    },
])))();
exports.onServiceUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.services.v2.service_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.bookingPolicy.createdDate' },
            { path: 'entity.bookingPolicy.updatedDate' },
            { path: 'entity.schedule.firstSessionStart' },
            { path: 'entity.schedule.lastSessionEnd' },
            { path: 'entity.supportedSlugs.createdDate' },
            { path: 'entity.mainSlug.createdDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.media.items.image' },
            { path: 'entity.media.mainMedia.image' },
            { path: 'entity.media.coverMedia.image' },
            { path: 'entity.staffMembers.mainMedia.image' },
            { path: 'entity.staffMemberDetails.staffMembers.mainMedia.image' },
        ],
    },
    {
        transformFn: address_1.transformRESTAddressToSDKAddress,
        paths: [
            { path: 'entity.locations.calculatedAddress' },
            { path: 'entity.locations.business.address' },
            { path: 'entity.locations.custom.address' },
        ],
    },
    {
        transformFn: page_url_v2_1.transformRESTPageURLV2ToSDKPageURLV2,
        paths: [
            { path: 'entity.urls.servicePage' },
            { path: 'entity.urls.bookingPage' },
            { path: 'entity.urls.calendarPage' },
        ],
    },
])))();
var bookings_services_v2_service_services_universal_js_2 = require("./bookings-services-v2-service-services.universal.js");
Object.defineProperty(exports, "Action", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.Action; } });
Object.defineProperty(exports, "AggregationType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.AggregationType; } });
Object.defineProperty(exports, "ApprovalStatus", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ApprovalStatus; } });
Object.defineProperty(exports, "BenefitType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.BenefitType; } });
Object.defineProperty(exports, "CalendarType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.CalendarType; } });
Object.defineProperty(exports, "CategoryNotificationEvent", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.CategoryNotificationEvent; } });
Object.defineProperty(exports, "CategoryStatus", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.CategoryStatus; } });
Object.defineProperty(exports, "CloneErrors", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.CloneErrors; } });
Object.defineProperty(exports, "ConferenceType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ConferenceType; } });
Object.defineProperty(exports, "CrudType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.CrudType; } });
Object.defineProperty(exports, "DateHistogramAggregationInterval", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.DateHistogramAggregationInterval; } });
Object.defineProperty(exports, "Day", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.Day; } });
Object.defineProperty(exports, "DayOfWeek", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.DayOfWeek; } });
Object.defineProperty(exports, "Event", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.Event; } });
Object.defineProperty(exports, "InvalidSlugError", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.InvalidSlugError; } });
Object.defineProperty(exports, "LocationStatus", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.LocationStatus; } });
Object.defineProperty(exports, "LocationType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.LocationType; } });
Object.defineProperty(exports, "LocationTypeEnumLocationType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.LocationTypeEnumLocationType; } });
Object.defineProperty(exports, "LocationsLocationType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.LocationsLocationType; } });
Object.defineProperty(exports, "MissingValues", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.MissingValues; } });
Object.defineProperty(exports, "Mode", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.Mode; } });
Object.defineProperty(exports, "NestedAggregationType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.NestedAggregationType; } });
Object.defineProperty(exports, "PlacementType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.PlacementType; } });
Object.defineProperty(exports, "RateType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.RateType; } });
Object.defineProperty(exports, "RecurringIntervalType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.RecurringIntervalType; } });
Object.defineProperty(exports, "RequestedFields", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.RequestedFields; } });
Object.defineProperty(exports, "ResolutionMethod", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ResolutionMethod; } });
Object.defineProperty(exports, "ResourceNotificationEvent", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ResourceNotificationEvent; } });
Object.defineProperty(exports, "ResourceStatus", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ResourceStatus; } });
Object.defineProperty(exports, "ScalarType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ScalarType; } });
Object.defineProperty(exports, "ScheduleStatus", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ScheduleStatus; } });
Object.defineProperty(exports, "ServiceType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.ServiceType; } });
Object.defineProperty(exports, "SessionType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.SessionType; } });
Object.defineProperty(exports, "SortDirection", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.SortDirection; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "SortType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.SortType; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.Status; } });
Object.defineProperty(exports, "Transparency", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.Transparency; } });
Object.defineProperty(exports, "V2RequestedFields", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.V2RequestedFields; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_services_v2_service_services_universal_js_2.WebhookIdentityType; } });
