// @wix/auto_sdk_bookings_external-calendars/build/internal/es/src/bookings-calendar-v2-external-calendar-external-calendars.meta.js
import * as ambassadorWixBookingsCalendarV2ExternalCalendar from './bookings-calendar-v2-external-calendar-external-calendars.http.js';
export function listProviders() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listProviders(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/providers',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getConnection() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.getConnection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/connections/{connectionId}',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listConnections() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listConnections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/connections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function connectByOAuth() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.connectByOAuth(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/external-calendars/connections:connectByOAuth',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function connectByCredentials() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.connectByCredentials(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/external-calendars/connections:connectByCredentials',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listCalendars() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listCalendars(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/connections/{connectionId}/calendars',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateSyncConfig() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.updateSyncConfig(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/external-calendars/connections/{connectionId}/sync-config',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function disconnect() {
    const payload = { connectionId: ':connectionId' };
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.disconnect(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/external-calendars/connections/{connectionId}/disconnect',
        pathParams: { connectionId: 'connectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listEvents() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCalendarV2ExternalCalendar.listEvents(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/external-calendars/events',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
