// @wix/auto_sdk_bookings_app-settings/build/es/src/bookings-v2-bookings-app-settings-app-settings.meta.js
import * as ambassadorWixBookingsV2BookingsAppSettings from './bookings-v2-bookings-app-settings-app-settings.http.js';
export function createBookingsAppSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2BookingsAppSettings.createBookingsAppSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings-app-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getBookingsAppSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2BookingsAppSettings.getBookingsAppSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateBookingsAppSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2BookingsAppSettings.updateBookingsAppSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/bookings-app-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
