// @wix/auto_sdk_ecom_tax-groups/build/cjs/src/billing-v1-tax-group-tax-groups.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listDefaultTaxGroupsByAppIds = exports.listDefaultTaxGroups = exports.queryTaxGroups = exports.deleteTaxGroup = exports.updateTaxGroup = exports.getTaxGroup = exports.createTaxGroup = void 0;
const ambassadorWixBillingV1TaxGroup = __importStar(require("./billing-v1-tax-group-tax-groups.http.js"));
function createTaxGroup() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.createTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-groups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createTaxGroup = createTaxGroup;
function getTaxGroup() {
    const payload = { taxGroupId: ':taxGroupId' };
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.getTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tax-groups/{taxGroupId}',
        pathParams: { taxGroupId: 'taxGroupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTaxGroup = getTaxGroup;
function updateTaxGroup() {
    const payload = { taxGroup: { id: ':taxGroupId' } };
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.updateTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tax-groups/{taxGroup.id}',
        pathParams: { taxGroupId: 'taxGroupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateTaxGroup = updateTaxGroup;
function deleteTaxGroup() {
    const payload = { taxGroupId: ':taxGroupId' };
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.deleteTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tax-groups/{taxGroupId}',
        pathParams: { taxGroupId: 'taxGroupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteTaxGroup = deleteTaxGroup;
function queryTaxGroups() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.queryTaxGroups(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-groups/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryTaxGroups = queryTaxGroups;
function listDefaultTaxGroups() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.listDefaultTaxGroups(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tax-groups/default-tax-groups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDefaultTaxGroups = listDefaultTaxGroups;
function listDefaultTaxGroupsByAppIds() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.listDefaultTaxGroupsByAppIds(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-groups/list-default-tax-groups-by-app-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDefaultTaxGroupsByAppIds = listDefaultTaxGroupsByAppIds;
