// @wix/auto_sdk_ecom_tax-regions/build/internal/es/src/billing-v1-tax-region-tax-regions.meta.js
import * as ambassadorWixBillingV1TaxRegion from './billing-v1-tax-region-tax-regions.http.js';
export function createTaxRegion() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.createTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-regions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTaxRegion() {
    const payload = { taxRegionId: ':taxRegionId' };
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.getTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tax-regions/{taxRegionId}',
        pathParams: { taxRegionId: 'taxRegionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateTaxRegion() {
    const payload = { taxRegion: { id: ':taxRegionId' } };
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.updateTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tax-regions/{taxRegion.id}',
        pathParams: { taxRegionId: 'taxRegionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteTaxRegion() {
    const payload = { taxRegionId: ':taxRegionId' };
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.deleteTaxRegion(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tax-regions/{taxRegionId}',
        pathParams: { taxRegionId: 'taxRegionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryTaxRegions() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxRegion.queryTaxRegions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-regions/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
