// @wix/auto_sdk_bookings_extended-bookings/build/es/src/bookings-reader-v2-extended-booking-extended-bookings.public.js
import { countExtendedBookings as universalCountExtendedBookings, query as universalQuery, queryExtendedBookings as universalQueryExtendedBookings, } from './bookings-reader-v2-extended-booking-extended-bookings.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/bookings' };
export function query(httpClient) {
    return (query, options) => universalQuery(query, options, 
    // @ts-ignore
    { httpClient });
}
export function queryExtendedBookings(httpClient) {
    return (query, options) => universalQueryExtendedBookings(query, options, 
    // @ts-ignore
    { httpClient });
}
export function countExtendedBookings(httpClient) {
    return (options) => universalCountExtendedBookings(options, 
    // @ts-ignore
    { httpClient });
}
export { Actor, AttendanceStatus, BookingFeeStatus, BookingStatus, IdentityType, LocationType, MultiServiceBookingType, PaymentStatus, Platform, SelectedPaymentOption, SortOrder, ValueType, } from './bookings-reader-v2-extended-booking-extended-bookings.universal.js';
