// @wix/auto_sdk_ecom_tax-groups/build/internal/cjs/src/billing-v1-tax-group-tax-groups.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.onTaxGroupUpdated = exports.onTaxGroupDeleted = exports.onTaxGroupCreated = exports.listDefaultTaxGroupsByAppIds = exports.listDefaultTaxGroups = exports.queryTaxGroups = exports.deleteTaxGroup = exports.updateTaxGroup = exports.getTaxGroup = exports.createTaxGroup = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const billing_v1_tax_group_tax_groups_universal_js_1 = require("./billing-v1-tax-group-tax-groups.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/ecom' };
function createTaxGroup(httpClient) {
    return (taxGroup) => (0, billing_v1_tax_group_tax_groups_universal_js_1.createTaxGroup)(taxGroup, 
    // @ts-ignore
    { httpClient });
}
exports.createTaxGroup = createTaxGroup;
function getTaxGroup(httpClient) {
    return (taxGroupId) => (0, billing_v1_tax_group_tax_groups_universal_js_1.getTaxGroup)(taxGroupId, 
    // @ts-ignore
    { httpClient });
}
exports.getTaxGroup = getTaxGroup;
function updateTaxGroup(httpClient) {
    return (_id, taxGroup) => (0, billing_v1_tax_group_tax_groups_universal_js_1.updateTaxGroup)(_id, taxGroup, 
    // @ts-ignore
    { httpClient });
}
exports.updateTaxGroup = updateTaxGroup;
function deleteTaxGroup(httpClient) {
    return (taxGroupId) => (0, billing_v1_tax_group_tax_groups_universal_js_1.deleteTaxGroup)(taxGroupId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteTaxGroup = deleteTaxGroup;
function queryTaxGroups(httpClient) {
    return () => (0, billing_v1_tax_group_tax_groups_universal_js_1.queryTaxGroups)(
    // @ts-ignore
    { httpClient });
}
exports.queryTaxGroups = queryTaxGroups;
function listDefaultTaxGroups(httpClient) {
    return () => (0, billing_v1_tax_group_tax_groups_universal_js_1.listDefaultTaxGroups)(
    // @ts-ignore
    { httpClient });
}
exports.listDefaultTaxGroups = listDefaultTaxGroups;
function listDefaultTaxGroupsByAppIds(httpClient) {
    return (appIds) => (0, billing_v1_tax_group_tax_groups_universal_js_1.listDefaultTaxGroupsByAppIds)(appIds, 
    // @ts-ignore
    { httpClient });
}
exports.listDefaultTaxGroupsByAppIds = listDefaultTaxGroupsByAppIds;
exports.onTaxGroupCreated = (0, sdk_types_1.EventDefinition)('wix.billing.v1.tax_group_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onTaxGroupDeleted = (0, sdk_types_1.EventDefinition)('wix.billing.v1.tax_group_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onTaxGroupUpdated = (0, sdk_types_1.EventDefinition)('wix.billing.v1.tax_group_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var billing_v1_tax_group_tax_groups_universal_js_2 = require("./billing-v1-tax-group-tax-groups.universal.js");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return billing_v1_tax_group_tax_groups_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return billing_v1_tax_group_tax_groups_universal_js_2.WebhookIdentityType; } });
