// @wix/auto_sdk_ecom_tax-groups/build/internal/cjs/src/billing-v1-tax-group-tax-groups.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.onTaxGroupUpdated = exports.onTaxGroupDeleted = exports.onTaxGroupCreated = exports.listDefaultTaxGroupsByAppIds = exports.listDefaultTaxGroups = exports.queryTaxGroups = exports.deleteTaxGroup = exports.updateTaxGroup = exports.getTaxGroup = exports.createTaxGroup = void 0;
const billing_v1_tax_group_tax_groups_public_js_1 = require("./billing-v1-tax-group-tax-groups.public.js");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
const billing_v1_tax_group_tax_groups_public_js_2 = require("./billing-v1-tax-group-tax-groups.public.js");
const billing_v1_tax_group_tax_groups_public_js_3 = require("./billing-v1-tax-group-tax-groups.public.js");
const billing_v1_tax_group_tax_groups_public_js_4 = require("./billing-v1-tax-group-tax-groups.public.js");
exports.createTaxGroup = (0, rest_modules_1.createRESTModule)(billing_v1_tax_group_tax_groups_public_js_1.createTaxGroup);
exports.getTaxGroup = (0, rest_modules_1.createRESTModule)(billing_v1_tax_group_tax_groups_public_js_1.getTaxGroup);
exports.updateTaxGroup = (0, rest_modules_1.createRESTModule)(billing_v1_tax_group_tax_groups_public_js_1.updateTaxGroup);
exports.deleteTaxGroup = (0, rest_modules_1.createRESTModule)(billing_v1_tax_group_tax_groups_public_js_1.deleteTaxGroup);
exports.queryTaxGroups = (0, rest_modules_1.createRESTModule)(billing_v1_tax_group_tax_groups_public_js_1.queryTaxGroups);
exports.listDefaultTaxGroups = (0, rest_modules_1.createRESTModule)(billing_v1_tax_group_tax_groups_public_js_1.listDefaultTaxGroups);
exports.listDefaultTaxGroupsByAppIds = (0, rest_modules_1.createRESTModule)(billing_v1_tax_group_tax_groups_public_js_1.listDefaultTaxGroupsByAppIds);
/**
 * Triggered when a tax group is created.
 */
exports.onTaxGroupCreated = (0, event_definition_modules_1.createEventModule)(billing_v1_tax_group_tax_groups_public_js_2.onTaxGroupCreated);
/**
 * Triggered when a tax group is deleted.
 */
exports.onTaxGroupDeleted = (0, event_definition_modules_1.createEventModule)(billing_v1_tax_group_tax_groups_public_js_3.onTaxGroupDeleted);
/**
 * Triggered when a tax group is updated.
 */
exports.onTaxGroupUpdated = (0, event_definition_modules_1.createEventModule)(billing_v1_tax_group_tax_groups_public_js_4.onTaxGroupUpdated);
var billing_v1_tax_group_tax_groups_universal_js_1 = require("./billing-v1-tax-group-tax-groups.universal.js");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return billing_v1_tax_group_tax_groups_universal_js_1.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return billing_v1_tax_group_tax_groups_universal_js_1.WebhookIdentityType; } });
