// @wix/auto_sdk_bookings_availability-calendar/build/cjs/src/bookings-availability-v1-slot-availability-availability-calendar.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortOrder = exports.LocationType = exports.LocationLocationType = exports.calculateMultiSlotAvailability = exports.getScheduleAvailability = exports.queryAvailability = exports.__metadata = void 0;
const bookings_availability_v1_slot_availability_availability_calendar_universal_js_1 = require("./bookings-availability-v1-slot-availability-availability-calendar.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function queryAvailability(httpClient) {
    return (query, options) => (0, bookings_availability_v1_slot_availability_availability_calendar_universal_js_1.queryAvailability)(query, options, 
    // @ts-ignore
    { httpClient });
}
exports.queryAvailability = queryAvailability;
function getScheduleAvailability(httpClient) {
    return (scheduleId) => (0, bookings_availability_v1_slot_availability_availability_calendar_universal_js_1.getScheduleAvailability)(scheduleId, 
    // @ts-ignore
    { httpClient });
}
exports.getScheduleAvailability = getScheduleAvailability;
function calculateMultiSlotAvailability(httpClient) {
    return (options) => (0, bookings_availability_v1_slot_availability_availability_calendar_universal_js_1.calculateMultiSlotAvailability)(options, 
    // @ts-ignore
    { httpClient });
}
exports.calculateMultiSlotAvailability = calculateMultiSlotAvailability;
var bookings_availability_v1_slot_availability_availability_calendar_universal_js_2 = require("./bookings-availability-v1-slot-availability-availability-calendar.universal.js");
Object.defineProperty(exports, "LocationLocationType", { enumerable: true, get: function () { return bookings_availability_v1_slot_availability_availability_calendar_universal_js_2.LocationLocationType; } });
Object.defineProperty(exports, "LocationType", { enumerable: true, get: function () { return bookings_availability_v1_slot_availability_availability_calendar_universal_js_2.LocationType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_availability_v1_slot_availability_availability_calendar_universal_js_2.SortOrder; } });
