// @wix/members/build/internal/meta.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  authentication: () => authentication,
  authorization: () => authorization,
  badges: () => badges,
  customFieldApplications: () => customFieldApplications,
  customFieldSuggestions: () => customFieldSuggestions,
  customFields: () => customFields,
  defaultPrivacy: () => defaultPrivacy,
  memberFollowers: () => memberFollowers,
  memberPrivacySettings: () => memberPrivacySettings,
  memberReport: () => memberReport,
  memberRoleDefinition: () => memberRoleDefinition,
  memberToMemberBlock: () => memberToMemberBlock,
  members: () => members,
  membersAbout: () => membersAbout,
  userMember: () => userMember
});
module.exports = __toCommonJS(meta_exports);
var badges = __toESM(require("@wix/auto_sdk_members_badges/meta"));
var authentication = __toESM(require("@wix/auto_sdk_members_authentication/meta"));
var membersAbout = __toESM(require("@wix/auto_sdk_members_members-about/meta"));
var userMember = __toESM(require("@wix/auto_sdk_members_user-member/meta"));
var customFields = __toESM(require("@wix/auto_sdk_members_custom-fields/meta"));
var customFieldApplications = __toESM(require("@wix/auto_sdk_members_custom-field-applications/meta"));
var customFieldSuggestions = __toESM(require("@wix/auto_sdk_members_custom-field-suggestions/meta"));
var defaultPrivacy = __toESM(require("@wix/auto_sdk_members_default-privacy/meta"));
var members = __toESM(require("@wix/auto_sdk_members_members/meta"));
var memberPrivacySettings = __toESM(require("@wix/auto_sdk_members_member-privacy-settings/meta"));
var memberReport = __toESM(require("@wix/auto_sdk_members_member-report/meta"));
var memberRoleDefinition = __toESM(require("@wix/auto_sdk_members_member-role-definition/meta"));
var memberToMemberBlock = __toESM(require("@wix/auto_sdk_members_member-to-member-block/meta"));
var authorization = __toESM(require("@wix/auto_sdk_members_authorization/meta"));
var memberFollowers = __toESM(require("@wix/auto_sdk_members_member-followers/meta"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  authentication,
  authorization,
  badges,
  customFieldApplications,
  customFieldSuggestions,
  customFields,
  defaultPrivacy,
  memberFollowers,
  memberPrivacySettings,
  memberReport,
  memberRoleDefinition,
  memberToMemberBlock,
  members,
  membersAbout,
  userMember
});
