// @wix/members/build/internal/meta.mjs
// meta.ts
import * as badges from "@wix/auto_sdk_members_badges/meta";
import * as authentication from "@wix/auto_sdk_members_authentication/meta";
import * as membersAbout from "@wix/auto_sdk_members_members-about/meta";
import * as userMember from "@wix/auto_sdk_members_user-member/meta";
import * as customFields from "@wix/auto_sdk_members_custom-fields/meta";
import * as customFieldApplications from "@wix/auto_sdk_members_custom-field-applications/meta";
import * as customFieldSuggestions from "@wix/auto_sdk_members_custom-field-suggestions/meta";
import * as defaultPrivacy from "@wix/auto_sdk_members_default-privacy/meta";
import * as members from "@wix/auto_sdk_members_members/meta";
import * as memberPrivacySettings from "@wix/auto_sdk_members_member-privacy-settings/meta";
import * as memberReport from "@wix/auto_sdk_members_member-report/meta";
import * as memberRoleDefinition from "@wix/auto_sdk_members_member-role-definition/meta";
import * as memberToMemberBlock from "@wix/auto_sdk_members_member-to-member-block/meta";
import * as authorization from "@wix/auto_sdk_members_authorization/meta";
import * as memberFollowers from "@wix/auto_sdk_members_member-followers/meta";
export {
  authentication,
  authorization,
  badges,
  customFieldApplications,
  customFieldSuggestions,
  customFields,
  defaultPrivacy,
  memberFollowers,
  memberPrivacySettings,
  memberReport,
  memberRoleDefinition,
  memberToMemberBlock,
  members,
  membersAbout,
  userMember
};
