// @wix/auto_sdk_bookings_service-categories/build/cjs/src/bookings-categories-v2-category-service-categories.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.Position = exports.onCategoryUpdated = exports.onCategoryDeleted = exports.onCategoryCreated = exports.moveCategory = exports.countCategories = exports.queryCategories = exports.deleteCategory = exports.updateCategory = exports.getCategory = exports.createCategory = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_categories_v2_category_service_categories_universal_js_1 = require("./bookings-categories-v2-category-service-categories.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function createCategory(httpClient) {
    return (category) => (0, bookings_categories_v2_category_service_categories_universal_js_1.createCategory)(category, 
    // @ts-ignore
    { httpClient });
}
exports.createCategory = createCategory;
function getCategory(httpClient) {
    return (categoryId) => (0, bookings_categories_v2_category_service_categories_universal_js_1.getCategory)(categoryId, 
    // @ts-ignore
    { httpClient });
}
exports.getCategory = getCategory;
function updateCategory(httpClient) {
    return (_id, category) => (0, bookings_categories_v2_category_service_categories_universal_js_1.updateCategory)(_id, category, 
    // @ts-ignore
    { httpClient });
}
exports.updateCategory = updateCategory;
function deleteCategory(httpClient) {
    return (categoryId) => (0, bookings_categories_v2_category_service_categories_universal_js_1.deleteCategory)(categoryId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCategory = deleteCategory;
function queryCategories(httpClient) {
    return () => (0, bookings_categories_v2_category_service_categories_universal_js_1.queryCategories)(
    // @ts-ignore
    { httpClient });
}
exports.queryCategories = queryCategories;
function countCategories(httpClient) {
    return (options) => (0, bookings_categories_v2_category_service_categories_universal_js_1.countCategories)(options, 
    // @ts-ignore
    { httpClient });
}
exports.countCategories = countCategories;
function moveCategory(httpClient) {
    return (categoryId, options) => (0, bookings_categories_v2_category_service_categories_universal_js_1.moveCategory)(categoryId, options, 
    // @ts-ignore
    { httpClient });
}
exports.moveCategory = moveCategory;
exports.onCategoryCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.categories.v2.category_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onCategoryDeleted = (0, sdk_types_1.EventDefinition)('wix.bookings.categories.v2.category_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onCategoryUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.categories.v2.category_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var bookings_categories_v2_category_service_categories_universal_js_2 = require("./bookings-categories-v2-category-service-categories.universal.js");
Object.defineProperty(exports, "Position", { enumerable: true, get: function () { return bookings_categories_v2_category_service_categories_universal_js_2.Position; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_categories_v2_category_service_categories_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_categories_v2_category_service_categories_universal_js_2.WebhookIdentityType; } });
