// @wix/auto_sdk_bookings_service-categories/build/cjs/src/bookings-categories-v2-category-service-categories.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.moveCategory = exports.countCategories = exports.queryCategories = exports.deleteCategory = exports.updateCategory = exports.getCategory = exports.createCategory = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBookingsCategoriesV2CategoriesServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/bookings/v2/categories',
                destPath: '/v2/categories',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/categories',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/bookings/v2/categories',
                destPath: '/v2/categories',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/bookings/v2/categories',
                destPath: '/v2/categories',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/bookings/v2/categories',
                destPath: '/v2/categories',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/bookings/v2/categories',
                destPath: '/v2/categories',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/bookings/v2/categories',
                destPath: '/v2/categories',
            },
        ],
        _: [
            {
                srcPath: '/_api/bookings/v2/categories',
                destPath: '/v2/categories',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_bookings_service-categories';
/**
 * Creates a Category.
 *
 *
 * The request body must include category name.
 */
function createCategory(payload) {
    function __createCategory({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'category.createdDate' },
                    { path: 'category.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.bookings.categories.v2.category',
            method: 'POST',
            methodFqn: 'wix.bookings.categories.v2.CategoriesService.CreateCategory',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                protoPath: '/v2/categories',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'category.createdDate' },
                        { path: 'category.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createCategory;
}
exports.createCategory = createCategory;
/** Retrieves a Category. */
function getCategory(payload) {
    function __getCategory({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.categories.v2.category',
            method: 'GET',
            methodFqn: 'wix.bookings.categories.v2.CategoriesService.GetCategory',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                protoPath: '/v2/categories/{categoryId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'category.createdDate' },
                        { path: 'category.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getCategory;
}
exports.getCategory = getCategory;
/**
 * Updates a Category.
 *
 *
 * Each time the Category is updated,
 * `revision` increments by 1.
 * The current `revision` must be passed when updating the Category.
 * This ensures you're working with the latest Category
 * and prevents unintended overwrites.
 */
function updateCategory(payload) {
    function __updateCategory({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'category.createdDate' },
                    { path: 'category.updatedDate' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.bookings.categories.v2.category',
            method: 'PATCH',
            methodFqn: 'wix.bookings.categories.v2.CategoriesService.UpdateCategory',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                protoPath: '/v2/categories/{category.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'category.createdDate' },
                        { path: 'category.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateCategory;
}
exports.updateCategory = updateCategory;
/**
 * Deletes a Category.
 *
 *
 * Deleting a Category permanently removes them from the Category List.
 */
function deleteCategory(payload) {
    function __deleteCategory({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.categories.v2.category',
            method: 'DELETE',
            methodFqn: 'wix.bookings.categories.v2.CategoriesService.DeleteCategory',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                protoPath: '/v2/categories/{categoryId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteCategory;
}
exports.deleteCategory = deleteCategory;
/**
 * Retrieves a list of Categories, given the provided [paging, filtering, and sorting][1].
 *
 * Up to 1,000 Categories can be returned per request.
 *
 * To learn how to query Categories, see [API Query Language][2].
 *
 * [1]: https://dev.wix.com/api/rest/getting-started/sorting-and-paging
 * [2]: https://dev.wix.com/api/rest/getting-started/api-query-language
 */
function queryCategories(payload) {
    function __queryCategories({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.categories.v2.category',
            method: 'POST',
            methodFqn: 'wix.bookings.categories.v2.CategoriesService.QueryCategories',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                protoPath: '/v2/categories/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'categories.createdDate' },
                        { path: 'categories.updatedDate' },
                    ],
                },
            ]),
            fallback: [
                {
                    method: 'POST',
                    url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                        protoPath: '/v2/categories/query',
                        data: payload,
                        host,
                    }),
                    data: payload,
                },
            ],
        };
        return metadata;
    }
    return __queryCategories;
}
exports.queryCategories = queryCategories;
/**
 * Counts categories according to given criteria.
 * Use [WQL
 * filter](https://dev.wix.com/docs/rest/articles/getting-started/api-query-language#the-filter-section)
 * to define the criteria.
 */
function countCategories(payload) {
    function __countCategories({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.categories.v2.category',
            method: 'POST',
            methodFqn: 'wix.bookings.categories.v2.CategoriesService.CountCategories',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                protoPath: '/v2/categories/count',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __countCategories;
}
exports.countCategories = countCategories;
/** Moves a category to a different position in the list */
function moveCategory(payload) {
    function __moveCategory({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.categories.v2.category',
            method: 'POST',
            methodFqn: 'wix.bookings.categories.v2.CategoriesService.MoveCategory',
            packageName: PACKAGE_NAME,
            url: resolveWixBookingsCategoriesV2CategoriesServiceUrl({
                protoPath: '/v2/categories/set-position/{categoryId}',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'category.createdDate' },
                        { path: 'category.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __moveCategory;
}
exports.moveCategory = moveCategory;
