// @wix/auto_sdk_bookings_service-categories/build/cjs/src/bookings-categories-v2-category-service-categories.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.moveCategory = exports.countCategories = exports.queryCategories = exports.deleteCategory = exports.updateCategory = exports.getCategory = exports.createCategory = void 0;
const ambassadorWixBookingsCategoriesV2Category = __importStar(require("./bookings-categories-v2-category-service-categories.http.js"));
function createCategory() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.createCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/categories',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createCategory = createCategory;
function getCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.getCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/categories/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getCategory = getCategory;
function updateCategory() {
    const payload = { category: { id: ':categoryId' } };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.updateCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/categories/{category.id}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateCategory = updateCategory;
function deleteCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.deleteCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/categories/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteCategory = deleteCategory;
function queryCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.queryCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/categories/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryCategories = queryCategories;
function countCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.countCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/categories/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.countCategories = countCategories;
function moveCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.moveCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/categories/set-position/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.moveCategory = moveCategory;
