// @wix/auto_sdk_bookings_booking-policies/build/internal/es/src/bookings-v1-booking-policy-booking-policies.context.js
import { createBookingPolicy as publicCreateBookingPolicy, getBookingPolicy as publicGetBookingPolicy, getStrictestBookingPolicy as publicGetStrictestBookingPolicy, updateBookingPolicy as publicUpdateBookingPolicy, setDefaultBookingPolicy as publicSetDefaultBookingPolicy, deleteBookingPolicy as publicDeleteBookingPolicy, queryBookingPolicies as publicQueryBookingPolicies, countBookingPolicies as publicCountBookingPolicies, } from './bookings-v1-booking-policy-booking-policies.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onBookingPolicyCreated as publicOnBookingPolicyCreated } from './bookings-v1-booking-policy-booking-policies.public.js';
import { onBookingPolicyDefaultBookingPolicySet as publicOnBookingPolicyDefaultBookingPolicySet } from './bookings-v1-booking-policy-booking-policies.public.js';
import { onBookingPolicyDeleted as publicOnBookingPolicyDeleted } from './bookings-v1-booking-policy-booking-policies.public.js';
import { onBookingPolicyUpdated as publicOnBookingPolicyUpdated } from './bookings-v1-booking-policy-booking-policies.public.js';
export const createBookingPolicy = /*#__PURE__*/ createRESTModule(publicCreateBookingPolicy);
export const getBookingPolicy = /*#__PURE__*/ createRESTModule(publicGetBookingPolicy);
export const getStrictestBookingPolicy = /*#__PURE__*/ createRESTModule(publicGetStrictestBookingPolicy);
export const updateBookingPolicy = /*#__PURE__*/ createRESTModule(publicUpdateBookingPolicy);
export const setDefaultBookingPolicy = /*#__PURE__*/ createRESTModule(publicSetDefaultBookingPolicy);
export const deleteBookingPolicy = /*#__PURE__*/ createRESTModule(publicDeleteBookingPolicy);
export const queryBookingPolicies = /*#__PURE__*/ createRESTModule(publicQueryBookingPolicies);
export const countBookingPolicies = /*#__PURE__*/ createRESTModule(publicCountBookingPolicies);
/**
 * Triggered when a booking policy is created.
 */
export const onBookingPolicyCreated = createEventModule(publicOnBookingPolicyCreated);
/**
 * Triggered when the site's default policy changes. Then,
 * _Booking Policy Updated_
 * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/bookings/booking-policies/on-booking-policy-updated) | [REST](https://dev.wix.com/docs/rest/business-solutions/bookings/policies/booking-policies/booking-policy-updated))
 * is also triggered both for the new and the previous default policy.
 */
export const onBookingPolicyDefaultBookingPolicySet = createEventModule(publicOnBookingPolicyDefaultBookingPolicySet);
/**
 * Triggered when a booking policy is deleted.
 */
export const onBookingPolicyDeleted = createEventModule(publicOnBookingPolicyDeleted);
/**
 * Triggered when a booking policy is updated, including when a policy's
 * `default` attribute changes.
 */
export const onBookingPolicyUpdated = createEventModule(publicOnBookingPolicyUpdated);
export { SortOrder, PlacementType, DayOfWeek, ResolutionMethod, State, SiteCreatedContext, Namespace, DeleteStatus, WebhookIdentityType, } from './bookings-v1-booking-policy-booking-policies.universal.js';
