// @wix/auto_sdk_members_custom-field-suggestions/build/internal/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  listCustomFieldSuggestions: () => listCustomFieldSuggestions2,
  queryCustomFieldSuggestions: () => queryCustomFieldSuggestions2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-custom-field-suggestion-custom-field-suggestions.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members-custom-fields-suggestions-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/custom-field-suggestions",
        destPath: "/v1/custom-field-suggestions"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-field-suggestions";
function listCustomFieldSuggestions(payload) {
  function __listCustomFieldSuggestions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_suggestion",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.suggestions.CustomFieldSuggestions.ListCustomFieldSuggestions",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(
        { protoPath: "/v1/custom-field-suggestions", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true)
    };
    return metadata;
  }
  return __listCustomFieldSuggestions;
}
function queryCustomFieldSuggestions(payload) {
  function __queryCustomFieldSuggestions({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_suggestion",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.suggestions.CustomFieldSuggestions.QueryCustomFieldSuggestions",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsSuggestionsCustomFieldSuggestionsUrl(
        { protoPath: "/v1/custom-field-suggestions/query", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __queryCustomFieldSuggestions;
}

// src/members-v1-custom-field-suggestion-custom-field-suggestions.meta.ts
function listCustomFieldSuggestions2() {
  const payload = {};
  const getRequestOptions = listCustomFieldSuggestions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/custom-field-suggestions",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCustomFieldSuggestions2() {
  const payload = {};
  const getRequestOptions = queryCustomFieldSuggestions(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-field-suggestions/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  listCustomFieldSuggestions,
  queryCustomFieldSuggestions
});
