// @wix/auto_sdk_bookings_staff-members/build/internal/es/src/bookings-staff-v1-staff-member-staff-members.context.js
import { createStaffMember as publicCreateStaffMember, getStaffMember as publicGetStaffMember, updateStaffMember as publicUpdateStaffMember, deleteStaffMember as publicDeleteStaffMember, queryStaffMembers as publicQueryStaffMembers, countStaffMembers as publicCountStaffMembers, connectStaffMemberToUser as publicConnectStaffMemberToUser, searchStaffMembers as publicSearchStaffMembers, disconnectStaffMemberFromUser as publicDisconnectStaffMemberFromUser, assignWorkingHoursSchedule as publicAssignWorkingHoursSchedule, assignCustomSchedule as publicAssignCustomSchedule, bulkUpdateStaffMemberTags as publicBulkUpdateStaffMemberTags, bulkUpdateStaffMemberTagsByFilter as publicBulkUpdateStaffMemberTagsByFilter, } from './bookings-staff-v1-staff-member-staff-members.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onStaffMemberConnectedToUser as publicOnStaffMemberConnectedToUser } from './bookings-staff-v1-staff-member-staff-members.public.js';
import { onStaffMemberCreated as publicOnStaffMemberCreated } from './bookings-staff-v1-staff-member-staff-members.public.js';
import { onStaffMemberDeleted as publicOnStaffMemberDeleted } from './bookings-staff-v1-staff-member-staff-members.public.js';
import { onStaffMemberDisconnectedFromUser as publicOnStaffMemberDisconnectedFromUser } from './bookings-staff-v1-staff-member-staff-members.public.js';
import { onStaffMemberFullyCreated as publicOnStaffMemberFullyCreated } from './bookings-staff-v1-staff-member-staff-members.public.js';
import { onStaffMemberUpdated as publicOnStaffMemberUpdated } from './bookings-staff-v1-staff-member-staff-members.public.js';
export const createStaffMember = /*#__PURE__*/ createRESTModule(publicCreateStaffMember);
export const getStaffMember = /*#__PURE__*/ createRESTModule(publicGetStaffMember);
export const updateStaffMember = /*#__PURE__*/ createRESTModule(publicUpdateStaffMember);
export const deleteStaffMember = /*#__PURE__*/ createRESTModule(publicDeleteStaffMember);
export const queryStaffMembers = /*#__PURE__*/ createRESTModule(publicQueryStaffMembers);
export const countStaffMembers = /*#__PURE__*/ createRESTModule(publicCountStaffMembers);
export const connectStaffMemberToUser = /*#__PURE__*/ createRESTModule(publicConnectStaffMemberToUser);
export const searchStaffMembers = /*#__PURE__*/ createRESTModule(publicSearchStaffMembers);
export const disconnectStaffMemberFromUser = /*#__PURE__*/ createRESTModule(publicDisconnectStaffMemberFromUser);
export const assignWorkingHoursSchedule = /*#__PURE__*/ createRESTModule(publicAssignWorkingHoursSchedule);
export const assignCustomSchedule = /*#__PURE__*/ createRESTModule(publicAssignCustomSchedule);
export const bulkUpdateStaffMemberTags = /*#__PURE__*/ createRESTModule(publicBulkUpdateStaffMemberTags);
export const bulkUpdateStaffMemberTagsByFilter = /*#__PURE__*/ createRESTModule(publicBulkUpdateStaffMemberTagsByFilter);
/**
 * Triggered when a Wix user is connected to a staff member.
 *
 *
 * The event is triggered when the connection status changes to `CONNECTED`. It
 * isn't triggered when Wix Bookings sends an invitation to become a Wix user or
 * join a site.
 */
export const onStaffMemberConnectedToUser = createEventModule(publicOnStaffMemberConnectedToUser);
/**
 * Triggered when a staff member is created.
 */
export const onStaffMemberCreated = createEventModule(publicOnStaffMemberCreated);
/**
 * Triggered when a staff member is deleted.
 */
export const onStaffMemberDeleted = createEventModule(publicOnStaffMemberDeleted);
/**
 * Triggered when a Wix user is disconnected from a staff member.
 */
export const onStaffMemberDisconnectedFromUser = createEventModule(publicOnStaffMemberDisconnectedFromUser);
/**
 * Triggered when a staff member is created.
 */
export const onStaffMemberFullyCreated = createEventModule(publicOnStaffMemberFullyCreated);
/**
 * Triggered when a staff member is updated.
 */
export const onStaffMemberUpdated = createEventModule(publicOnStaffMemberUpdated);
export { IdentityType, AssociatedWixIdentityConnectionStatusEnumConnectionStatus, ConnectionStatus, RequestedFields, SortOrder, Mode, ScalarType, AggregationType, Day, Transparency, RecurringIntervalType, LocationType, LocationStatus, LocationsLocationType, DayOfWeek, ApprovalStatus, ScheduleStatus, ConferenceType, CalendarType, Status, SessionType, WebhookIdentityType, } from './bookings-staff-v1-staff-member-staff-members.universal.js';
