// @wix/auto_sdk_members_member-privacy-settings/build/internal/meta.mjs
// src/members-v1-member-privacy-settings-member-privacy-settings.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(opts) {
  const domainToMappings = {
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/privacy-settings",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-privacy-settings";
function getMemberPrivacySettings(payload) {
  function __getMemberPrivacySettings({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_privacy_settings",
      method: "GET",
      methodFqn: "com.wixpress.members.memberprivacysettings.MemberPrivacySettingsService.GetMemberPrivacySettings",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(
        { protoPath: "/v1/privacy-settings", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberPrivacySettings.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getMemberPrivacySettings;
}
function setMemberPrivacySettings(payload) {
  function __setMemberPrivacySettings({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "memberPrivacySettings.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member_privacy_settings",
      method: "POST",
      methodFqn: "com.wixpress.members.memberprivacysettings.MemberPrivacySettingsService.SetMemberPrivacySettings",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(
        { protoPath: "/v1/privacy-settings", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberPrivacySettings.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __setMemberPrivacySettings;
}

// src/members-v1-member-privacy-settings-member-privacy-settings.meta.ts
function getMemberPrivacySettings2() {
  const payload = {};
  const getRequestOptions = getMemberPrivacySettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/privacy-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setMemberPrivacySettings2() {
  const payload = {};
  const getRequestOptions = setMemberPrivacySettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/privacy-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getMemberPrivacySettings2 as getMemberPrivacySettings,
  setMemberPrivacySettings2 as setMemberPrivacySettings
};
