// @wix/auto_sdk_members_member-privacy-settings/build/internal/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getMemberPrivacySettings: () => getMemberPrivacySettings2,
  setMemberPrivacySettings: () => setMemberPrivacySettings2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-member-privacy-settings-member-privacy-settings.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(opts) {
  const domainToMappings = {
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/privacy-settings",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/members/v1/privacy-settings",
        destPath: "/v1/privacy-settings"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-privacy-settings";
function getMemberPrivacySettings(payload) {
  function __getMemberPrivacySettings({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_privacy_settings",
      method: "GET",
      methodFqn: "com.wixpress.members.memberprivacysettings.MemberPrivacySettingsService.GetMemberPrivacySettings",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(
        { protoPath: "/v1/privacy-settings", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberPrivacySettings.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getMemberPrivacySettings;
}
function setMemberPrivacySettings(payload) {
  function __setMemberPrivacySettings({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "memberPrivacySettings.updatedDate" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member_privacy_settings",
      method: "POST",
      methodFqn: "com.wixpress.members.memberprivacysettings.MemberPrivacySettingsService.SetMemberPrivacySettings",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMemberprivacysettingsMemberPrivacySettingsServiceUrl(
        { protoPath: "/v1/privacy-settings", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "memberPrivacySettings.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __setMemberPrivacySettings;
}

// src/members-v1-member-privacy-settings-member-privacy-settings.meta.ts
function getMemberPrivacySettings2() {
  const payload = {};
  const getRequestOptions = getMemberPrivacySettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/privacy-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function setMemberPrivacySettings2() {
  const payload = {};
  const getRequestOptions = setMemberPrivacySettings(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/privacy-settings",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getMemberPrivacySettings,
  setMemberPrivacySettings
});
