// @wix/auto_sdk_bookings_booking-policies/build/es/src/bookings-v1-booking-policy-booking-policies.meta.js
import * as ambassadorWixBookingsV1BookingPolicy from './bookings-v1-booking-policy-booking-policies.http.js';
export function createBookingPolicy() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.createBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getBookingPolicy() {
    const payload = { bookingPolicyId: ':bookingPolicyId' };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.getBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/booking-policies/{bookingPolicyId}',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getStrictestBookingPolicy() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.getStrictestBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/strictest',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateBookingPolicy() {
    const payload = { bookingPolicy: { id: ':bookingPolicyId' } };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.updateBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/booking-policies/{bookingPolicy.id}',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setDefaultBookingPolicy() {
    const payload = { bookingPolicyId: ':bookingPolicyId' };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.setDefaultBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/{bookingPolicyId}:setDefault',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteBookingPolicy() {
    const payload = { bookingPolicyId: ':bookingPolicyId' };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.deleteBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/booking-policies/{bookingPolicyId}',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryBookingPolicies() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.queryBookingPolicies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countBookingPolicies() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.countBookingPolicies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
