// @wix/auto_sdk_members_members-about/build/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  Alignment: () => Alignment,
  AppType: () => AppType,
  BackgroundType: () => BackgroundType,
  Crop: () => Crop,
  DecorationType: () => DecorationType,
  Direction: () => Direction,
  FontType: () => FontType,
  GIFType: () => GIFType,
  InitialExpandedItems: () => InitialExpandedItems,
  LayoutType: () => LayoutType,
  LineStyle: () => LineStyle,
  MapType: () => MapType,
  NodeType: () => NodeType,
  NullValue: () => NullValue,
  Orientation: () => Orientation,
  PluginContainerDataAlignment: () => PluginContainerDataAlignment,
  PollLayoutDirection: () => PollLayoutDirection,
  PollLayoutType: () => PollLayoutType,
  Position: () => Position,
  SortOrder: () => SortOrder,
  Source: () => Source,
  Target: () => Target,
  TextAlignment: () => TextAlignment,
  ThumbnailsAlignment: () => ThumbnailsAlignment,
  Type: () => Type,
  VerticalAlignment: () => VerticalAlignment,
  ViewMode: () => ViewMode,
  ViewRole: () => ViewRole,
  VoteRole: () => VoteRole,
  WebhookIdentityType: () => WebhookIdentityType,
  Width: () => Width,
  WidthType: () => WidthType,
  createMemberAbout: () => createMemberAbout4,
  deleteMemberAbout: () => deleteMemberAbout4,
  getMemberAbout: () => getMemberAbout4,
  getMyMemberAbout: () => getMyMemberAbout4,
  onMemberAboutCreated: () => onMemberAboutCreated2,
  onMemberAboutDeleted: () => onMemberAboutDeleted2,
  onMemberAboutUpdated: () => onMemberAboutUpdated2,
  queryMemberAbouts: () => queryMemberAbouts4,
  updateMemberAbout: () => updateMemberAbout4
});
module.exports = __toCommonJS(index_exports);

// src/members-about-v2-member-about-members-about.public.ts
var import_rename_all_nested_keys2 = require("@wix/sdk-runtime/rename-all-nested-keys");
var import_float3 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp3 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var import_sdk_types = require("@wix/sdk-types");

// src/members-about-v2-member-about-members-about.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/members-about-v2-member-about-members-about.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixMembersAboutV2MembersAboutUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_members-about";
function createMemberAbout(payload) {
  function __createMemberAbout({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "memberAbout.content.nodes.galleryData.options.item.ratio" },
          { path: "memberAbout.content.nodes.imageData.image.duration" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
          { path: "memberAbout.content.nodes.pollData.poll.image.duration" },
          {
            path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
          { path: "memberAbout.content.nodes.videoData.video.duration" },
          { path: "memberAbout.content.nodes.videoData.thumbnail.duration" },
          { path: "memberAbout.content.nodes.audioData.audio.duration" },
          { path: "memberAbout.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "memberAbout.content.metadata.createdTimestamp" },
          { path: "memberAbout.content.metadata.updatedTimestamp" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "POST",
      methodFqn: "wix.members.about.v2.MembersAbout.CreateMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createMemberAbout;
}
function updateMemberAbout(payload) {
  function __updateMemberAbout({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "memberAbout.content.nodes.galleryData.options.item.ratio" },
          { path: "memberAbout.content.nodes.imageData.image.duration" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
          { path: "memberAbout.content.nodes.pollData.poll.image.duration" },
          {
            path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
          { path: "memberAbout.content.nodes.videoData.video.duration" },
          { path: "memberAbout.content.nodes.videoData.thumbnail.duration" },
          { path: "memberAbout.content.nodes.audioData.audio.duration" },
          { path: "memberAbout.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "memberAbout.content.metadata.createdTimestamp" },
          { path: "memberAbout.content.metadata.updatedTimestamp" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "PATCH",
      methodFqn: "wix.members.about.v2.MembersAbout.UpdateMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{memberAbout.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMemberAbout;
}
function deleteMemberAbout(payload) {
  function __deleteMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "DELETE",
      methodFqn: "wix.members.about.v2.MembersAbout.DeleteMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMemberAbout;
}
function getMemberAbout(payload) {
  function __getMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "GET",
      methodFqn: "wix.members.about.v2.MembersAbout.GetMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMemberAbout;
}
function getMyMemberAbout(payload) {
  function __getMyMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "GET",
      methodFqn: "wix.members.about.v2.MembersAbout.GetMyMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/my",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMyMemberAbout;
}
function queryMemberAbouts(payload) {
  function __queryMemberAbouts({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "POST",
      methodFqn: "wix.members.about.v2.MembersAbout.QueryMemberAbouts",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbouts.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbouts.content.nodes.imageData.image.duration" },
            { path: "memberAbouts.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbouts.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbouts.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.appEmbedData.image.duration"
            },
            { path: "memberAbouts.content.nodes.videoData.video.duration" },
            {
              path: "memberAbouts.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbouts.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbouts.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbouts.content.metadata.createdTimestamp" },
            { path: "memberAbouts.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMemberAbouts;
}

// src/members-about-v2-member-about-members-about.universal.ts
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["LINK"] = "LINK";
  Type2["ACTION"] = "ACTION";
  return Type2;
})(Type || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["CENTER"] = "CENTER";
  Alignment2["LEFT"] = "LEFT";
  Alignment2["RIGHT"] = "RIGHT";
  return Alignment2;
})(Alignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  return Source2;
})(Source || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  Position2["HIDDEN"] = "HIDDEN";
  return Position2;
})(Position || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function createMemberAbout2(memberAbout) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberAbout
  });
  const reqOpts = createMemberAbout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.memberAbout;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberAbout: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberAbout"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateMemberAbout2(_id, memberAbout) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberAbout: { ...memberAbout, id: _id }
  });
  const reqOpts = updateMemberAbout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.memberAbout;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { memberAbout: "$[1]" },
        explicitPathsToArguments: { "memberAbout.id": "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id", "memberAbout"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteMemberAbout2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = deleteMemberAbout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMemberAbout2(_id) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ id: _id });
  const reqOpts = getMemberAbout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data)?.memberAbout;
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { id: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["_id"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getMyMemberAbout2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({});
  const reqOpts = getMyMemberAbout(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryMemberAbouts2() {
  const { httpClient, sideEffects } = arguments[0];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryMemberAbouts(payload);
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, {}];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
        ...args?.[1],
        query: args?.[0]
      });
    },
    responseTransformer: ({
      data
    }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(data);
      return {
        items: transformedData?.memberAbouts,
        pagingMetadata: transformedData?.metadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}

// src/members-about-v2-member-about-members-about.public.ts
function createMemberAbout3(httpClient) {
  return (memberAbout) => createMemberAbout2(
    memberAbout,
    // @ts-ignore
    { httpClient }
  );
}
function updateMemberAbout3(httpClient) {
  return (_id, memberAbout) => updateMemberAbout2(
    _id,
    memberAbout,
    // @ts-ignore
    { httpClient }
  );
}
function deleteMemberAbout3(httpClient) {
  return (_id) => deleteMemberAbout2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function getMemberAbout3(httpClient) {
  return (_id) => getMemberAbout2(
    _id,
    // @ts-ignore
    { httpClient }
  );
}
function getMyMemberAbout3(httpClient) {
  return () => getMyMemberAbout2(
    // @ts-ignore
    { httpClient }
  );
}
function queryMemberAbouts3(httpClient) {
  return () => queryMemberAbouts2(
    // @ts-ignore
    { httpClient }
  );
}
var onMemberAboutCreated = (0, import_sdk_types.EventDefinition)(
  "wix.members.about.v2.member_about_created",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          {
            path: "entity.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "entity.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "entity.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "entity.content.nodes.galleryData.options.item.ratio" },
          { path: "entity.content.nodes.imageData.image.duration" },
          { path: "entity.content.nodes.mapData.mapSettings.lat" },
          { path: "entity.content.nodes.mapData.mapSettings.lng" },
          { path: "entity.content.nodes.pollData.poll.image.duration" },
          {
            path: "entity.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "entity.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "entity.content.nodes.appEmbedData.image.duration" },
          { path: "entity.content.nodes.videoData.video.duration" },
          { path: "entity.content.nodes.videoData.thumbnail.duration" },
          { path: "entity.content.nodes.audioData.audio.duration" },
          { path: "entity.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.content.metadata.createdTimestamp" },
          { path: "entity.content.metadata.updatedTimestamp" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMemberAboutDeleted = (0, import_sdk_types.EventDefinition)(
  "wix.members.about.v2.member_about_deleted",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          {
            path: "undefined.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "undefined.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "undefined.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "undefined.content.nodes.galleryData.options.item.ratio" },
          { path: "undefined.content.nodes.imageData.image.duration" },
          { path: "undefined.content.nodes.mapData.mapSettings.lat" },
          { path: "undefined.content.nodes.mapData.mapSettings.lng" },
          { path: "undefined.content.nodes.pollData.poll.image.duration" },
          {
            path: "undefined.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "undefined.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "undefined.content.nodes.appEmbedData.image.duration" },
          { path: "undefined.content.nodes.videoData.video.duration" },
          { path: "undefined.content.nodes.videoData.thumbnail.duration" },
          { path: "undefined.content.nodes.audioData.audio.duration" },
          { path: "undefined.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "undefined.content.metadata.createdTimestamp" },
          { path: "undefined.content.metadata.updatedTimestamp" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();
var onMemberAboutUpdated = (0, import_sdk_types.EventDefinition)(
  "wix.members.about.v2.member_about_updated",
  true,
  (event) => (0, import_rename_all_nested_keys2.renameKeysFromRESTResponseToSDKResponse)(
    (0, import_transform_paths2.transformPaths)(event, [
      {
        transformFn: import_float3.transformRESTFloatToSDKFloat,
        paths: [
          {
            path: "entity.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "entity.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "entity.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "entity.content.nodes.galleryData.options.item.ratio" },
          { path: "entity.content.nodes.imageData.image.duration" },
          { path: "entity.content.nodes.mapData.mapSettings.lat" },
          { path: "entity.content.nodes.mapData.mapSettings.lng" },
          { path: "entity.content.nodes.pollData.poll.image.duration" },
          {
            path: "entity.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "entity.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "entity.content.nodes.appEmbedData.image.duration" },
          { path: "entity.content.nodes.videoData.video.duration" },
          { path: "entity.content.nodes.videoData.thumbnail.duration" },
          { path: "entity.content.nodes.audioData.audio.duration" },
          { path: "entity.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: import_timestamp3.transformRESTTimestampToSDKTimestamp,
        paths: [
          { path: "entity.content.metadata.createdTimestamp" },
          { path: "entity.content.metadata.updatedTimestamp" },
          { path: "metadata.eventTime" }
        ]
      }
    ])
  )
)();

// src/members-about-v2-member-about-members-about.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var import_event_definition_modules = require("@wix/sdk-runtime/event-definition-modules");
var createMemberAbout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(createMemberAbout3);
var updateMemberAbout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(updateMemberAbout3);
var deleteMemberAbout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(deleteMemberAbout3);
var getMemberAbout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getMemberAbout3);
var getMyMemberAbout4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(getMyMemberAbout3);
var queryMemberAbouts4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(queryMemberAbouts3);
var onMemberAboutCreated2 = (0, import_event_definition_modules.createEventModule)(onMemberAboutCreated);
var onMemberAboutDeleted2 = (0, import_event_definition_modules.createEventModule)(onMemberAboutDeleted);
var onMemberAboutUpdated2 = (0, import_event_definition_modules.createEventModule)(onMemberAboutUpdated);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Alignment,
  AppType,
  BackgroundType,
  Crop,
  DecorationType,
  Direction,
  FontType,
  GIFType,
  InitialExpandedItems,
  LayoutType,
  LineStyle,
  MapType,
  NodeType,
  NullValue,
  Orientation,
  PluginContainerDataAlignment,
  PollLayoutDirection,
  PollLayoutType,
  Position,
  SortOrder,
  Source,
  Target,
  TextAlignment,
  ThumbnailsAlignment,
  Type,
  VerticalAlignment,
  ViewMode,
  ViewRole,
  VoteRole,
  WebhookIdentityType,
  Width,
  WidthType,
  createMemberAbout,
  deleteMemberAbout,
  getMemberAbout,
  getMyMemberAbout,
  onMemberAboutCreated,
  onMemberAboutDeleted,
  onMemberAboutUpdated,
  queryMemberAbouts,
  updateMemberAbout
});
