// @wix/auto_sdk_members_members-about/build/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createMemberAbout: () => createMemberAbout2,
  deleteMemberAbout: () => deleteMemberAbout2,
  getMemberAbout: () => getMemberAbout2,
  getMyMemberAbout: () => getMyMemberAbout2,
  queryMemberAbouts: () => queryMemberAbouts2,
  updateMemberAbout: () => updateMemberAbout2
});
module.exports = __toCommonJS(meta_exports);

// src/members-about-v2-member-about-members-about.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixMembersAboutV2MembersAboutUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/members-about-api",
        destPath: ""
      },
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v2/abouts",
        destPath: "/v2/abouts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_members-about";
function createMemberAbout(payload) {
  function __createMemberAbout({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "memberAbout.content.nodes.galleryData.options.item.ratio" },
          { path: "memberAbout.content.nodes.imageData.image.duration" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
          { path: "memberAbout.content.nodes.pollData.poll.image.duration" },
          {
            path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
          { path: "memberAbout.content.nodes.videoData.video.duration" },
          { path: "memberAbout.content.nodes.videoData.thumbnail.duration" },
          { path: "memberAbout.content.nodes.audioData.audio.duration" },
          { path: "memberAbout.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "memberAbout.content.metadata.createdTimestamp" },
          { path: "memberAbout.content.metadata.updatedTimestamp" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "POST",
      methodFqn: "wix.members.about.v2.MembersAbout.CreateMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createMemberAbout;
}
function updateMemberAbout(payload) {
  function __updateMemberAbout({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          {
            path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
          },
          { path: "memberAbout.content.nodes.galleryData.options.item.ratio" },
          { path: "memberAbout.content.nodes.imageData.image.duration" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
          { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
          { path: "memberAbout.content.nodes.pollData.poll.image.duration" },
          {
            path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
          { path: "memberAbout.content.nodes.videoData.video.duration" },
          { path: "memberAbout.content.nodes.videoData.thumbnail.duration" },
          { path: "memberAbout.content.nodes.audioData.audio.duration" },
          { path: "memberAbout.content.nodes.audioData.coverImage.duration" }
        ]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "memberAbout.content.metadata.createdTimestamp" },
          { path: "memberAbout.content.metadata.updatedTimestamp" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "PATCH",
      methodFqn: "wix.members.about.v2.MembersAbout.UpdateMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{memberAbout.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMemberAbout;
}
function deleteMemberAbout(payload) {
  function __deleteMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "DELETE",
      methodFqn: "wix.members.about.v2.MembersAbout.DeleteMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMemberAbout;
}
function getMemberAbout(payload) {
  function __getMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "GET",
      methodFqn: "wix.members.about.v2.MembersAbout.GetMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMemberAbout;
}
function getMyMemberAbout(payload) {
  function __getMyMemberAbout({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "GET",
      methodFqn: "wix.members.about.v2.MembersAbout.GetMyMemberAbout",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/my",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbout.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbout.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbout.content.nodes.imageData.image.duration" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbout.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbout.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbout.content.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "memberAbout.content.nodes.appEmbedData.image.duration" },
            { path: "memberAbout.content.nodes.videoData.video.duration" },
            {
              path: "memberAbout.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbout.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbout.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbout.content.metadata.createdTimestamp" },
            { path: "memberAbout.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMyMemberAbout;
}
function queryMemberAbouts(payload) {
  function __queryMemberAbouts({ host }) {
    const metadata = {
      entityFqdn: "wix.members.about.v2.member_about",
      method: "POST",
      methodFqn: "wix.members.about.v2.MembersAbout.QueryMemberAbouts",
      packageName: PACKAGE_NAME,
      url: resolveWixMembersAboutV2MembersAboutUrl({
        protoPath: "/v2/abouts/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            {
              path: "memberAbouts.content.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "memberAbouts.content.nodes.galleryData.options.item.ratio"
            },
            { path: "memberAbouts.content.nodes.imageData.image.duration" },
            { path: "memberAbouts.content.nodes.mapData.mapSettings.lat" },
            { path: "memberAbouts.content.nodes.mapData.mapSettings.lng" },
            {
              path: "memberAbouts.content.nodes.pollData.poll.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "memberAbouts.content.nodes.appEmbedData.image.duration"
            },
            { path: "memberAbouts.content.nodes.videoData.video.duration" },
            {
              path: "memberAbouts.content.nodes.videoData.thumbnail.duration"
            },
            { path: "memberAbouts.content.nodes.audioData.audio.duration" },
            {
              path: "memberAbouts.content.nodes.audioData.coverImage.duration"
            }
          ]
        },
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "memberAbouts.content.metadata.createdTimestamp" },
            { path: "memberAbouts.content.metadata.updatedTimestamp" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMemberAbouts;
}

// src/members-about-v2-member-about-members-about.meta.ts
function createMemberAbout2() {
  const payload = {};
  const getRequestOptions = createMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/abouts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMemberAbout2() {
  const payload = { memberAbout: { id: ":memberAboutId" } };
  const getRequestOptions = updateMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v2/abouts/{memberAbout.id}",
    pathParams: { memberAboutId: "memberAboutId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberAbout2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v2/abouts/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMemberAbout2() {
  const payload = { id: ":id" };
  const getRequestOptions = getMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/abouts/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMyMemberAbout2() {
  const payload = {};
  const getRequestOptions = getMyMemberAbout(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/abouts/my",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMemberAbouts2() {
  const payload = {};
  const getRequestOptions = queryMemberAbouts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v2/abouts/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createMemberAbout,
  deleteMemberAbout,
  getMemberAbout,
  getMyMemberAbout,
  queryMemberAbouts,
  updateMemberAbout
});
