// @wix/auto_sdk_ecom_tax-groups/build/es/src/billing-v1-tax-group-tax-groups.meta.js
import * as ambassadorWixBillingV1TaxGroup from './billing-v1-tax-group-tax-groups.http.js';
export function createTaxGroup() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.createTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-groups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTaxGroup() {
    const payload = { taxGroupId: ':taxGroupId' };
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.getTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tax-groups/{taxGroupId}',
        pathParams: { taxGroupId: 'taxGroupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateTaxGroup() {
    const payload = { taxGroup: { id: ':taxGroupId' } };
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.updateTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/tax-groups/{taxGroup.id}',
        pathParams: { taxGroupId: 'taxGroupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteTaxGroup() {
    const payload = { taxGroupId: ':taxGroupId' };
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.deleteTaxGroup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/tax-groups/{taxGroupId}',
        pathParams: { taxGroupId: 'taxGroupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryTaxGroups() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.queryTaxGroups(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-groups/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDefaultTaxGroups() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.listDefaultTaxGroups(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/tax-groups/default-tax-groups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDefaultTaxGroupsByAppIds() {
    const payload = {};
    const getRequestOptions = ambassadorWixBillingV1TaxGroup.listDefaultTaxGroupsByAppIds(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/tax-groups/list-default-tax-groups-by-app-ids',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
