// @wix/auto_sdk_bookings_booking-policy-snapshots/build/internal/es/src/bookings-policy-snapshots-v1-booking-policy-snapshot-booking-policy-snapshots.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveComWixpressBookingsPolicySnapshotsV1BookingPolicySnapshotsUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/booking-policy-snapshots',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/booking-policy-snapshots',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/_api/booking-policy-snapshots',
                destPath: '',
            },
            {
                srcPath: '/booking-policy-snapshots',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_bookings_booking-policy-snapshots';
/**
 * Retrieves a list of booking policy snapshots by *booking IDs*
 * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/bookings/bookings/introduction) | [REST](https://dev.wix.com/docs/rest/business-solutions/bookings/bookings/about-the-bookings-apis)).
 */
export function listPolicySnapshotsByBookingIds(payload) {
    function __listPolicySnapshotsByBookingIds({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.policy_snapshots.v1.booking_policy_snapshot',
            method: 'GET',
            methodFqn: 'com.wixpress.bookings.policy.snapshots.v1.BookingPolicySnapshots.ListPolicySnapshotsByBookingIds',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsPolicySnapshotsV1BookingPolicySnapshotsUrl({ protoPath: '/v1/policy-snapshots', data: payload, host }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'bookingPolicySnapshots.createdDate' },
                        { path: 'bookingPolicySnapshots.policy.createdDate' },
                        { path: 'bookingPolicySnapshots.policy.updatedDate' },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listPolicySnapshotsByBookingIds;
}
