// @wix/auto_sdk_bookings_app-settings/build/internal/cjs/src/bookings-v2-bookings-app-settings-app-settings.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateBookingsAppSettings = exports.getBookingsAppSettings = exports.createBookingsAppSettings = void 0;
const ambassadorWixBookingsV2BookingsAppSettings = __importStar(require("./bookings-v2-bookings-app-settings-app-settings.http.js"));
function createBookingsAppSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2BookingsAppSettings.createBookingsAppSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings-app-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createBookingsAppSettings = createBookingsAppSettings;
function getBookingsAppSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2BookingsAppSettings.getBookingsAppSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getBookingsAppSettings = getBookingsAppSettings;
function updateBookingsAppSettings() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2BookingsAppSettings.updateBookingsAppSettings(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/bookings-app-settings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateBookingsAppSettings = updateBookingsAppSettings;
