// @wix/auto_sdk_members_member-to-member-block/build/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  blockMember: () => blockMember2,
  listCurrentMemberBlocking: () => listCurrentMemberBlocking2,
  listMemberBlocks: () => listMemberBlocks2,
  unblockMember: () => unblockMember2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-member-to-member-block-member-to-member-block.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/member-to-member-blocks",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/member-to-member-blocks",
        destPath: "/v1/member-to-member-blocks"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/member-to-member-blocks",
        destPath: "/v1/member-to-member-blocks"
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/_api/members/v1/member-to-member-blocks",
        destPath: "/v1/member-to-member-blocks"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-to-member-block";
function blockMember(payload) {
  function __blockMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "POST",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.BlockMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/my",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __blockMember;
}
function unblockMember(payload) {
  function __unblockMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "DELETE",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.UnblockMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/my/{memberId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __unblockMember;
}
function listCurrentMemberBlocking(payload) {
  function __listCurrentMemberBlocking({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "GET",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.ListCurrentMemberBlocking",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/my",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listCurrentMemberBlocking;
}
function listMemberBlocks(payload) {
  function __listMemberBlocks({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "GET",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.ListMemberBlocks",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/{memberId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMemberBlocks;
}

// src/members-v1-member-to-member-block-member-to-member-block.meta.ts
function blockMember2() {
  const payload = {};
  const getRequestOptions = blockMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/member-to-member-blocks/members/my",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function unblockMember2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = unblockMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/member-to-member-blocks/members/my/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCurrentMemberBlocking2() {
  const payload = {};
  const getRequestOptions = listCurrentMemberBlocking(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/member-to-member-blocks/members/my",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMemberBlocks2() {
  const payload = { memberId: ":memberId" };
  const getRequestOptions = listMemberBlocks(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/member-to-member-blocks/members/{memberId}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  blockMember,
  listCurrentMemberBlocking,
  listMemberBlocks,
  unblockMember
});
