// @wix/auto_sdk_ecom_shipping-options/build/internal/es/src/ecom-v1-shipping-option-shipping-options.public.js
import { createShippingOption as universalCreateShippingOption, deleteShippingOption as universalDeleteShippingOption, getShippingOption as universalGetShippingOption, queryShippingOptions as universalQueryShippingOptions, updateExtendedFields as universalUpdateExtendedFields, updateShippingOption as universalUpdateShippingOption, } from './ecom-v1-shipping-option-shipping-options.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/ecom' };
export function createShippingOption(httpClient) {
    return (shippingOption) => universalCreateShippingOption(shippingOption, 
    // @ts-ignore
    { httpClient });
}
export function getShippingOption(httpClient) {
    return (shippingOptionId) => universalGetShippingOption(shippingOptionId, 
    // @ts-ignore
    { httpClient });
}
export function updateShippingOption(httpClient) {
    return (_id, shippingOption) => universalUpdateShippingOption(_id, shippingOption, 
    // @ts-ignore
    { httpClient });
}
export function deleteShippingOption(httpClient) {
    return (shippingOptionId) => universalDeleteShippingOption(shippingOptionId, 
    // @ts-ignore
    { httpClient });
}
export function queryShippingOptions(httpClient) {
    return () => universalQueryShippingOptions(
    // @ts-ignore
    { httpClient });
}
export function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => universalUpdateExtendedFields(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
export { ConditionType, LogicalOperator, SortOrder, WebhookIdentityType, } from './ecom-v1-shipping-option-shipping-options.universal.js';
