// @wix/auto_sdk_bookings_categories/build/es/src/bookings-v1-category-categories.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { EventDefinition } from '@wix/sdk-types';
import { createCategory as universalCreateCategory, deleteCategory as universalDeleteCategory, listCategories as universalListCategories, updateCategory as universalUpdateCategory, } from './bookings-v1-category-categories.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/bookings' };
export function listCategories(httpClient) {
    return (options) => universalListCategories(options, 
    // @ts-ignore
    { httpClient });
}
export function createCategory(httpClient) {
    return (category) => universalCreateCategory(category, 
    // @ts-ignore
    { httpClient });
}
export function updateCategory(httpClient) {
    return (_id, category) => universalUpdateCategory(_id, category, 
    // @ts-ignore
    { httpClient });
}
export function deleteCategory(httpClient) {
    return (_id, options) => universalDeleteCategory(_id, options, 
    // @ts-ignore
    { httpClient });
}
export const onCategoryNotification = EventDefinition('com.wixpress.bookings.services.api.v1.CategoryNotification', false, (event) => renameKeysFromRESTResponseToSDKResponse(event))();
export { Event, Status, WebhookIdentityType, } from './bookings-v1-category-categories.universal.js';
