// @wix/auto_sdk_bookings_resource-types/build/internal/cjs/src/bookings-resources-v2-resource-type-resource-types.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.countResourceTypes = exports.queryResourceTypes = exports.deleteResourceType = exports.updateResourceType = exports.getResourceType = exports.createResourceType = void 0;
const ambassadorWixBookingsResourcesV2ResourceType = __importStar(require("./bookings-resources-v2-resource-type-resource-types.http.js"));
function createResourceType() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.createResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/resource-types',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createResourceType = createResourceType;
function getResourceType() {
    const payload = { resourceTypeId: ':resourceTypeId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.getResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/resources/resource-types/{resourceTypeId}',
        pathParams: { resourceTypeId: 'resourceTypeId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getResourceType = getResourceType;
function updateResourceType() {
    const payload = { resourceType: { id: ':resourceTypeId' } };
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.updateResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/resources/resource-types/{resourceType.id}',
        pathParams: { resourceTypeId: 'resourceTypeId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateResourceType = updateResourceType;
function deleteResourceType() {
    const payload = { resourceTypeId: ':resourceTypeId' };
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.deleteResourceType(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/resources/resource-types/{resourceTypeId}',
        pathParams: { resourceTypeId: 'resourceTypeId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteResourceType = deleteResourceType;
function queryResourceTypes() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.queryResourceTypes(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/resource-types/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryResourceTypes = queryResourceTypes;
function countResourceTypes() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsResourcesV2ResourceType.countResourceTypes(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/resources/resource-types/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.countResourceTypes = countResourceTypes;
