// @wix/auto_sdk_bookings_availability-calendar/build/es/src/bookings-availability-v1-slot-availability-availability-calendar.meta.js
import * as ambassadorWixBookingsAvailabilityV1SlotAvailability from './bookings-availability-v1-slot-availability-availability-calendar.http.js';
export function queryAvailability() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsAvailabilityV1SlotAvailability.queryAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/availability/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getScheduleAvailability() {
    const payload = { scheduleId: ':scheduleId' };
    const getRequestOptions = ambassadorWixBookingsAvailabilityV1SlotAvailability.getScheduleAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/availability/schedule/{scheduleId}',
        pathParams: { scheduleId: 'scheduleId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function calculateMultiSlotAvailability() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsAvailabilityV1SlotAvailability.calculateMultiSlotAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/availability/multislot',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
