// @wix/auto_sdk_bookings_booking-fees/build/internal/es/src/bookings-fees-v1-booking-fee-booking-fees.meta.js
import * as ambassadorWixBookingsFeesV1BookingFee from './bookings-fees-v1-booking-fee-booking-fees.http.js';
export function listBookingFeesByBookingIds() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsFeesV1BookingFee.listBookingFeesByBookingIds(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/booking-fees',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function applyBookingFeesToOrder() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsFeesV1BookingFee.applyBookingFeesToOrder(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-fees/apply',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function collectAppliedBookingFees() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsFeesV1BookingFee.collectAppliedBookingFees(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-fees/collect',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
