// @wix/auto_sdk_bookings_attendance/build/cjs/src/bookings-v2-attendance-attendance.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.queryAttendance = exports.bulkSetAttendance = exports.setAttendance = exports.getAttendance = void 0;
const ambassadorWixBookingsV2Attendance = __importStar(require("./bookings-v2-attendance-attendance.http.js"));
function getAttendance() {
    const payload = { attendanceId: ':attendanceId' };
    const getRequestOptions = ambassadorWixBookingsV2Attendance.getAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/attendance/{attendanceId}',
        pathParams: { attendanceId: 'attendanceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getAttendance = getAttendance;
function setAttendance() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Attendance.setAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/attendance/set',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setAttendance = setAttendance;
function bulkSetAttendance() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Attendance.bulkSetAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/attendance/set',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkSetAttendance = bulkSetAttendance;
function queryAttendance() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Attendance.queryAttendance(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/attendance/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryAttendance = queryAttendance;
