// @wix/auto_sdk_members_members/build/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  approveMember: () => approveMember2,
  blockMember: () => blockMember2,
  bulkApproveMembers: () => bulkApproveMembers2,
  bulkBlockMembers: () => bulkBlockMembers2,
  bulkDeleteMembers: () => bulkDeleteMembers2,
  bulkDeleteMembersByFilter: () => bulkDeleteMembersByFilter2,
  createMember: () => createMember2,
  deleteMember: () => deleteMember2,
  deleteMemberAddresses: () => deleteMemberAddresses2,
  deleteMemberEmails: () => deleteMemberEmails2,
  deleteMemberPhones: () => deleteMemberPhones2,
  deleteMyMember: () => deleteMyMember2,
  disconnectMember: () => disconnectMember2,
  getCurrentMember: () => getCurrentMember,
  getMember: () => getMember2,
  joinCommunity: () => joinCommunity2,
  leaveCommunity: () => leaveCommunity2,
  listMembers: () => listMembers2,
  muteMember: () => muteMember2,
  queryMembers: () => queryMembers2,
  unmuteMember: () => unmuteMember2,
  updateCurrentMemberSlug: () => updateCurrentMemberSlug,
  updateMember: () => updateMember2,
  updateMemberSlug: () => updateMemberSlug2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-member-members.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersApiMembersUrl(opts) {
  const domainToMappings = {
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/members-ng-api",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/_api/members/v1/members",
        destPath: "/v1/members"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_members";
function updateMySlug(payload) {
  function __updateMySlug({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.UpdateMySlug",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/my/slug",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMySlug;
}
function updateMemberSlug(payload) {
  function __updateMemberSlug({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.UpdateMemberSlug",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/slug",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMemberSlug;
}
function joinCommunity(payload) {
  function __joinCommunity({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.JoinCommunity",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/join-community",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __joinCommunity;
}
function leaveCommunity(payload) {
  function __leaveCommunity({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.LeaveCommunity",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/leave-community",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __leaveCommunity;
}
function getMyMember(payload) {
  function __getMyMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "GET",
      methodFqn: "com.wixpress.members.api.Members.GetMyMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/my",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMyMember;
}
function getMember(payload) {
  function __getMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "GET",
      methodFqn: "com.wixpress.members.api.Members.GetMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMember;
}
function listMembers(payload) {
  function __listMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "GET",
      methodFqn: "com.wixpress.members.api.Members.ListMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload, true),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "members.createdDate" },
            { path: "members.updatedDate" },
            { path: "members.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listMembers;
}
function queryMembers(payload) {
  function __queryMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.QueryMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "members.createdDate" },
            { path: "members.updatedDate" },
            { path: "members.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryMembers;
}
function muteMember(payload) {
  function __muteMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.MuteMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/mute",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __muteMember;
}
function unmuteMember(payload) {
  function __unmuteMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.UnmuteMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/unmute",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __unmuteMember;
}
function approveMember(payload) {
  function __approveMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.ApproveMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/approve",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __approveMember;
}
function blockMember(payload) {
  function __blockMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.BlockMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/block",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __blockMember;
}
function disconnectMember(payload) {
  function __disconnectMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.DisconnectMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/disconnect",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __disconnectMember;
}
function deleteMember(payload) {
  function __deleteMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "DELETE",
      methodFqn: "com.wixpress.members.api.Members.DeleteMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMember;
}
function deleteMyMember(payload) {
  function __deleteMyMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "DELETE",
      methodFqn: "com.wixpress.members.api.Members.DeleteMyMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/my",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteMyMember;
}
function bulkDeleteMembers(payload) {
  function __bulkDeleteMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.BulkDeleteMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/bulk/delete",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteMembers;
}
function bulkDeleteMembersByFilter(payload) {
  function __bulkDeleteMembersByFilter({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.BulkDeleteMembersByFilter",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/bulk/delete-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkDeleteMembersByFilter;
}
function bulkApproveMembers(payload) {
  function __bulkApproveMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.BulkApproveMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/bulk/approve-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkApproveMembers;
}
function bulkBlockMembers(payload) {
  function __bulkBlockMembers({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.BulkBlockMembers",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/bulk/block-by-filter",
        data: payload,
        host
      }),
      data: payload
    };
    return metadata;
  }
  return __bulkBlockMembers;
}
function createMember(payload) {
  function __createMember({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "member.createdDate" },
          { path: "member.updatedDate" },
          { path: "member.lastLoginDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "POST",
      methodFqn: "com.wixpress.members.api.Members.CreateMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createMember;
}
function updateMember(payload) {
  function __updateMember({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "member.createdDate" },
          { path: "member.updatedDate" },
          { path: "member.lastLoginDate" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "PATCH",
      methodFqn: "com.wixpress.members.api.Members.UpdateMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{member.id}",
        data: serializedData,
        host
      }),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateMember;
}
function deleteMemberPhones(payload) {
  function __deleteMemberPhones({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "DELETE",
      methodFqn: "com.wixpress.members.api.Members.DeleteMemberPhones",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/phones",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteMemberPhones;
}
function deleteMemberEmails(payload) {
  function __deleteMemberEmails({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "DELETE",
      methodFqn: "com.wixpress.members.api.Members.DeleteMemberEmails",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/emails",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteMemberEmails;
}
function deleteMemberAddresses(payload) {
  function __deleteMemberAddresses({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member",
      method: "DELETE",
      methodFqn: "com.wixpress.members.api.Members.DeleteMemberAddresses",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersApiMembersUrl({
        protoPath: "/v1/members/{id}/addresses",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "member.createdDate" },
            { path: "member.updatedDate" },
            { path: "member.lastLoginDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __deleteMemberAddresses;
}

// src/members-v1-member-members.meta.ts
function updateCurrentMemberSlug() {
  const payload = {};
  const getRequestOptions = updateMySlug(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/my/slug",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMemberSlug2() {
  const payload = { id: ":id" };
  const getRequestOptions = updateMemberSlug(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/{id}/slug",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function joinCommunity2() {
  const payload = {};
  const getRequestOptions = joinCommunity(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/join-community",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function leaveCommunity2() {
  const payload = {};
  const getRequestOptions = leaveCommunity(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/leave-community",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCurrentMember() {
  const payload = {};
  const getRequestOptions = getMyMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/members/my",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMember2() {
  const payload = { id: ":id" };
  const getRequestOptions = getMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/members/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listMembers2() {
  const payload = {};
  const getRequestOptions = listMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryMembers2() {
  const payload = {};
  const getRequestOptions = queryMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function muteMember2() {
  const payload = { id: ":id" };
  const getRequestOptions = muteMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/{id}/mute",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function unmuteMember2() {
  const payload = { id: ":id" };
  const getRequestOptions = unmuteMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/{id}/unmute",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function approveMember2() {
  const payload = { id: ":id" };
  const getRequestOptions = approveMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/{id}/approve",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function blockMember2() {
  const payload = { id: ":id" };
  const getRequestOptions = blockMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/{id}/block",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function disconnectMember2() {
  const payload = { id: ":id" };
  const getRequestOptions = disconnectMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/{id}/disconnect",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMember2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/members/{id}",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMyMember2() {
  const payload = {};
  const getRequestOptions = deleteMyMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/members/my",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteMembers2() {
  const payload = {};
  const getRequestOptions = bulkDeleteMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/bulk/delete",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteMembersByFilter2() {
  const payload = {};
  const getRequestOptions = bulkDeleteMembersByFilter(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/bulk/delete-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkApproveMembers2() {
  const payload = {};
  const getRequestOptions = bulkApproveMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/bulk/approve-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkBlockMembers2() {
  const payload = {};
  const getRequestOptions = bulkBlockMembers(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members/bulk/block-by-filter",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function createMember2() {
  const payload = {};
  const getRequestOptions = createMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateMember2() {
  const payload = { member: { id: ":memberId" } };
  const getRequestOptions = updateMember(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/members/{member.id}",
    pathParams: { memberId: "memberId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberPhones2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteMemberPhones(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/members/{id}/phones",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberEmails2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteMemberEmails(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/members/{id}/emails",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteMemberAddresses2() {
  const payload = { id: ":id" };
  const getRequestOptions = deleteMemberAddresses(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/members/{id}/addresses",
    pathParams: { id: "id" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  approveMember,
  blockMember,
  bulkApproveMembers,
  bulkBlockMembers,
  bulkDeleteMembers,
  bulkDeleteMembersByFilter,
  createMember,
  deleteMember,
  deleteMemberAddresses,
  deleteMemberEmails,
  deleteMemberPhones,
  deleteMyMember,
  disconnectMember,
  getCurrentMember,
  getMember,
  joinCommunity,
  leaveCommunity,
  listMembers,
  muteMember,
  queryMembers,
  unmuteMember,
  updateCurrentMemberSlug,
  updateMember,
  updateMemberSlug
});
