// @wix/auto_sdk_bookings_categories/build/internal/cjs/src/bookings-v1-category-categories.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Status = exports.Event = exports.onCategoryNotification = exports.deleteCategory = exports.updateCategory = exports.createCategory = exports.listCategories = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_v1_category_categories_universal_js_1 = require("./bookings-v1-category-categories.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function listCategories(httpClient) {
    return (options) => (0, bookings_v1_category_categories_universal_js_1.listCategories)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listCategories = listCategories;
function createCategory(httpClient) {
    return (category) => (0, bookings_v1_category_categories_universal_js_1.createCategory)(category, 
    // @ts-ignore
    { httpClient });
}
exports.createCategory = createCategory;
function updateCategory(httpClient) {
    return (_id, category) => (0, bookings_v1_category_categories_universal_js_1.updateCategory)(_id, category, 
    // @ts-ignore
    { httpClient });
}
exports.updateCategory = updateCategory;
function deleteCategory(httpClient) {
    return (_id, options) => (0, bookings_v1_category_categories_universal_js_1.deleteCategory)(_id, options, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCategory = deleteCategory;
exports.onCategoryNotification = (0, sdk_types_1.EventDefinition)('com.wixpress.bookings.services.api.v1.CategoryNotification', false, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(event))();
var bookings_v1_category_categories_universal_js_2 = require("./bookings-v1-category-categories.universal.js");
Object.defineProperty(exports, "Event", { enumerable: true, get: function () { return bookings_v1_category_categories_universal_js_2.Event; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return bookings_v1_category_categories_universal_js_2.Status; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_v1_category_categories_universal_js_2.WebhookIdentityType; } });
