// @wix/auto_sdk_bookings_categories/build/internal/cjs/src/bookings-v1-category-categories.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports._delete = exports.update = exports.create = exports.list = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl(opts) {
    const domainToMappings = {
        'www.wixapis.com': [
            {
                srcPath: '/bookings/v1/categories',
                destPath: '/v1/categories',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/bookings/v1/categories',
                destPath: '/v1/categories',
            },
            {
                srcPath: '/_api/services-server/v1/categories',
                destPath: '/v1/categories',
            },
            {
                srcPath: '/_api/services-server/v1/batch/categories',
                destPath: '/v1/batch/categories',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/bookings/v1/categories',
                destPath: '/v1/categories',
            },
            {
                srcPath: '/_api/bookings/v1/batch/categories',
                destPath: '/v1/batch/categories',
            },
            {
                srcPath: '/_api/services-server/v1/categories',
                destPath: '/v1/categories',
            },
            {
                srcPath: '/_api/services-server/v1/batch/categories',
                destPath: '/v1/batch/categories',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/bookings/v1/categories',
                destPath: '/v1/categories',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/bookings/v1/categories',
                destPath: '/v1/categories',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/bookings/v1/categories',
                destPath: '/v1/categories',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/categories-proxy',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/services-server/v1/categories',
                destPath: '/v1/categories',
            },
            {
                srcPath: '/_api/services-server/v1/batch/categories',
                destPath: '/v1/batch/categories',
            },
        ],
        'bookings._base_domain_': [
            {
                srcPath: '/_api/services-server/v1/categories',
                destPath: '/v1/categories',
            },
            {
                srcPath: '/_api/services-server/v1/batch/categories',
                destPath: '/v1/batch/categories',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_bookings_categories';
/** Retrieves all categories. */
function list(payload) {
    function __list({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.v1.category',
            method: 'GET',
            methodFqn: 'com.wixpress.bookings.services.api.v1.CategoriesService.List',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
                protoPath: '/v1/categories',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __list;
}
exports.list = list;
/** Creates a category. */
function create(payload) {
    function __create({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.v1.category',
            method: 'POST',
            methodFqn: 'com.wixpress.bookings.services.api.v1.CategoriesService.Create',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
                protoPath: '/v1/categories',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __create;
}
exports.create = create;
/**
 * Updates a category.
 *
 *
 * Each time the category is updated, revision increments by 1. You must include
 * the number of the existing revision when updating the category. This ensures
 * you're working with the latest service information and prevents unintended
 * overwrites.
 */
function update(payload) {
    function __update({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.v1.category',
            method: 'PUT',
            methodFqn: 'com.wixpress.bookings.services.api.v1.CategoriesService.Update',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
                protoPath: '/v1/categories/{category.id}',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __update;
}
exports.update = update;
/**
 * Deletes a category.
 *
 *
 * You can specify `{"deleteServices": true}` to also delete all associated
 * services. Learn more about *deleting a service*
 * ([SDK](https://dev.wix.com/docs/sdk/backend-modules/bookings/services/delete-service) | [REST](https://dev.wix.com/docs/rest/business-solutions/bookings/services/services-v2/delete-service)).
 */
function _delete(payload) {
    function ___delete({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.v1.category',
            method: 'DELETE',
            methodFqn: 'com.wixpress.bookings.services.api.v1.CategoriesService._delete',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsServicesApiV1CategoriesServiceUrl({
                protoPath: '/v1/categories/{id}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return ___delete;
}
exports._delete = _delete;
