// @wix/auto_sdk_bookings_attendance/build/es/src/bookings-v2-attendance-attendance.public.js
import { bulkSetAttendance as universalBulkSetAttendance, getAttendance as universalGetAttendance, queryAttendance as universalQueryAttendance, setAttendance as universalSetAttendance, } from './bookings-v2-attendance-attendance.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/bookings' };
export function getAttendance(httpClient) {
    return (attendanceId) => universalGetAttendance(attendanceId, 
    // @ts-ignore
    { httpClient });
}
export function setAttendance(httpClient) {
    return (attendance, options) => universalSetAttendance(attendance, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkSetAttendance(httpClient) {
    return (options) => universalBulkSetAttendance(options, 
    // @ts-ignore
    { httpClient });
}
export function queryAttendance(httpClient) {
    return () => universalQueryAttendance(
    // @ts-ignore
    { httpClient });
}
export { AttendanceStatus, SortOrder, WebhookIdentityType, } from './bookings-v2-attendance-attendance.universal.js';
