// @wix/auto_sdk_bookings_booking-fees/build/internal/cjs/src/bookings-fees-v1-booking-fee-booking-fees.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Trigger = exports.BookingFeeStatus = exports.collectAppliedBookingFees = exports.applyBookingFeesToOrder = exports.listBookingFeesByBookingIds = exports.__metadata = void 0;
const bookings_fees_v1_booking_fee_booking_fees_universal_js_1 = require("./bookings-fees-v1-booking-fee-booking-fees.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function listBookingFeesByBookingIds(httpClient) {
    return (options) => (0, bookings_fees_v1_booking_fee_booking_fees_universal_js_1.listBookingFeesByBookingIds)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listBookingFeesByBookingIds = listBookingFeesByBookingIds;
function applyBookingFeesToOrder(httpClient) {
    return (bookingIds, options) => (0, bookings_fees_v1_booking_fee_booking_fees_universal_js_1.applyBookingFeesToOrder)(bookingIds, options, 
    // @ts-ignore
    { httpClient });
}
exports.applyBookingFeesToOrder = applyBookingFeesToOrder;
function collectAppliedBookingFees(httpClient) {
    return (orderId, options) => (0, bookings_fees_v1_booking_fee_booking_fees_universal_js_1.collectAppliedBookingFees)(orderId, options, 
    // @ts-ignore
    { httpClient });
}
exports.collectAppliedBookingFees = collectAppliedBookingFees;
var bookings_fees_v1_booking_fee_booking_fees_universal_js_2 = require("./bookings-fees-v1-booking-fee-booking-fees.universal.js");
Object.defineProperty(exports, "BookingFeeStatus", { enumerable: true, get: function () { return bookings_fees_v1_booking_fee_booking_fees_universal_js_2.BookingFeeStatus; } });
Object.defineProperty(exports, "Trigger", { enumerable: true, get: function () { return bookings_fees_v1_booking_fee_booking_fees_universal_js_2.Trigger; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_fees_v1_booking_fee_booking_fees_universal_js_2.WebhookIdentityType; } });
