// @wix/auto_sdk_members_custom-field-applications/build/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  createCustomFieldApplication: () => createCustomFieldApplication2,
  deleteCustomFieldApplication: () => deleteCustomFieldApplication2,
  getCustomFieldApplication: () => getCustomFieldApplication2,
  getCustomFieldApplications: () => getCustomFieldApplications2,
  getMembersCustomFieldApplications: () => getMembersCustomFieldApplications2,
  getRolesCustomFieldApplications: () => getRolesCustomFieldApplications2,
  updateCustomFieldApplication: () => updateCustomFieldApplication2
});
module.exports = __toCommonJS(meta_exports);

// src/members-v1-custom-field-application-custom-field-applications.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-field-applications";
function createCustomFieldApplication(payload) {
  function __createCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.CreateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        { protoPath: "/v1/custom-fields-applications", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __createCustomFieldApplication;
}
function deleteCustomFieldApplication(payload) {
  function __deleteCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "DELETE",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.DeleteCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteCustomFieldApplication;
}
function updateCustomFieldApplication(payload) {
  function __updateCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "PATCH",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.UpdateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{application.customFieldId}",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __updateCustomFieldApplication;
}
function getCustomFieldApplication(payload) {
  function __getCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getCustomFieldApplication;
}
function getCustomFieldApplications(payload) {
  function __getCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/applications",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __getCustomFieldApplications;
}
function getMembersCustomFieldApplications(payload) {
  function __getMembersCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetMembersCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/members",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMembersCustomFieldApplications;
}
function getRolesCustomFieldApplications(payload) {
  function __getRolesCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetRolesCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/roles",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRolesCustomFieldApplications;
}

// src/members-v1-custom-field-application-custom-field-applications.meta.ts
function createCustomFieldApplication2() {
  const payload = {};
  const getRequestOptions = createCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCustomFieldApplication2() {
  const payload = { customFieldId: ":customFieldId" };
  const getRequestOptions = deleteCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/custom-fields-applications/{customFieldId}",
    pathParams: { customFieldId: "customFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCustomFieldApplication2() {
  const payload = {
    application: { customFieldId: ":applicationCustomFieldId" }
  };
  const getRequestOptions = updateCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/custom-fields-applications/{application.customFieldId}",
    pathParams: { applicationCustomFieldId: "applicationCustomFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomFieldApplication2() {
  const payload = { customFieldId: ":customFieldId" };
  const getRequestOptions = getCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/custom-fields-applications/{customFieldId}",
    pathParams: { customFieldId: "customFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/applications",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMembersCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getMembersCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRolesCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getRolesCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/roles",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createCustomFieldApplication,
  deleteCustomFieldApplication,
  getCustomFieldApplication,
  getCustomFieldApplications,
  getMembersCustomFieldApplications,
  getRolesCustomFieldApplications,
  updateCustomFieldApplication
});
