// @wix/auto_sdk_members_custom-field-applications/build/meta.mjs
// src/members-v1-custom-field-application-custom-field-applications.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(opts) {
  const domainToMappings = {
    _: [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/custom-fields-applications",
        destPath: "/v1/custom-fields-applications"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_custom-field-applications";
function createCustomFieldApplication(payload) {
  function __createCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.CreateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        { protoPath: "/v1/custom-fields-applications", data: payload, host }
      ),
      data: payload
    };
    return metadata;
  }
  return __createCustomFieldApplication;
}
function deleteCustomFieldApplication(payload) {
  function __deleteCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "DELETE",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.DeleteCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCustomFieldApplication;
}
function updateCustomFieldApplication(payload) {
  function __updateCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "PATCH",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.UpdateCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{application.customFieldId}",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __updateCustomFieldApplication;
}
function getCustomFieldApplication(payload) {
  function __getCustomFieldApplication({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "GET",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplication",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/{customFieldId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __getCustomFieldApplication;
}
function getCustomFieldApplications(payload) {
  function __getCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/applications",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __getCustomFieldApplications;
}
function getMembersCustomFieldApplications(payload) {
  function __getMembersCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetMembersCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/members",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getMembersCustomFieldApplications;
}
function getRolesCustomFieldApplications(payload) {
  function __getRolesCustomFieldApplications({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.custom_field_application",
      method: "POST",
      methodFqn: "com.wixpress.members.customfields.applications.CustomFieldApplications.GetRolesCustomFieldApplications",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersCustomfieldsApplicationsCustomFieldApplicationsUrl(
        {
          protoPath: "/v1/custom-fields-applications/roles",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.applications.customField.createdDate" },
            { path: "results.applications.customField.updatedDate" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getRolesCustomFieldApplications;
}

// src/members-v1-custom-field-application-custom-field-applications.meta.ts
function createCustomFieldApplication2() {
  const payload = {};
  const getRequestOptions = createCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCustomFieldApplication2() {
  const payload = { customFieldId: ":customFieldId" };
  const getRequestOptions = deleteCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v1/custom-fields-applications/{customFieldId}",
    pathParams: { customFieldId: "customFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCustomFieldApplication2() {
  const payload = {
    application: { customFieldId: ":applicationCustomFieldId" }
  };
  const getRequestOptions = updateCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v1/custom-fields-applications/{application.customFieldId}",
    pathParams: { applicationCustomFieldId: "applicationCustomFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomFieldApplication2() {
  const payload = { customFieldId: ":customFieldId" };
  const getRequestOptions = getCustomFieldApplication(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/custom-fields-applications/{customFieldId}",
    pathParams: { customFieldId: "customFieldId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/applications",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getMembersCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getMembersCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/members",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getRolesCustomFieldApplications2() {
  const payload = {};
  const getRequestOptions = getRolesCustomFieldApplications(
    payload
  );
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v1/custom-fields-applications/roles",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createCustomFieldApplication2 as createCustomFieldApplication,
  deleteCustomFieldApplication2 as deleteCustomFieldApplication,
  getCustomFieldApplication2 as getCustomFieldApplication,
  getCustomFieldApplications2 as getCustomFieldApplications,
  getMembersCustomFieldApplications2 as getMembersCustomFieldApplications,
  getRolesCustomFieldApplications2 as getRolesCustomFieldApplications,
  updateCustomFieldApplication2 as updateCustomFieldApplication
};
