// @wix/auto_sdk_bookings_service-categories/build/es/src/bookings-categories-v2-category-service-categories.context.js
import { createCategory as publicCreateCategory, getCategory as publicGetCategory, updateCategory as publicUpdateCategory, deleteCategory as publicDeleteCategory, queryCategories as publicQueryCategories, countCategories as publicCountCategories, moveCategory as publicMoveCategory, } from './bookings-categories-v2-category-service-categories.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onCategoryCreated as publicOnCategoryCreated } from './bookings-categories-v2-category-service-categories.public.js';
import { onCategoryDeleted as publicOnCategoryDeleted } from './bookings-categories-v2-category-service-categories.public.js';
import { onCategoryUpdated as publicOnCategoryUpdated } from './bookings-categories-v2-category-service-categories.public.js';
export const createCategory = /*#__PURE__*/ createRESTModule(publicCreateCategory);
export const getCategory = /*#__PURE__*/ createRESTModule(publicGetCategory);
export const updateCategory = /*#__PURE__*/ createRESTModule(publicUpdateCategory);
export const deleteCategory = /*#__PURE__*/ createRESTModule(publicDeleteCategory);
export const queryCategories = /*#__PURE__*/ createRESTModule(publicQueryCategories);
export const countCategories = /*#__PURE__*/ createRESTModule(publicCountCategories);
export const moveCategory = /*#__PURE__*/ createRESTModule(publicMoveCategory);
/** */
export const onCategoryCreated = createEventModule(publicOnCategoryCreated);
/** */
export const onCategoryDeleted = createEventModule(publicOnCategoryDeleted);
/** */
export const onCategoryUpdated = createEventModule(publicOnCategoryUpdated);
export { SortOrder, Position, WebhookIdentityType, } from './bookings-categories-v2-category-service-categories.universal.js';
