// @wix/auto_sdk_bookings_service-categories/build/es/src/bookings-categories-v2-category-service-categories.meta.js
import * as ambassadorWixBookingsCategoriesV2Category from './bookings-categories-v2-category-service-categories.http.js';
export function createCategory() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.createCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/categories',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.getCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/categories/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCategory() {
    const payload = { category: { id: ':categoryId' } };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.updateCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/categories/{category.id}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.deleteCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/categories/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.queryCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/categories/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.countCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/categories/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function moveCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBookingsCategoriesV2Category.moveCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/categories/set-position/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
