// @wix/auto_sdk_bookings_service-categories/build/internal/es/src/bookings-categories-v2-category-service-categories.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { countCategories as universalCountCategories, createCategory as universalCreateCategory, deleteCategory as universalDeleteCategory, getCategory as universalGetCategory, moveCategory as universalMoveCategory, queryCategories as universalQueryCategories, updateCategory as universalUpdateCategory, } from './bookings-categories-v2-category-service-categories.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/bookings' };
export function createCategory(httpClient) {
    return (category) => universalCreateCategory(category, 
    // @ts-ignore
    { httpClient });
}
export function getCategory(httpClient) {
    return (categoryId) => universalGetCategory(categoryId, 
    // @ts-ignore
    { httpClient });
}
export function updateCategory(httpClient) {
    return (_id, category) => universalUpdateCategory(_id, category, 
    // @ts-ignore
    { httpClient });
}
export function deleteCategory(httpClient) {
    return (categoryId) => universalDeleteCategory(categoryId, 
    // @ts-ignore
    { httpClient });
}
export function queryCategories(httpClient) {
    return () => universalQueryCategories(
    // @ts-ignore
    { httpClient });
}
export function countCategories(httpClient) {
    return (options) => universalCountCategories(options, 
    // @ts-ignore
    { httpClient });
}
export function moveCategory(httpClient) {
    return (categoryId, options) => universalMoveCategory(categoryId, options, 
    // @ts-ignore
    { httpClient });
}
export const onCategoryCreated = EventDefinition('wix.bookings.categories.v2.category_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onCategoryDeleted = EventDefinition('wix.bookings.categories.v2.category_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onCategoryUpdated = EventDefinition('wix.bookings.categories.v2.category_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export { Position, SortOrder, WebhookIdentityType, } from './bookings-categories-v2-category-service-categories.universal.js';
