// @wix/auto_sdk_bookings_bookings/build/internal/cjs/src/bookings-v2-booking-bookings.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.ValueType = exports.SortOrder = exports.SelectedPaymentOption = exports.Platform = exports.PaymentStatus = exports.MultiServiceBookingType = exports.LocationType = exports.LocationLocationType = exports.IdentityType = exports.IdentificationDataIdentityType = exports.BookingStatus = exports.Actor = exports.onBookingUpdated = exports.onBookingRescheduled = exports.onBookingNumberOfParticipantsUpdated = exports.onBookingDeclined = exports.onBookingCreated = exports.onBookingConfirmed = exports.onBookingCanceled = exports.removeBookingsFromMultiServiceBooking = exports.addBookingsToMultiServiceBooking = exports.getMultiServiceBooking = exports.bulkGetMultiServiceBookingAllowedActions = exports.declineMultiServiceBooking = exports.confirmMultiServiceBooking = exports.markMultiServiceBookingAsPending = exports.cancelMultiServiceBooking = exports.getMultiServiceBookingAvailability = exports.rescheduleMultiServiceBooking = exports.createMultiServiceBooking = exports.updateNumberOfParticipants = exports.cancelBooking = exports.declineBooking = exports.updateExtendedFields = exports.confirmBooking = exports.rescheduleBooking = exports.bulkCreateBooking = exports.createBooking = exports.bulkConfirmOrDeclineBooking = exports.confirmOrDeclineBooking = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const bookings_v2_booking_bookings_universal_js_1 = require("./bookings-v2-booking-bookings.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/bookings' };
function confirmOrDeclineBooking(httpClient) {
    return (bookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.confirmOrDeclineBooking)(bookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.confirmOrDeclineBooking = confirmOrDeclineBooking;
function bulkConfirmOrDeclineBooking(httpClient) {
    return (details, options) => (0, bookings_v2_booking_bookings_universal_js_1.bulkConfirmOrDeclineBooking)(details, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkConfirmOrDeclineBooking = bulkConfirmOrDeclineBooking;
function createBooking(httpClient) {
    return (booking, options) => (0, bookings_v2_booking_bookings_universal_js_1.createBooking)(booking, options, 
    // @ts-ignore
    { httpClient });
}
exports.createBooking = createBooking;
function bulkCreateBooking(httpClient) {
    return (createBookingsInfo, options) => (0, bookings_v2_booking_bookings_universal_js_1.bulkCreateBooking)(createBookingsInfo, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateBooking = bulkCreateBooking;
function rescheduleBooking(httpClient) {
    return (bookingId, slot, options) => (0, bookings_v2_booking_bookings_universal_js_1.rescheduleBooking)(bookingId, slot, options, 
    // @ts-ignore
    { httpClient });
}
exports.rescheduleBooking = rescheduleBooking;
function confirmBooking(httpClient) {
    return (bookingId, revision, options) => (0, bookings_v2_booking_bookings_universal_js_1.confirmBooking)(bookingId, revision, options, 
    // @ts-ignore
    { httpClient });
}
exports.confirmBooking = confirmBooking;
function updateExtendedFields(httpClient) {
    return (_id, namespace, options) => (0, bookings_v2_booking_bookings_universal_js_1.updateExtendedFields)(_id, namespace, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateExtendedFields = updateExtendedFields;
function declineBooking(httpClient) {
    return (bookingId, revision, options) => (0, bookings_v2_booking_bookings_universal_js_1.declineBooking)(bookingId, revision, options, 
    // @ts-ignore
    { httpClient });
}
exports.declineBooking = declineBooking;
function cancelBooking(httpClient) {
    return (bookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.cancelBooking)(bookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.cancelBooking = cancelBooking;
function updateNumberOfParticipants(httpClient) {
    return (bookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.updateNumberOfParticipants)(bookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateNumberOfParticipants = updateNumberOfParticipants;
function createMultiServiceBooking(httpClient) {
    return (bookings, options) => (0, bookings_v2_booking_bookings_universal_js_1.createMultiServiceBooking)(bookings, options, 
    // @ts-ignore
    { httpClient });
}
exports.createMultiServiceBooking = createMultiServiceBooking;
function rescheduleMultiServiceBooking(httpClient) {
    return (multiServiceBookingId, rescheduleBookingsInfo, options) => (0, bookings_v2_booking_bookings_universal_js_1.rescheduleMultiServiceBooking)(multiServiceBookingId, rescheduleBookingsInfo, options, 
    // @ts-ignore
    { httpClient });
}
exports.rescheduleMultiServiceBooking = rescheduleMultiServiceBooking;
function getMultiServiceBookingAvailability(httpClient) {
    return (multiServiceBookingId) => (0, bookings_v2_booking_bookings_universal_js_1.getMultiServiceBookingAvailability)(multiServiceBookingId, 
    // @ts-ignore
    { httpClient });
}
exports.getMultiServiceBookingAvailability = getMultiServiceBookingAvailability;
function cancelMultiServiceBooking(httpClient) {
    return (multiServiceBookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.cancelMultiServiceBooking)(multiServiceBookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.cancelMultiServiceBooking = cancelMultiServiceBooking;
function markMultiServiceBookingAsPending(httpClient) {
    return (multiServiceBookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.markMultiServiceBookingAsPending)(multiServiceBookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.markMultiServiceBookingAsPending = markMultiServiceBookingAsPending;
function confirmMultiServiceBooking(httpClient) {
    return (multiServiceBookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.confirmMultiServiceBooking)(multiServiceBookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.confirmMultiServiceBooking = confirmMultiServiceBooking;
function declineMultiServiceBooking(httpClient) {
    return (multiServiceBookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.declineMultiServiceBooking)(multiServiceBookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.declineMultiServiceBooking = declineMultiServiceBooking;
function bulkGetMultiServiceBookingAllowedActions(httpClient) {
    return (multiServiceBookingIds) => (0, bookings_v2_booking_bookings_universal_js_1.bulkGetMultiServiceBookingAllowedActions)(multiServiceBookingIds, 
    // @ts-ignore
    { httpClient });
}
exports.bulkGetMultiServiceBookingAllowedActions = bulkGetMultiServiceBookingAllowedActions;
function getMultiServiceBooking(httpClient) {
    return (multiServiceBookingId) => (0, bookings_v2_booking_bookings_universal_js_1.getMultiServiceBooking)(multiServiceBookingId, 
    // @ts-ignore
    { httpClient });
}
exports.getMultiServiceBooking = getMultiServiceBooking;
function addBookingsToMultiServiceBooking(httpClient) {
    return (multiServiceBookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.addBookingsToMultiServiceBooking)(multiServiceBookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.addBookingsToMultiServiceBooking = addBookingsToMultiServiceBooking;
function removeBookingsFromMultiServiceBooking(httpClient) {
    return (multiServiceBookingId, options) => (0, bookings_v2_booking_bookings_universal_js_1.removeBookingsFromMultiServiceBooking)(multiServiceBookingId, options, 
    // @ts-ignore
    { httpClient });
}
exports.removeBookingsFromMultiServiceBooking = removeBookingsFromMultiServiceBooking;
exports.onBookingCanceled = (0, sdk_types_1.EventDefinition)('wix.bookings.v2.booking_canceled', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.booking.createdDate' },
            { path: 'data.booking.startDate' },
            { path: 'data.booking.endDate' },
            { path: 'data.booking.updatedDate' },
            { path: 'data.booking.canceledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.latitude',
            },
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.longitude',
            },
        ],
    },
])))();
exports.onBookingConfirmed = (0, sdk_types_1.EventDefinition)('wix.bookings.v2.booking_confirmed', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.booking.createdDate' },
            { path: 'data.booking.startDate' },
            { path: 'data.booking.endDate' },
            { path: 'data.booking.updatedDate' },
            { path: 'data.booking.canceledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.latitude',
            },
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.longitude',
            },
        ],
    },
])))();
exports.onBookingCreated = (0, sdk_types_1.EventDefinition)('wix.bookings.v2.booking_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.startDate' },
            { path: 'entity.endDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.canceledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.contactDetails.fullAddress.geocode.latitude' },
            { path: 'entity.contactDetails.fullAddress.geocode.longitude' },
        ],
    },
])))();
exports.onBookingDeclined = (0, sdk_types_1.EventDefinition)('wix.bookings.v2.booking_declined', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.booking.createdDate' },
            { path: 'data.booking.startDate' },
            { path: 'data.booking.endDate' },
            { path: 'data.booking.updatedDate' },
            { path: 'data.booking.canceledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.latitude',
            },
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.longitude',
            },
        ],
    },
])))();
exports.onBookingNumberOfParticipantsUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.v2.booking_number_of_participants_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.booking.createdDate' },
            { path: 'data.booking.startDate' },
            { path: 'data.booking.endDate' },
            { path: 'data.booking.updatedDate' },
            { path: 'data.booking.canceledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.latitude',
            },
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.longitude',
            },
        ],
    },
])))();
exports.onBookingRescheduled = (0, sdk_types_1.EventDefinition)('wix.bookings.v2.booking_rescheduled', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'data.previousStartDate' },
            { path: 'data.previousEndDate' },
            { path: 'data.booking.createdDate' },
            { path: 'data.booking.startDate' },
            { path: 'data.booking.endDate' },
            { path: 'data.booking.updatedDate' },
            { path: 'data.booking.canceledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.latitude',
            },
            {
                path: 'data.booking.contactDetails.fullAddress.geocode.longitude',
            },
        ],
    },
])))();
exports.onBookingUpdated = (0, sdk_types_1.EventDefinition)('wix.bookings.v2.booking_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.startDate' },
            { path: 'entity.endDate' },
            { path: 'entity.updatedDate' },
            { path: 'entity.canceledDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.contactDetails.fullAddress.geocode.latitude' },
            { path: 'entity.contactDetails.fullAddress.geocode.longitude' },
        ],
    },
])))();
var bookings_v2_booking_bookings_universal_js_2 = require("./bookings-v2-booking-bookings.universal.js");
Object.defineProperty(exports, "Actor", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.Actor; } });
Object.defineProperty(exports, "BookingStatus", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.BookingStatus; } });
Object.defineProperty(exports, "IdentificationDataIdentityType", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.IdentificationDataIdentityType; } });
Object.defineProperty(exports, "IdentityType", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.IdentityType; } });
Object.defineProperty(exports, "LocationLocationType", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.LocationLocationType; } });
Object.defineProperty(exports, "LocationType", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.LocationType; } });
Object.defineProperty(exports, "MultiServiceBookingType", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.MultiServiceBookingType; } });
Object.defineProperty(exports, "PaymentStatus", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.PaymentStatus; } });
Object.defineProperty(exports, "Platform", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.Platform; } });
Object.defineProperty(exports, "SelectedPaymentOption", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.SelectedPaymentOption; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "ValueType", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.ValueType; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return bookings_v2_booking_bookings_universal_js_2.WebhookIdentityType; } });
