// @wix/auto_sdk_bookings_bookings/build/internal/cjs/src/bookings-v2-booking-bookings.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removeBookingsFromMultiServiceBooking = exports.addBookingsToMultiServiceBooking = exports.getMultiServiceBooking = exports.bulkGetMultiServiceBookingAllowedActions = exports.declineMultiServiceBooking = exports.confirmMultiServiceBooking = exports.markMultiServiceBookingAsPending = exports.cancelMultiServiceBooking = exports.getMultiServiceBookingAvailability = exports.rescheduleMultiServiceBooking = exports.createMultiServiceBooking = exports.updateNumberOfParticipants = exports.cancelBooking = exports.declineBooking = exports.updateExtendedFields = exports.confirmBooking = exports.rescheduleBooking = exports.bulkCreateBooking = exports.createBooking = exports.bulkConfirmOrDeclineBooking = exports.confirmOrDeclineBooking = void 0;
const ambassadorWixBookingsV2Booking = __importStar(require("./bookings-v2-booking-bookings.http.js"));
function confirmOrDeclineBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.confirmOrDeclineBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/confirmation/{bookingId}:confirmOrDecline',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.confirmOrDeclineBooking = confirmOrDeclineBooking;
function bulkConfirmOrDeclineBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.bulkConfirmOrDeclineBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/bookings/confirmOrDecline',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkConfirmOrDeclineBooking = bulkConfirmOrDeclineBooking;
function createBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.createBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createBooking = createBooking;
function bulkCreateBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.bulkCreateBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/bookings/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateBooking = bulkCreateBooking;
function rescheduleBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.rescheduleBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/reschedule',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.rescheduleBooking = rescheduleBooking;
function confirmBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsConfirmBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/confirm',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.confirmBooking = confirmBooking;
function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{id}/update_extended_fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateExtendedFields = updateExtendedFields;
function declineBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsDeclineBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/decline',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.declineBooking = declineBooking;
function cancelBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsCancelBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/cancel',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.cancelBooking = cancelBooking;
function updateNumberOfParticipants() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsUpdateNumberOfParticipants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/update_number_of_participants',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateNumberOfParticipants = updateNumberOfParticipants;
function createMultiServiceBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.createMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createMultiServiceBooking = createMultiServiceBooking;
function rescheduleMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.rescheduleMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/reschedule',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.rescheduleMultiServiceBooking = rescheduleMultiServiceBooking;
function getMultiServiceBookingAvailability() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.getMultiServiceBookingAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/get_availability',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getMultiServiceBookingAvailability = getMultiServiceBookingAvailability;
function cancelMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.cancelMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/cancel',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.cancelMultiServiceBooking = cancelMultiServiceBooking;
function markMultiServiceBookingAsPending() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.markMultiServiceBookingAsPending(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/mark_as_pending',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.markMultiServiceBookingAsPending = markMultiServiceBookingAsPending;
function confirmMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.confirmMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/confirm',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.confirmMultiServiceBooking = confirmMultiServiceBooking;
function declineMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.declineMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/decline',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.declineMultiServiceBooking = declineMultiServiceBooking;
function bulkGetMultiServiceBookingAllowedActions() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.bulkGetMultiServiceBookingAllowedActions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/multi_service_bookings/get_allowed_actions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkGetMultiServiceBookingAllowedActions = bulkGetMultiServiceBookingAllowedActions;
function getMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.getMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getMultiServiceBooking = getMultiServiceBooking;
function addBookingsToMultiServiceBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.addBookingsToMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/add_bookings_to_multi_service_booking',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addBookingsToMultiServiceBooking = addBookingsToMultiServiceBooking;
function removeBookingsFromMultiServiceBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.removeBookingsFromMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/remove_bookings_from_multi_service_booking',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeBookingsFromMultiServiceBooking = removeBookingsFromMultiServiceBooking;
