// @wix/auto_sdk_ecom_tax-regions/build/cjs/src/billing-v1-tax-region-tax-regions.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.onTaxRegionUpdated = exports.onTaxRegionDeleted = exports.onTaxRegionCreated = exports.queryTaxRegions = exports.deleteTaxRegion = exports.updateTaxRegion = exports.getTaxRegion = exports.createTaxRegion = void 0;
const billing_v1_tax_region_tax_regions_public_js_1 = require("./billing-v1-tax-region-tax-regions.public.js");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
const billing_v1_tax_region_tax_regions_public_js_2 = require("./billing-v1-tax-region-tax-regions.public.js");
const billing_v1_tax_region_tax_regions_public_js_3 = require("./billing-v1-tax-region-tax-regions.public.js");
const billing_v1_tax_region_tax_regions_public_js_4 = require("./billing-v1-tax-region-tax-regions.public.js");
exports.createTaxRegion = (0, rest_modules_1.createRESTModule)(billing_v1_tax_region_tax_regions_public_js_1.createTaxRegion);
exports.getTaxRegion = (0, rest_modules_1.createRESTModule)(billing_v1_tax_region_tax_regions_public_js_1.getTaxRegion);
exports.updateTaxRegion = (0, rest_modules_1.createRESTModule)(billing_v1_tax_region_tax_regions_public_js_1.updateTaxRegion);
exports.deleteTaxRegion = (0, rest_modules_1.createRESTModule)(billing_v1_tax_region_tax_regions_public_js_1.deleteTaxRegion);
exports.queryTaxRegions = (0, rest_modules_1.createRESTModule)(billing_v1_tax_region_tax_regions_public_js_1.queryTaxRegions);
/** */
exports.onTaxRegionCreated = (0, event_definition_modules_1.createEventModule)(billing_v1_tax_region_tax_regions_public_js_2.onTaxRegionCreated);
/**
 * Triggered when a tax region is deleted.
 */
exports.onTaxRegionDeleted = (0, event_definition_modules_1.createEventModule)(billing_v1_tax_region_tax_regions_public_js_3.onTaxRegionDeleted);
/**
 * Triggered when a tax region is updated.
 */
exports.onTaxRegionUpdated = (0, event_definition_modules_1.createEventModule)(billing_v1_tax_region_tax_regions_public_js_4.onTaxRegionUpdated);
var billing_v1_tax_region_tax_regions_universal_js_1 = require("./billing-v1-tax-region-tax-regions.universal.js");
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return billing_v1_tax_region_tax_regions_universal_js_1.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return billing_v1_tax_region_tax_regions_universal_js_1.WebhookIdentityType; } });
