// @wix/auto_sdk_bookings_services/build/es/src/bookings-services-v2-service-services.meta.js
import * as ambassadorWixBookingsServicesV2Service from './bookings-services-v2-service-services.http.js';
export function createService() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.createService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.getService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/services/{serviceId}',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateService() {
    const payload = { service: { id: ':serviceId' } };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.updateService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v2/services/{service.id}',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkUpdateServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateServicesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkUpdateServicesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/update-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.deleteService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/services/{serviceId}',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkDeleteServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/delete',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteServicesByFilter() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.bulkDeleteServicesByFilter(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/services/delete-by-filter',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function searchServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.searchServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/search',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryPolicies() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryPolicies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/policies/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryBookingForms() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryBookingForms(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/booking-forms/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function countServices() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.countServices(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryLocations() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryLocations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/locations/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.queryCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/categories/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setServiceLocations() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.setServiceLocations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/locations',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function enablePricingPlansForService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.enablePricingPlansForService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/pricing-plans/add',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function disablePricingPlansForService() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.disablePricingPlansForService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/pricing-plans/remove',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function setCustomSlug() {
    const payload = { serviceId: ':serviceId' };
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.setCustomSlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/{serviceId}/slugs/custom',
        pathParams: { serviceId: 'serviceId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function validateSlug() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.validateSlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/slugs/validate',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cloneService() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsServicesV2Service.cloneService(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/services/clone',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
