// @wix/auto_sdk_bookings_bookings/build/es/src/bookings-v2-booking-bookings.meta.js
import * as ambassadorWixBookingsV2Booking from './bookings-v2-booking-bookings.http.js';
export function confirmOrDeclineBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.confirmOrDeclineBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/confirmation/{bookingId}:confirmOrDecline',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkConfirmOrDeclineBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.bulkConfirmOrDeclineBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/bookings/confirmOrDecline',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.createBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.bulkCreateBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/bookings/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function rescheduleBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.rescheduleBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/reschedule',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function confirmBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsConfirmBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/confirm',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateExtendedFields() {
    const payload = { id: ':id' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.updateExtendedFields(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{id}/update_extended_fields',
        pathParams: { id: 'id' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function declineBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsDeclineBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/decline',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cancelBooking() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsCancelBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/cancel',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateNumberOfParticipants() {
    const payload = { bookingId: ':bookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.bookingsUpdateNumberOfParticipants(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bookings/{bookingId}/update_number_of_participants',
        pathParams: { bookingId: 'bookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createMultiServiceBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.createMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function rescheduleMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.rescheduleMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/reschedule',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getMultiServiceBookingAvailability() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.getMultiServiceBookingAvailability(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/get_availability',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cancelMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.cancelMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/cancel',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function markMultiServiceBookingAsPending() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.markMultiServiceBookingAsPending(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/mark_as_pending',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function confirmMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.confirmMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/confirm',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function declineMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.declineMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}/decline',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkGetMultiServiceBookingAllowedActions() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.bulkGetMultiServiceBookingAllowedActions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/bulk/multi_service_bookings/get_allowed_actions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getMultiServiceBooking() {
    const payload = { multiServiceBookingId: ':multiServiceBookingId' };
    const getRequestOptions = ambassadorWixBookingsV2Booking.getMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/multi_service_bookings/{multiServiceBookingId}',
        pathParams: { multiServiceBookingId: 'multiServiceBookingId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addBookingsToMultiServiceBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.addBookingsToMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/add_bookings_to_multi_service_booking',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeBookingsFromMultiServiceBooking() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV2Booking.removeBookingsFromMultiServiceBooking(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/multi_service_bookings/remove_bookings_from_multi_service_booking',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
