// @wix/auto_sdk_bookings_extended-bookings/build/internal/cjs/src/bookings-reader-v2-extended-booking-extended-bookings.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.countExtendedBookings = exports.queryExtendedBookings = exports.query = void 0;
const float_1 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressBookingsReaderV2BookingsReaderUrl(opts) {
    const domainToMappings = {
        _: [
            {
                srcPath: '/_api/bookings-reader',
                destPath: '',
            },
        ],
        'editor._base_domain': [
            {
                srcPath: '/_api/bookings-reader',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/bookings-reader',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/_api/bookings-reader',
                destPath: '',
            },
            {
                srcPath: '/bookings/bookings-reader',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/bookings-reader',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/bookings-reader',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/bookings-reader',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_1.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_bookings_extended-bookings';
/**
 * > **Deprecation Notice**
 * >
 * > **This endpoint has been replaced with [Query Extended Bookings](https://dev.wix.com/api/rest/wix-bookings/bookings-reader-v2/query-extended-bookings) and will be removed on May 31, 2025.**
 * > **If your app uses this endpoint, we recommend updating your code as soon as possible.**
 *
 *
 * Retrieves a list of bookings, given the provided paging, filtering, and sorting.
 *
 *
 * You can also retrieve information about which actions the customer can perform
 * for the bookings. To do so, pass `withBookingAllowedActions` as `true`.
 *
 * Query Bookings runs with these defaults:
 *
 * - `createdDate` sorted in `DESC` order
 * - `cursorPaging.limit` is `50`
 *
 * You can check the overview about all
 * [supported filters](https://dev.wix.com/api/rest/wix-bookings/bookings-reader-v2/supported-filters)
 * for more information.
 *
 * `query.fields` and `query.fieldsets` aren't supported for this endpoint.
 *
 * When using filters for dates, you must use [UTC time](https://en.wikipedia.org/wiki/Coordinated_Universal_Time).
 *
 * Bookings belonging to a schedule don't have a `sessionId`. Therefore you
 * must use the `sessionId` filter that isn't part of the `query` object to
 * filter bookings for courses.
 *
 * To learn about working with _Query_ endpoints, see
 * [API Query Language](https://dev.wix.com/api/rest/getting-started/api-query-language),
 * [Sorting and Paging](https://dev.wix.com/api/rest/getting-started/sorting-and-paging),
 * and [Field Projection](https://dev.wix.com/api/rest/getting-started/field-projection).
 *
 * When calling Query Bookings, the retrieved data may not contain your most recent changes. See
 * [Wix-data and Eventual Consistency](reference/wix-data/introduction#wix-data_introduction_wix-data-and-eventual-consistency)
 * for more information.
 * @deprecated
 */
function query(payload) {
    function __query({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.reader.v2.extended_booking',
            method: 'POST',
            methodFqn: 'com.wixpress.bookings.reader.v2.BookingsReader.Query',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsReaderV2BookingsReaderUrl({
                protoPath: '/v2/extended-booking/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'extendedBookings.booking.createdDate' },
                        { path: 'extendedBookings.booking.startDate' },
                        { path: 'extendedBookings.booking.endDate' },
                        { path: 'extendedBookings.booking.updatedDate' },
                        { path: 'extendedBookings.booking.canceledDate' },
                    ],
                },
                {
                    transformFn: float_1.transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'extendedBookings.booking.contactDetails.fullAddress.geocode.latitude',
                        },
                        {
                            path: 'extendedBookings.booking.contactDetails.fullAddress.geocode.longitude',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __query;
}
exports.query = query;
/**
 * Retrieves a list of bookings, including additional extended information, given the provided paging, filtering, and sorting.
 *
 * `queryExtendedBookings()` doesn't use a query builder, instead it uses [API query language](https://dev.wix.com/docs/sdk/articles/work-with-the-sdk/api-query-language#with-the-api-query-language).
 *
 * Up to 100 extended bookings can be returned per request.
 *
 * `queryExtendedBookings()` runs with these defaults, which you can override:
 *
 * - `createdDate` sorted in `DESC` order
 * - `cursorPaging.limit` is `50`
 *
 *
 * You can retrieve information about which actions the customer can perform
 * for the bookings. To do so, pass `withBookingAllowedActions` as `true`.
 *
 *
 * For field support, see
 * [supported filters](https://www.wix.com/velo/reference/wix-bookings-v2/extendedbookings/supported-filters)
 * for more information.
 *
 * You can specify a filter only once per query. If you specify a filter
 * more than once, only the first filter determines the extended bookings that are returned.
 *
 * When filtering by date, you must use [UTC time](https://en.wikipedia.org/wiki/Coordinated_Universal_Time).
 */
function queryExtendedBookings(payload) {
    function __queryExtendedBookings({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.reader.v2.extended_booking',
            method: 'POST',
            methodFqn: 'com.wixpress.bookings.reader.v2.BookingsReader.QueryExtendedBookings',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsReaderV2BookingsReaderUrl({
                protoPath: '/v2/extended-bookings/query',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'extendedBookings.booking.createdDate' },
                        { path: 'extendedBookings.booking.startDate' },
                        { path: 'extendedBookings.booking.endDate' },
                        { path: 'extendedBookings.booking.updatedDate' },
                        { path: 'extendedBookings.booking.canceledDate' },
                    ],
                },
                {
                    transformFn: float_1.transformRESTFloatToSDKFloat,
                    paths: [
                        {
                            path: 'extendedBookings.booking.contactDetails.fullAddress.geocode.latitude',
                        },
                        {
                            path: 'extendedBookings.booking.contactDetails.fullAddress.geocode.longitude',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryExtendedBookings;
}
exports.queryExtendedBookings = queryExtendedBookings;
/**
 * Counts the number of bookings matching the specified filters.
 *
 *
 * For field support see
 * [supported filters](https://dev.wix.com/api/rest/wix-bookings/bookings-reader-v2/supported-filters)
 * for more information.
 */
function countExtendedBookings(payload) {
    function __countExtendedBookings({ host }) {
        const metadata = {
            entityFqdn: 'wix.bookings.reader.v2.extended_booking',
            method: 'POST',
            methodFqn: 'com.wixpress.bookings.reader.v2.BookingsReader.CountExtendedBookings',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressBookingsReaderV2BookingsReaderUrl({
                protoPath: '/v2/extended-bookings/count',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __countExtendedBookings;
}
exports.countExtendedBookings = countExtendedBookings;
