// @wix/auto_sdk_bookings_booking-policies/build/internal/cjs/src/bookings-v1-booking-policy-booking-policies.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.countBookingPolicies = exports.queryBookingPolicies = exports.deleteBookingPolicy = exports.setDefaultBookingPolicy = exports.updateBookingPolicy = exports.getStrictestBookingPolicy = exports.getBookingPolicy = exports.createBookingPolicy = void 0;
const ambassadorWixBookingsV1BookingPolicy = __importStar(require("./bookings-v1-booking-policy-booking-policies.http.js"));
function createBookingPolicy() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.createBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createBookingPolicy = createBookingPolicy;
function getBookingPolicy() {
    const payload = { bookingPolicyId: ':bookingPolicyId' };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.getBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/booking-policies/{bookingPolicyId}',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getBookingPolicy = getBookingPolicy;
function getStrictestBookingPolicy() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.getStrictestBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/strictest',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getStrictestBookingPolicy = getStrictestBookingPolicy;
function updateBookingPolicy() {
    const payload = { bookingPolicy: { id: ':bookingPolicyId' } };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.updateBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v1/booking-policies/{bookingPolicy.id}',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateBookingPolicy = updateBookingPolicy;
function setDefaultBookingPolicy() {
    const payload = { bookingPolicyId: ':bookingPolicyId' };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.setDefaultBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/{bookingPolicyId}:setDefault',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.setDefaultBookingPolicy = setDefaultBookingPolicy;
function deleteBookingPolicy() {
    const payload = { bookingPolicyId: ':bookingPolicyId' };
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.deleteBookingPolicy(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/booking-policies/{bookingPolicyId}',
        pathParams: { bookingPolicyId: 'bookingPolicyId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteBookingPolicy = deleteBookingPolicy;
function queryBookingPolicies() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.queryBookingPolicies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryBookingPolicies = queryBookingPolicies;
function countBookingPolicies() {
    const payload = {};
    const getRequestOptions = ambassadorWixBookingsV1BookingPolicy.countBookingPolicies(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/booking-policies/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.countBookingPolicies = countBookingPolicies;
