// @wix/auto_sdk_members_member-to-member-block/build/internal/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.ts
var index_exports = {};
__export(index_exports, {
  BlockDirection: () => BlockDirection,
  WebhookIdentityType: () => WebhookIdentityType,
  blockMember: () => blockMember4,
  listCurrentMemberBlocking: () => listCurrentMemberBlocking4,
  listMemberBlocks: () => listMemberBlocks4,
  unblockMember: () => unblockMember4
});
module.exports = __toCommonJS(index_exports);

// src/members-v1-member-to-member-block-member-to-member-block.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/members-v1-member-to-member-block-member-to-member-block.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/members/v1/member-to-member-blocks",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/members/v1/member-to-member-blocks",
        destPath: "/v1/member-to-member-blocks"
      }
    ],
    _: [
      {
        srcPath: "/_api/members/v1/member-to-member-blocks",
        destPath: "/v1/member-to-member-blocks"
      }
    ],
    "members.wixapps.net": [
      {
        srcPath: "/_api/members/v1/member-to-member-blocks",
        destPath: "/v1/member-to-member-blocks"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_members_member-to-member-block";
function blockMember(payload) {
  function __blockMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "POST",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.BlockMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/my",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __blockMember;
}
function unblockMember(payload) {
  function __unblockMember({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "DELETE",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.UnblockMember",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/my/{memberId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __unblockMember;
}
function listCurrentMemberBlocking(payload) {
  function __listCurrentMemberBlocking({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "GET",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.ListCurrentMemberBlocking",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/my",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listCurrentMemberBlocking;
}
function listMemberBlocks(payload) {
  function __listMemberBlocks({ host }) {
    const metadata = {
      entityFqdn: "wix.members.v1.member_to_member_block",
      method: "GET",
      methodFqn: "com.wixpress.members.membertomemberblocks.MemberToMemberBlocks.ListMemberBlocks",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressMembersMembertomemberblocksMemberToMemberBlocksUrl(
        {
          protoPath: "/v1/member-to-member-blocks/members/{memberId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __listMemberBlocks;
}

// src/members-v1-member-to-member-block-member-to-member-block.universal.ts
var BlockDirection = /* @__PURE__ */ ((BlockDirection2) => {
  BlockDirection2["UNKNOWN"] = "UNKNOWN";
  BlockDirection2["BLOCKING"] = "BLOCKING";
  BlockDirection2["BLOCKED"] = "BLOCKED";
  return BlockDirection2;
})(BlockDirection || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
async function blockMember2(memberId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ memberId });
  const reqOpts = blockMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function unblockMember2(memberId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({ memberId });
  const reqOpts = unblockMember(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { memberId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["memberId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listCurrentMemberBlocking2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listCurrentMemberBlocking(
    payload
  );
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { cursorPaging: "$[0].cursorPaging" },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listMemberBlocks2(memberId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)({
    memberId,
    cursorPaging: options?.cursorPaging
  });
  const reqOpts = listMemberBlocks(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          memberId: "$[0]",
          cursorPaging: "$[1].cursorPaging"
        },
        singleArgumentUnchanged: false
      },
      ["memberId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/members-v1-member-to-member-block-member-to-member-block.public.ts
function blockMember3(httpClient) {
  return (memberId) => blockMember2(
    memberId,
    // @ts-ignore
    { httpClient }
  );
}
function unblockMember3(httpClient) {
  return (memberId) => unblockMember2(
    memberId,
    // @ts-ignore
    { httpClient }
  );
}
function listCurrentMemberBlocking3(httpClient) {
  return (options) => listCurrentMemberBlocking2(
    options,
    // @ts-ignore
    { httpClient }
  );
}
function listMemberBlocks3(httpClient) {
  return (memberId, options) => listMemberBlocks2(
    memberId,
    options,
    // @ts-ignore
    { httpClient }
  );
}

// src/members-v1-member-to-member-block-member-to-member-block.context.ts
var import_rest_modules3 = require("@wix/sdk-runtime/rest-modules");
var blockMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(blockMember3);
var unblockMember4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(unblockMember3);
var listCurrentMemberBlocking4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listCurrentMemberBlocking3);
var listMemberBlocks4 = /* @__PURE__ */ (0, import_rest_modules3.createRESTModule)(listMemberBlocks3);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BlockDirection,
  WebhookIdentityType,
  blockMember,
  listCurrentMemberBlocking,
  listMemberBlocks,
  unblockMember
});
